{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "CLLeQw9Mxab4"
      },
      "source": [
        "## Ссылка на диск с картированными чтениями"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "iX7NcoC_CLY0"
      },
      "source": [
        "\n",
        "| | SRA accession  | Stage    |\n",
        "|---:|:-------------|:-----------|\n",
        "| 1 | SRR5836473  | 8 cell       |\n",
        "| 2 | SRR3824222 | Epiblast    |\n",
        "| 3 | SRR5836475  | ICM       |\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "RQAK_lggxsXi"
      },
      "source": [
        "[Bam-файлы и отчеты](https://drive.google.com/drive/folders/1hGzGjuUZNLIeUsCdSxbEzgthcV_BaqMw?usp=sharing)"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "5l0a7C6JHe40"
      },
      "source": [
        "# Установка FastQC"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 1,
      "metadata": {
        "id": "esYQuHKO8nAg",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "e7b210ce-b2fd-42b8-e0f6-2b141cbfc63c"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2024-06-25 09:26:16--  https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip\n",
            "Resolving www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)... 149.155.133.4\n",
            "Connecting to www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)|149.155.133.4|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 10249221 (9.8M) [application/zip]\n",
            "Saving to: ‘fastqc_v0.11.9.zip’\n",
            "\n",
            "fastqc_v0.11.9.zip  100%[===================>]   9.77M  4.06MB/s    in 2.4s    \n",
            "\n",
            "2024-06-25 09:26:20 (4.06 MB/s) - ‘fastqc_v0.11.9.zip’ saved [10249221/10249221]\n",
            "\n",
            "Archive:  fastqc_v0.11.9.zip\n",
            "  inflating: FastQC/cisd-jhdf5.jar   \n",
            "   creating: FastQC/Configuration/\n",
            "  inflating: FastQC/Configuration/adapter_list.txt  \n",
            "  inflating: FastQC/Configuration/contaminant_list.txt  \n",
            "  inflating: FastQC/Configuration/limits.txt  \n",
            "  inflating: FastQC/fastqc           \n",
            "  inflating: FastQC/fastqc_icon.ico  \n",
            "   creating: FastQC/Help/\n",
            "   creating: FastQC/Help/1 Introduction/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/\n",
            "  inflating: FastQC/Help/1 Introduction/.svn/entries  \n",
            "   creating: FastQC/Help/1 Introduction/.svn/props/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/text-base/\n",
            "  inflating: FastQC/Help/1 Introduction/.svn/text-base/1.1 What is FastQC.html.svn-base  \n",
            "   creating: FastQC/Help/1 Introduction/.svn/tmp/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/tmp/props/\n",
            "  inflating: FastQC/Help/1 Introduction/1.1 What is FastQC.html  \n",
            "   creating: FastQC/Help/2 Basic Operations/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/\n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/entries  \n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/props/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/text-base/\n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.1 Opening a sequence file.html.svn-base  \n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.2 Evaluating Results.html.svn-base  \n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.3 Saving a Report.html.svn-base  \n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/tmp/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/tmp/props/\n",
            "  inflating: FastQC/Help/2 Basic Operations/2.1 Opening a sequence file.html  \n",
            "  inflating: FastQC/Help/2 Basic Operations/2.2 Evaluating Results.html  \n",
            "  inflating: FastQC/Help/2 Basic Operations/2.3 Saving a Report.html  \n",
            "   creating: FastQC/Help/3 Analysis Modules/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/entries  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/prop-base/\n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/duplication_levels.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/prop-base/kmer_profiles.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_gc_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_n_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_sequence_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_sequence_gc_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_sequence_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_tile_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/sequence_length_distribution.png.svn-base  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/props/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/text-base/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/1 Basic Statistics.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/10 Adapter Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/11 Kmer Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/12 Per Tile Sequence Quality.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/2 Per Base Sequence Quality.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/3 Per Sequence Quality Scores.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/4 Per Base Sequence Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/5 Per Sequence GC Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/6 Per Base N Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/7 Sequence Length Distribution.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/8 Duplicate Sequences.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/9 Overrepresented Sequences.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/duplication_levels.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/kmer_profiles.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_gc_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_n_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_sequence_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_sequence_gc_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_sequence_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_tile_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/sequence_length_distribution.png.svn-base  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/tmp/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/tmp/props/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/1 Basic Statistics.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/10 Adapter Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/11 Kmer Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/12 Per Tile Sequence Quality.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/2 Per Base Sequence Quality.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/3 Per Sequence Quality Scores.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/4 Per Base Sequence Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/5 Per Sequence GC Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/6 Per Base N Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/7 Sequence Length Distribution.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/8 Duplicate Sequences.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/9 Overrepresented Sequences.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/duplication_levels.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/kmer_profiles.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_gc_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_n_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_sequence_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_sequence_gc_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_sequence_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_tile_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/sequence_length_distribution.png  \n",
            "  inflating: FastQC/INSTALL.txt      \n",
            "  inflating: FastQC/jbzip2-0.9.jar   \n",
            "  inflating: FastQC/LICENSE          \n",
            "  inflating: FastQC/LICENSE.txt      \n",
            "  inflating: FastQC/LICENSE_JHDF5.txt  \n",
            "   creating: FastQC/net/\n",
            "   creating: FastQC/net/sourceforge/\n",
            "   creating: FastQC/net/sourceforge/iharder/\n",
            "   creating: FastQC/net/sourceforge/iharder/base64/\n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$1.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$InputStream.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$OutputStream.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64.class  \n",
            "   creating: FastQC/org/\n",
            "   creating: FastQC/org/apache/\n",
            "   creating: FastQC/org/apache/commons/\n",
            "   creating: FastQC/org/apache/commons/math3/\n",
            "   creating: FastQC/org/apache/commons/math3/analysis/\n",
            "   creating: FastQC/org/apache/commons/math3/analysis/solvers/\n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/AbstractUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/AllowedSolution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BaseAbstractUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BaseUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BracketedUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BrentSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/UnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/UnivariateSolverUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/UnivariateFunction.class  \n",
            "   creating: FastQC/org/apache/commons/math3/distribution/\n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractIntegerDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractRealDistribution$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractRealDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/BetaDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/BinomialDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/CauchyDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/ChiSquaredDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/FDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/GammaDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/HypergeometricDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/IntegerDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/NormalDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/PascalDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/PoissonDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/RealDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/SaddlePointExpansion.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/TDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/WeibullDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/ZipfDistribution.class  \n",
            "   creating: FastQC/org/apache/commons/math3/exception/\n",
            "  inflating: FastQC/org/apache/commons/math3/exception/ConvergenceException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/DimensionMismatchException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathArithmeticException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalArgumentException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalNumberException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalStateException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathInternalError.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MaxCountExceededException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NoBracketingException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotFiniteNumberException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotPositiveException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotStrictlyPositiveException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NullArgumentException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NumberIsTooLargeException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NumberIsTooSmallException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/OutOfRangeException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/TooManyEvaluationsException.class  \n",
            "   creating: FastQC/org/apache/commons/math3/exception/util/\n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ArgUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ExceptionContext.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ExceptionContextProvider.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/Localizable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/LocalizedFormats.class  \n",
            "   creating: FastQC/org/apache/commons/math3/random/\n",
            "  inflating: FastQC/org/apache/commons/math3/random/AbstractWell.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/BitsStreamGenerator.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomData.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomDataImpl.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomGenerator.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/Well19937c.class  \n",
            "   creating: FastQC/org/apache/commons/math3/special/\n",
            "  inflating: FastQC/org/apache/commons/math3/special/Beta$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Beta.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Erf.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Gamma$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Gamma.class  \n",
            "   creating: FastQC/org/apache/commons/math3/util/\n",
            "  inflating: FastQC/org/apache/commons/math3/util/ArithmeticUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/ContinuedFraction.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/DoubleArray.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$ExpFracTable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$ExpIntTable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$lnMant.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMathCalc.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMathLiteralArrays.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor$MaxCountExceededCallback.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/MathUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Precision.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/ResizableDoubleArray.class  \n",
            "  inflating: FastQC/README.md        \n",
            "  inflating: FastQC/README.txt       \n",
            "  inflating: FastQC/RELEASE_NOTES.txt  \n",
            "  inflating: FastQC/run_fastqc.bat   \n",
            "  inflating: FastQC/sam-1.103.jar    \n",
            "   creating: FastQC/Templates/\n",
            "  inflating: FastQC/Templates/fastqc2fo.xsl  \n",
            "  inflating: FastQC/Templates/header_template.html  \n",
            "   creating: FastQC/Templates/Icons/\n",
            " extracting: FastQC/Templates/Icons/error.png  \n",
            " extracting: FastQC/Templates/Icons/fastqc_icon.png  \n",
            " extracting: FastQC/Templates/Icons/tick.png  \n",
            " extracting: FastQC/Templates/Icons/warning.png  \n",
            "   creating: FastQC/uk/\n",
            "   creating: FastQC/uk/ac/\n",
            "   creating: FastQC/uk/ac/babraham/\n",
            "   creating: FastQC/uk/ac/babraham/FastQC/\n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Analysis/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisListener.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisQueue.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisRunner.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/OfflineRunner.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Dialogs/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/AboutDialog$1.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/AboutDialog.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/FastQCTitlePanel$SmoothJLabel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/FastQCTitlePanel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/WelcomePanel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCApplication$1.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCApplication.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCConfig.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCMenuBar.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/FileFilters/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/BAMFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/CasavaFastQFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/FastQFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/GobyFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/MappedBAMFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/SequenceFileFilter.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Graphs/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/BaseGroup.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/LineGraph.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/QualityBoxPlot.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/TileGraph.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Help/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpDialog.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpIndexRoot$FileSorter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpIndexRoot.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPage.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPageDisplay$HelpEditor.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPageDisplay.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpSearchPanel.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Modules/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AbstractQCModule.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent$Adapter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/BasicStats$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/BasicStats.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/DuplicationLevel.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/GCModel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/GCModelValue.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent$Kmer.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/ModuleConfig.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/ModuleFactory.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/NContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs$OverrepresentedSeq.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerBaseQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerBaseSequenceContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerSequenceGCContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerSequenceQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerTileQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/QCModule.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/SequenceLengthDistribution.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Report/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Report/HTMLReportArchive.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Report/stylesheet.txt  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Resources/\n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/error.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon.png  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon.svg  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon_100.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/tick.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/warning.png  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Results/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Results/ResultsPanel$ModuleRenderer.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Results/ResultsPanel.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/BAMFile.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/Contaminant.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/ContaminantHit.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/ContaminentFinder.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Fast5File.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/FastQFile.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/QualityEncoding/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/QualityEncoding/PhredEncoding.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Sequence.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFactory.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFile.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFileGroup.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFormatException.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Statistics/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Statistics/NormalDistribution.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Statistics/PearsonCorrelation.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Utilities/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/CasavaBasename.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/HotColdColourGradient.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/ImageToBase64.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/MultiMemberGZIPInputStream.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/NameFormatException.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/NanoporeBasename.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/QualityCount.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/RGB.class  \n"
          ]
        }
      ],
      "source": [
        "!wget https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip\n",
        "!unzip fastqc_v0.11.9.zip\n",
        "!chmod a+x FastQC/fastqc"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Скачайте любой из запусков и проведите анализ QC прочтений.Для скачивания можно воспользоваться SRAtoolkit или скачать с European Nucleotide Archive. Воспользуемся 2 способом"
      ],
      "metadata": {
        "id": "Jnt0wzNquNJX"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "Скачиваем и разархивируем"
      ],
      "metadata": {
        "id": "Eyk01UE9vERr"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "! wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR583/003/SRR5836473/SRR5836473_1.fastq.gz\n",
        "! gunzip SRR5836473_1.fastq.gz"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "DJ8xHfYQuKoL",
        "outputId": "ace2571d-f12e-4e5e-bdc9-9697ce05831b"
      },
      "execution_count": 2,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2024-06-25 09:29:36--  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR583/003/SRR5836473/SRR5836473_1.fastq.gz\n",
            "           => ‘SRR5836473_1.fastq.gz’\n",
            "Resolving ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)... 193.62.193.165\n",
            "Connecting to ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)|193.62.193.165|:21... connected.\n",
            "Logging in as anonymous ... Logged in!\n",
            "==> SYST ... done.    ==> PWD ... done.\n",
            "==> TYPE I ... done.  ==> CWD (1) /vol1/fastq/SRR583/003/SRR5836473 ... done.\n",
            "==> SIZE SRR5836473_1.fastq.gz ... 6856386252\n",
            "==> PASV ... done.    ==> RETR SRR5836473_1.fastq.gz ... done.\n",
            "Length: 6856386252 (6.4G) (unauthoritative)\n",
            "\n",
            "SRR5836473_1.fastq. 100%[===================>]   6.38G  16.4MB/s    in 7m 2s   \n",
            "\n",
            "2024-06-25 09:36:42 (15.5 MB/s) - ‘SRR5836473_1.fastq.gz’ saved [6856386252]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Применяем fastqc"
      ],
      "metadata": {
        "id": "9uQ6pHyxvHfk"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!ls SRR* | xargs -P 2 -tI{} ./FastQC/fastqc {}"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "OV4NJSwTwPoc",
        "outputId": "0676810e-b569-45d4-b78d-584609214652"
      },
      "execution_count": 3,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "./FastQC/fastqc SRR5836473_1.fastq\n",
            "Started analysis of SRR5836473_1.fastq\n",
            "Approx 5% complete for SRR5836473_1.fastq\n",
            "Approx 10% complete for SRR5836473_1.fastq\n",
            "Approx 15% complete for SRR5836473_1.fastq\n",
            "Approx 20% complete for SRR5836473_1.fastq\n",
            "Approx 25% complete for SRR5836473_1.fastq\n",
            "Approx 30% complete for SRR5836473_1.fastq\n",
            "Approx 35% complete for SRR5836473_1.fastq\n",
            "Approx 40% complete for SRR5836473_1.fastq\n",
            "Approx 45% complete for SRR5836473_1.fastq\n",
            "Approx 50% complete for SRR5836473_1.fastq\n",
            "Approx 55% complete for SRR5836473_1.fastq\n",
            "Approx 60% complete for SRR5836473_1.fastq\n",
            "Approx 65% complete for SRR5836473_1.fastq\n",
            "Approx 70% complete for SRR5836473_1.fastq\n",
            "Approx 75% complete for SRR5836473_1.fastq\n",
            "Approx 80% complete for SRR5836473_1.fastq\n",
            "Approx 85% complete for SRR5836473_1.fastq\n",
            "Approx 90% complete for SRR5836473_1.fastq\n",
            "Approx 95% complete for SRR5836473_1.fastq\n",
            "Analysis complete for SRR5836473_1.fastq\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Удаляем"
      ],
      "metadata": {
        "id": "i7DMT_-vvL4t"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!rm SRR5836473_1.fastq"
      ],
      "metadata": {
        "id": "NBzsb_CowV0l"
      },
      "execution_count": 4,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Теперь то же самое для второго файла, скачиваем:"
      ],
      "metadata": {
        "id": "xjQjJizlvPI6"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "! wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR583/003/SRR5836473/SRR5836473_2.fastq.gz\n",
        "! gunzip SRR5836473_2.fastq.gz"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "MtkIUDEPwaPM",
        "outputId": "2988b949-2d6e-48b7-e0ff-3fb58b302bad"
      },
      "execution_count": 5,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2024-06-25 10:09:28--  ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR583/003/SRR5836473/SRR5836473_2.fastq.gz\n",
            "           => ‘SRR5836473_2.fastq.gz’\n",
            "Resolving ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)... 193.62.193.165\n",
            "Connecting to ftp.sra.ebi.ac.uk (ftp.sra.ebi.ac.uk)|193.62.193.165|:21... connected.\n",
            "Logging in as anonymous ... Logged in!\n",
            "==> SYST ... done.    ==> PWD ... done.\n",
            "==> TYPE I ... done.  ==> CWD (1) /vol1/fastq/SRR583/003/SRR5836473 ... done.\n",
            "==> SIZE SRR5836473_2.fastq.gz ... 8127008133\n",
            "==> PASV ... done.    ==> RETR SRR5836473_2.fastq.gz ... done.\n",
            "Length: 8127008133 (7.6G) (unauthoritative)\n",
            "\n",
            "SRR5836473_2.fastq. 100%[===================>]   7.57G  17.1MB/s    in 7m 42s  \n",
            "\n",
            "2024-06-25 10:17:14 (16.8 MB/s) - ‘SRR5836473_2.fastq.gz’ saved [8127008133]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Применяем fastqc"
      ],
      "metadata": {
        "id": "g27FxTFQvTQU"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!ls SRR* | xargs -P 2 -tI{} ./FastQC/fastqc {}"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "7kn5PGshwpSB",
        "outputId": "405e603d-7e0b-49cf-cc97-887488c14abc"
      },
      "execution_count": 6,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "./FastQC/fastqc SRR5836473_1_fastqc.html\n",
            "./FastQC/fastqc SRR5836473_1_fastqc.zip\n",
            "Failed to process SRR5836473_1_fastqc.zip\n",
            "uk.ac.babraham.FastQC.Sequence.SequenceFormatException: ID line didn't start with '@'\n",
            "\tat uk.ac.babraham.FastQC.Sequence.FastQFile.readNext(FastQFile.java:158)\n",
            "\tat uk.ac.babraham.FastQC.Sequence.FastQFile.<init>(FastQFile.java:89)\n",
            "\tat uk.ac.babraham.FastQC.Sequence.SequenceFactory.getSequenceFile(SequenceFactory.java:106)\n",
            "\tat uk.ac.babraham.FastQC.Sequence.SequenceFactory.getSequenceFile(SequenceFactory.java:62)\n",
            "\tat uk.ac.babraham.FastQC.Analysis.OfflineRunner.processFile(OfflineRunner.java:159)\n",
            "\tat uk.ac.babraham.FastQC.Analysis.OfflineRunner.<init>(OfflineRunner.java:121)\n",
            "\tat uk.ac.babraham.FastQC.FastQCApplication.main(FastQCApplication.java:316)\n",
            "./FastQC/fastqc SRR5836473_2.fastq\n",
            "Failed to process SRR5836473_1_fastqc.html\n",
            "uk.ac.babraham.FastQC.Sequence.SequenceFormatException: ID line didn't start with '@'\n",
            "\tat uk.ac.babraham.FastQC.Sequence.FastQFile.readNext(FastQFile.java:158)\n",
            "\tat uk.ac.babraham.FastQC.Sequence.FastQFile.<init>(FastQFile.java:89)\n",
            "\tat uk.ac.babraham.FastQC.Sequence.SequenceFactory.getSequenceFile(SequenceFactory.java:106)\n",
            "\tat uk.ac.babraham.FastQC.Sequence.SequenceFactory.getSequenceFile(SequenceFactory.java:62)\n",
            "\tat uk.ac.babraham.FastQC.Analysis.OfflineRunner.processFile(OfflineRunner.java:159)\n",
            "\tat uk.ac.babraham.FastQC.Analysis.OfflineRunner.<init>(OfflineRunner.java:121)\n",
            "\tat uk.ac.babraham.FastQC.FastQCApplication.main(FastQCApplication.java:316)\n",
            "Started analysis of SRR5836473_2.fastq\n",
            "Approx 5% complete for SRR5836473_2.fastq\n",
            "Approx 10% complete for SRR5836473_2.fastq\n",
            "Approx 15% complete for SRR5836473_2.fastq\n",
            "Approx 20% complete for SRR5836473_2.fastq\n",
            "Approx 25% complete for SRR5836473_2.fastq\n",
            "Approx 30% complete for SRR5836473_2.fastq\n",
            "Approx 35% complete for SRR5836473_2.fastq\n",
            "Approx 40% complete for SRR5836473_2.fastq\n",
            "Approx 45% complete for SRR5836473_2.fastq\n",
            "Approx 50% complete for SRR5836473_2.fastq\n",
            "Approx 55% complete for SRR5836473_2.fastq\n",
            "Approx 60% complete for SRR5836473_2.fastq\n",
            "Approx 65% complete for SRR5836473_2.fastq\n",
            "Approx 70% complete for SRR5836473_2.fastq\n",
            "Approx 75% complete for SRR5836473_2.fastq\n",
            "Approx 80% complete for SRR5836473_2.fastq\n",
            "Approx 85% complete for SRR5836473_2.fastq\n",
            "Approx 90% complete for SRR5836473_2.fastq\n",
            "Approx 95% complete for SRR5836473_2.fastq\n",
            "Analysis complete for SRR5836473_2.fastq\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Удаляем"
      ],
      "metadata": {
        "id": "VvfQTyP4vWRM"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!rm SRR5836473_2.fastq"
      ],
      "metadata": {
        "id": "yxNuy4ERwmlh"
      },
      "execution_count": 7,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "Ugk3bV0LI5El"
      },
      "source": [
        "# Установка Bismark с зависимостями (+bowtie2, samtools)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 2,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "uSOZLyn5n5Nl",
        "outputId": "27ba341d-f6cc-4150-c289-039629cf054e"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\u001b[33m\n",
            "\u001b[0m✨🍰✨ Everything looks OK!\n"
          ]
        }
      ],
      "source": [
        "! pip install -q condacolab\n",
        "import condacolab\n",
        "condacolab.install()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 3,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "npb--DNtn52o",
        "outputId": "bc081da0-1761-460d-8b7b-d365584240f9"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "✨🍰✨ Everything looks OK!\n"
          ]
        }
      ],
      "source": [
        "import condacolab\n",
        "condacolab.check()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 4,
      "metadata": {
        "id": "MfBSN4YuoCL6",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "5a75750a-c2ef-4986-8c9f-a7bc1f3b2f91"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Channels:\n",
            " - bioconda\n",
            " - conda-forge\n",
            "Platform: linux-64\n",
            "Collecting package metadata (repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
            "Solving environment: / \b\b- \b\b\\ \b\b| \b\bdone\n",
            "\n",
            "\n",
            "==> WARNING: A newer version of conda exists. <==\n",
            "    current version: 23.11.0\n",
            "    latest version: 24.5.0\n",
            "\n",
            "Please update conda by running\n",
            "\n",
            "    $ conda update -n base -c conda-forge conda\n",
            "\n",
            "\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local\n",
            "\n",
            "  added / updated specs:\n",
            "    - bismark\n",
            "\n",
            "\n",
            "The following packages will be downloaded:\n",
            "\n",
            "    package                    |            build\n",
            "    ---------------------------|-----------------\n",
            "    bismark-0.24.2             |       hdfd78af_0         7.7 MB  bioconda\n",
            "    bowtie2-2.5.3              |  py310ha0a81b8_1        14.1 MB  bioconda\n",
            "    ca-certificates-2024.6.2   |       hbcca054_0         152 KB  conda-forge\n",
            "    certifi-2024.6.2           |     pyhd8ed1ab_0         157 KB  conda-forge\n",
            "    coreutils-8.25             |                1         9.1 MB  bioconda\n",
            "    hisat2-2.2.1               |       hdbdd923_6        16.2 MB  bioconda\n",
            "    htslib-1.19.1              |       h81da01d_2         2.9 MB  bioconda\n",
            "    k8-0.2.5                   |       hdcf5f25_4         1.7 MB  bioconda\n",
            "    libdeflate-1.20            |       hd590300_0          70 KB  conda-forge\n",
            "    libgcc-7.2.0               |       h69d50b8_2         304 KB  conda-forge\n",
            "    libhwloc-2.9.3             |default_h554bfaf_1009         2.5 MB  conda-forge\n",
            "    libxcrypt-4.4.36           |       hd590300_1          98 KB  conda-forge\n",
            "    minimap2-2.28              |       he4a0461_1         1.1 MB  bioconda\n",
            "    openssl-3.3.1              |       h4ab18f5_0         2.8 MB  conda-forge\n",
            "    perl-5.32.1                | 7_hd590300_perl5        12.7 MB  conda-forge\n",
            "    samtools-1.19.2            |       h50ea8bc_1         453 KB  bioconda\n",
            "    tbb-2021.11.0              |       h00ab1b0_1         191 KB  conda-forge\n",
            "    zlib-1.2.13                |       hd590300_5          91 KB  conda-forge\n",
            "    ------------------------------------------------------------\n",
            "                                           Total:        72.2 MB\n",
            "\n",
            "The following NEW packages will be INSTALLED:\n",
            "\n",
            "  bismark            bioconda/noarch::bismark-0.24.2-hdfd78af_0 \n",
            "  bowtie2            bioconda/linux-64::bowtie2-2.5.3-py310ha0a81b8_1 \n",
            "  coreutils          bioconda/linux-64::coreutils-8.25-1 \n",
            "  hisat2             bioconda/linux-64::hisat2-2.2.1-hdbdd923_6 \n",
            "  htslib             bioconda/linux-64::htslib-1.19.1-h81da01d_2 \n",
            "  k8                 bioconda/linux-64::k8-0.2.5-hdcf5f25_4 \n",
            "  libdeflate         conda-forge/linux-64::libdeflate-1.20-hd590300_0 \n",
            "  libgcc             conda-forge/linux-64::libgcc-7.2.0-h69d50b8_2 \n",
            "  libhwloc           conda-forge/linux-64::libhwloc-2.9.3-default_h554bfaf_1009 \n",
            "  libxcrypt          conda-forge/linux-64::libxcrypt-4.4.36-hd590300_1 \n",
            "  minimap2           bioconda/linux-64::minimap2-2.28-he4a0461_1 \n",
            "  perl               conda-forge/linux-64::perl-5.32.1-7_hd590300_perl5 \n",
            "  samtools           bioconda/linux-64::samtools-1.19.2-h50ea8bc_1 \n",
            "  tbb                conda-forge/linux-64::tbb-2021.11.0-h00ab1b0_1 \n",
            "  zlib               conda-forge/linux-64::zlib-1.2.13-hd590300_5 \n",
            "\n",
            "The following packages will be UPDATED:\n",
            "\n",
            "  ca-certificates                     2023.11.17-hbcca054_0 --> 2024.6.2-hbcca054_0 \n",
            "  certifi                           2023.11.17-pyhd8ed1ab_0 --> 2024.6.2-pyhd8ed1ab_0 \n",
            "  openssl                                  3.2.0-hd590300_1 --> 3.3.1-h4ab18f5_0 \n",
            "\n",
            "\n",
            "\n",
            "Downloading and Extracting Packages:\n",
            "hisat2-2.2.1         | 16.2 MB   | :   0% 0/1 [00:00<?, ?it/s]\n",
            "bowtie2-2.5.3        | 14.1 MB   | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\n",
            "\n",
            "perl-5.32.1          | 12.7 MB   | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "coreutils-8.25       | 9.1 MB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "bismark-0.24.2       | 7.7 MB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "htslib-1.19.1        | 2.9 MB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "openssl-3.3.1        | 2.8 MB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libhwloc-2.9.3       | 2.5 MB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "k8-0.2.5             | 1.7 MB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "minimap2-2.28        | 1.1 MB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "samtools-1.19.2      | 453 KB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libgcc-7.2.0         | 304 KB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "tbb-2021.11.0        | 191 KB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "certifi-2024.6.2     | 157 KB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "ca-certificates-2024 | 152 KB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libxcrypt-4.4.36     | 98 KB     | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "zlib-1.2.13          | 91 KB     | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libdeflate-1.20      | 70 KB     | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "bowtie2-2.5.3        | 14.1 MB   | :   0% 0.0011091937771141992/1 [00:00<03:01, 181.97s/it]\u001b[A\n",
            "\n",
            "perl-5.32.1          | 12.7 MB   | :   0% 0.0012277751453917628/1 [00:00<03:07, 187.44s/it]\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "hisat2-2.2.1         | 16.2 MB   | :   0% 0.0009635402010155163/1 [00:00<04:28, 269.15s/it]\n",
            "\n",
            "\n",
            "coreutils-8.25       | 9.1 MB    | :   0% 0.0017100384706469664/1 [00:00<02:36, 156.63s/it]\u001b[A\u001b[A\u001b[A\n",
            "bowtie2-2.5.3        | 14.1 MB   | :  30% 0.3017007073750622/1 [00:00<00:00,  1.25it/s]    \u001b[A\n",
            "\n",
            "perl-5.32.1          | 12.7 MB   | :  26% 0.25906055567766195/1 [00:00<00:00,  1.01s/it]   \u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "hisat2-2.2.1         | 16.2 MB   | :  13% 0.1300779271370947/1 [00:00<00:01,  2.21s/it]    \n",
            "\n",
            "\n",
            "coreutils-8.25       | 9.1 MB    | :  40% 0.4001490021313901/1 [00:00<00:00,  1.36it/s]    \u001b[A\u001b[A\u001b[A\n",
            "bowtie2-2.5.3        | 14.1 MB   | :  48% 0.47917171171333406/1 [00:00<00:00,  1.38it/s]\u001b[A\n",
            "\n",
            "perl-5.32.1          | 12.7 MB   | :  56% 0.5586376911532521/1 [00:00<00:00,  1.67it/s] \u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "hisat2-2.2.1         | 16.2 MB   | :  36% 0.3603640351798031/1 [00:00<00:00,  1.06it/s]\n",
            "\n",
            "\n",
            "coreutils-8.25       | 9.1 MB    | :  77% 0.7678072733204878/1 [00:00<00:00,  2.12it/s]\u001b[A\u001b[A\u001b[A\n",
            "bowtie2-2.5.3        | 14.1 MB   | :  71% 0.7065564360217449/1 [00:00<00:00,  1.64it/s] \u001b[A\n",
            "\n",
            "hisat2-2.2.1         | 16.2 MB   | :  57% 0.5704157990011857/1 [00:00<00:00,  1.36it/s]\n",
            "hisat2-2.2.1         | 16.2 MB   | :  81% 0.8074466884510026/1 [00:00<00:00,  1.62it/s]\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "htslib-1.19.1        | 2.9 MB    | :   1% 0.005438038588698877/1 [00:00<02:28, 148.85s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "openssl-3.3.1        | 2.8 MB    | :   1% 0.005657126480835034/1 [00:00<02:31, 152.83s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "htslib-1.19.1        | 2.9 MB    | :  88% 0.880962251369218/1 [00:00<00:00,  1.31it/s]    \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libhwloc-2.9.3       | 2.5 MB    | :   1% 0.006263799072128885/1 [00:01<02:52, 173.51s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "k8-0.2.5             | 1.7 MB    | :   1% 0.009349703684470327/1 [00:01<01:55, 116.73s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "minimap2-2.28        | 1.1 MB    | :   1% 0.014633667080501246/1 [00:01<01:14, 75.54s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libgcc-7.2.0         | 304 KB    | :   5% 0.05255779785906529/1 [00:01<00:22, 23.44s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "tbb-2021.11.0        | 191 KB    | :   8% 0.08378848317479799/1 [00:01<00:13, 15.28s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "certifi-2024.6.2     | 157 KB    | :  10% 0.10205365540696262/1 [00:01<00:11, 12.59s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "ca-certificates-2024 | 152 KB    | :  11% 0.10500208286602365/1 [00:01<00:11, 12.47s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libxcrypt-4.4.36     | 98 KB     | :  16% 0.163198629386511/1 [00:01<00:06,  8.02s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "zlib-1.2.13          | 91 KB     | :  18% 0.1765041745219499/1 [00:01<00:06,  7.46s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libdeflate-1.20      | 70 KB     | :  23% 0.22914685314685315/1 [00:01<00:04,  5.78s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "samtools-1.19.2      | 453 KB    | :   4% 0.035356216470507/1 [00:01<00:38, 40.39s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "samtools-1.19.2      | 453 KB    | :  35% 0.35356216470506996/1 [00:01<00:02,  3.25s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "bismark-0.24.2       | 7.7 MB    | : 100% 1.0/1 [00:03<00:00,  4.17s/it]               \u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "bismark-0.24.2       | 7.7 MB    | : 100% 1.0/1 [00:03<00:00,  4.17s/it]\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "coreutils-8.25       | 9.1 MB    | : 100% 1.0/1 [00:04<00:00,  2.12it/s]               \u001b[A\u001b[A\u001b[A\n",
            "\n",
            "perl-5.32.1          | 12.7 MB   | : 100% 1.0/1 [00:05<00:00,  2.01it/s]               \u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "openssl-3.3.1        | 2.8 MB    | : 100% 1.0/1 [00:05<00:00,  5.16s/it]                  \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "openssl-3.3.1        | 2.8 MB    | : 100% 1.0/1 [00:05<00:00,  5.16s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "htslib-1.19.1        | 2.9 MB    | : 100% 1.0/1 [00:05<00:00,  1.31it/s]              \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "minimap2-2.28        | 1.1 MB    | : 100% 1.0/1 [00:05<00:00,  5.43s/it]                 \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "minimap2-2.28        | 1.1 MB    | : 100% 1.0/1 [00:05<00:00,  5.43s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "k8-0.2.5             | 1.7 MB    | : 100% 1.0/1 [00:06<00:00,  5.81s/it]                  \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "k8-0.2.5             | 1.7 MB    | : 100% 1.0/1 [00:06<00:00,  5.81s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libgcc-7.2.0         | 304 KB    | : 100% 1.0/1 [00:06<00:00,  5.93s/it]                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libgcc-7.2.0         | 304 KB    | : 100% 1.0/1 [00:06<00:00,  5.93s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "bowtie2-2.5.3        | 14.1 MB   | : 100% 1.0/1 [00:08<00:00,  1.92it/s]               \u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libhwloc-2.9.3       | 2.5 MB    | : 100% 1.0/1 [00:09<00:00,  8.81s/it]                  \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libhwloc-2.9.3       | 2.5 MB    | : 100% 1.0/1 [00:09<00:00,  8.81s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "certifi-2024.6.2     | 157 KB    | : 100% 1.0/1 [00:09<00:00,  9.02s/it]                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "certifi-2024.6.2     | 157 KB    | : 100% 1.0/1 [00:09<00:00,  9.02s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "tbb-2021.11.0        | 191 KB    | : 100% 1.0/1 [00:09<00:00,  8.99s/it]                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "tbb-2021.11.0        | 191 KB    | : 100% 1.0/1 [00:09<00:00,  8.99s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libxcrypt-4.4.36     | 98 KB     | : 100% 1.0/1 [00:09<00:00,  9.23s/it]              \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libxcrypt-4.4.36     | 98 KB     | : 100% 1.0/1 [00:09<00:00,  9.23s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "ca-certificates-2024 | 152 KB    | : 100% 1.0/1 [00:09<00:00,  9.08s/it]                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "ca-certificates-2024 | 152 KB    | : 100% 1.0/1 [00:09<00:00,  9.08s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "zlib-1.2.13          | 91 KB     | : 100% 1.0/1 [00:09<00:00,  9.30s/it]               \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "zlib-1.2.13          | 91 KB     | : 100% 1.0/1 [00:09<00:00,  9.30s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libdeflate-1.20      | 70 KB     | : 100% 1.0/1 [00:09<00:00,  9.48s/it]                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "libdeflate-1.20      | 70 KB     | : 100% 1.0/1 [00:09<00:00,  9.48s/it]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "samtools-1.19.2      | 453 KB    | : 100% 1.0/1 [00:09<00:00,  9.74s/it]                \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \n",
            "                                                                        \u001b[A\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "Preparing transaction: - \b\b\\ \b\bdone\n",
            "Verifying transaction: / \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n",
            "Executing transaction: - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n"
          ]
        }
      ],
      "source": [
        "! conda install -c bioconda bismark"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 5,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "13VM_exnoHEl",
        "outputId": "13d28a8f-6934-4640-a1f8-b3d95f2e06ef"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "Usage: samtools view [options] <in.bam>|<in.sam>|<in.cram> [region ...]\n",
            "\n",
            "Output options:\n",
            "  -b, --bam                  Output BAM\n",
            "  -C, --cram                 Output CRAM (requires -T)\n",
            "  -1, --fast                 Use fast BAM compression (and default to --bam)\n",
            "  -u, --uncompressed         Uncompressed BAM output (and default to --bam)\n",
            "  -h, --with-header          Include header in SAM output\n",
            "  -H, --header-only          Print SAM header only (no alignments)\n",
            "      --no-header            Print SAM alignment records only [default]\n",
            "  -c, --count                Print only the count of matching records\n",
            "  -o, --output FILE          Write output to FILE [standard output]\n",
            "  -U, --unoutput FILE, --output-unselected FILE\n",
            "                             Output reads not selected by filters to FILE\n",
            "  -p, --unmap                Set flag to UNMAP on reads not selected\n",
            "                             then write to output file.\n",
            "  -P, --fetch-pairs          Retrieve complete pairs even when outside of region\n",
            "Input options:\n",
            "  -t, --fai-reference FILE   FILE listing reference names and lengths\n",
            "  -M, --use-index            Use index and multi-region iterator for regions\n",
            "      --region[s]-file FILE  Use index to include only reads overlapping FILE\n",
            "  -X, --customized-index     Expect extra index file argument after <in.bam>\n",
            "\n",
            "Filtering options (Only include in output reads that...):\n",
            "  -L, --target[s]-file FILE  ...overlap (BED) regions in FILE\n",
            "  -N, --qname-file [^]FILE   ...whose read name is listed in FILE (\"^\" negates)\n",
            "  -r, --read-group STR       ...are in read group STR\n",
            "  -R, --read-group-file [^]FILE\n",
            "                             ...are in a read group listed in FILE\n",
            "  -d, --tag STR1[:STR2]      ...have a tag STR1 (with associated value STR2)\n",
            "  -D, --tag-file STR:FILE    ...have a tag STR whose value is listed in FILE\n",
            "  -q, --min-MQ INT           ...have mapping quality >= INT\n",
            "  -l, --library STR          ...are in library STR\n",
            "  -m, --min-qlen INT         ...cover >= INT query bases (as measured via CIGAR)\n",
            "  -e, --expr STR             ...match the filter expression STR\n",
            "  -f, --require-flags FLAG   ...have all of the FLAGs present\n",
            "  -F, --excl[ude]-flags FLAG ...have none of the FLAGs present\n",
            "      --rf, --incl-flags, --include-flags FLAG\n",
            "                             ...have some of the FLAGs present\n",
            "  -G FLAG                    EXCLUDE reads with all of the FLAGs present\n",
            "      --subsample FLOAT      Keep only FLOAT fraction of templates/read pairs\n",
            "      --subsample-seed INT   Influence WHICH reads are kept in subsampling [0]\n",
            "  -s INT.FRAC                Same as --subsample 0.FRAC --subsample-seed INT\n",
            "\n",
            "Processing options:\n",
            "      --add-flags FLAG       Add FLAGs to reads\n",
            "      --remove-flags FLAG    Remove FLAGs from reads\n",
            "  -x, --remove-tag STR\n",
            "               Comma-separated read tags to strip (repeatable) [null]\n",
            "      --keep-tag STR\n",
            "               Comma-separated read tags to preserve (repeatable) [null].\n",
            "               Equivalent to \"-x ^STR\"\n",
            "  -B, --remove-B             Collapse the backward CIGAR operation\n",
            "  -z, --sanitize FLAGS       Perform sanitity checking and fixing on records.\n",
            "                             FLAGS is comma separated (see manual). [off]\n",
            "\n",
            "General options:\n",
            "  -?, --help   Print long help, including note about region specification\n",
            "  -S           Ignored (input format is auto-detected)\n",
            "      --no-PG  Do not add a PG line\n",
            "      --input-fmt-option OPT[=VAL]\n",
            "               Specify a single input file format option in the form\n",
            "               of OPTION or OPTION=VALUE\n",
            "  -O, --output-fmt FORMAT[,OPT[=VAL]]...\n",
            "               Specify output format (SAM, BAM, CRAM)\n",
            "      --output-fmt-option OPT[=VAL]\n",
            "               Specify a single output file format option in the form\n",
            "               of OPTION or OPTION=VALUE\n",
            "  -T, --reference FILE\n",
            "               Reference sequence FASTA FILE [null]\n",
            "  -@, --threads INT\n",
            "               Number of additional threads to use [0]\n",
            "      --write-index\n",
            "               Automatically index the output files [off]\n",
            "      --verbosity INT\n",
            "               Set level of verbosity\n",
            "\n"
          ]
        }
      ],
      "source": [
        "! samtools view"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "1. Скачиваем чтения из SRA с помощью SRAtoolkit или любым другим способом\n",
        "\n",
        "2. Индексируем геном\n",
        "\n",
        "3. Строим выравнивание\n",
        "\n",
        "(Описание шагов 2-3 см https://www.bioinformatics.babraham.ac.uk/projects/bismark/Bismark_User_Guide.pdf)\n",
        "\n"
      ],
      "metadata": {
        "id": "XVeS3m8fWWNR"
      }
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "_zEx3USYPKoR"
      },
      "source": [
        "Можно получать данные секвенировия с ENA. Например, https://www.ebi.ac.uk/ena/browser/view/SRR5836473\n",
        "\n",
        "## но не нужно! Файлы там очень большие"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "c5QMGKJYxrza"
      },
      "outputs": [],
      "source": [
        "#! wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR583/003/SRR5836473/SRR5836473_1.fastq.gz\n",
        "#! wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR583/003/SRR5836473/SRR5836473_2.fastq.gz"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "#Внимание!\n",
        "\n",
        "Т.к. вычислительные мощности нужны большие, то этот шаг был сделан за вас.\n",
        " Достаточно скачать выравнивания с диска https://drive.google.com/drive/folders/1hGzGjuUZNLIeUsCdSxbEzgthcV_BaqMw?usp=sharing или с сервера майнора)"
      ],
      "metadata": {
        "id": "rtzJRhe7Uufp"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "Будем скачивать с диска"
      ],
      "metadata": {
        "id": "rnEm1EkoBWc8"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "! pip install -U --no-cache-dir gdown --pre"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "3eLRDGOfBZZg",
        "outputId": "b4140645-cae8-4320-9e9a-b54782fd923f"
      },
      "execution_count": 6,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting gdown\n",
            "  Downloading gdown-5.2.0-py3-none-any.whl.metadata (5.8 kB)\n",
            "Collecting beautifulsoup4 (from gdown)\n",
            "  Downloading beautifulsoup4-4.13.0b2-py3-none-any.whl.metadata (3.8 kB)\n",
            "Collecting filelock (from gdown)\n",
            "  Downloading filelock-3.15.4-py3-none-any.whl.metadata (2.9 kB)\n",
            "Requirement already satisfied: requests[socks] in /usr/local/lib/python3.10/site-packages (from gdown) (2.31.0)\n",
            "Requirement already satisfied: tqdm in /usr/local/lib/python3.10/site-packages (from gdown) (4.66.1)\n",
            "Collecting soupsieve>1.2 (from beautifulsoup4->gdown)\n",
            "  Downloading soupsieve-2.5-py3-none-any.whl.metadata (4.7 kB)\n",
            "Collecting typing-extensions (from beautifulsoup4->gdown)\n",
            "  Downloading typing_extensions-4.12.2-py3-none-any.whl.metadata (3.0 kB)\n",
            "Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.10/site-packages (from requests[socks]->gdown) (3.3.2)\n",
            "Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.10/site-packages (from requests[socks]->gdown) (3.6)\n",
            "Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.10/site-packages (from requests[socks]->gdown) (2.1.0)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.10/site-packages (from requests[socks]->gdown) (2024.6.2)\n",
            "Requirement already satisfied: PySocks!=1.5.7,>=1.5.6 in /usr/local/lib/python3.10/site-packages (from requests[socks]->gdown) (1.7.1)\n",
            "Downloading gdown-5.2.0-py3-none-any.whl (18 kB)\n",
            "Downloading beautifulsoup4-4.13.0b2-py3-none-any.whl (179 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m179.6/179.6 kB\u001b[0m \u001b[31m6.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading filelock-3.15.4-py3-none-any.whl (16 kB)\n",
            "Downloading soupsieve-2.5-py3-none-any.whl (36 kB)\n",
            "Downloading typing_extensions-4.12.2-py3-none-any.whl (37 kB)\n",
            "Installing collected packages: typing-extensions, soupsieve, filelock, beautifulsoup4, gdown\n",
            "Successfully installed beautifulsoup4-4.13.0b2 filelock-3.15.4 gdown-5.2.0 soupsieve-2.5 typing-extensions-4.12.2\n",
            "\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\u001b[33m\n",
            "\u001b[0m"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Качаем bam файл и отчет SRR5836473"
      ],
      "metadata": {
        "id": "yhKZpKmbDOQ-"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "! gdown --id 1h8kqppLRjNoK2uwcnXfK7iQbe_sXQmor\n",
        "! gdown --id 1ybuIMaW1BWdz-ehkzg7BmlZVWu4R6pte"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "n909mXZgBgN5",
        "outputId": "c9ff68a1-45f3-495a-fa44-7e67677a7331"
      },
      "execution_count": 7,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "/usr/local/lib/python3.10/site-packages/gdown/__main__.py:140: FutureWarning: Option `--id` was deprecated in version 4.3.1 and will be removed in 5.0. You don't need to pass it anymore to use a file ID.\n",
            "  warnings.warn(\n",
            "Downloading...\n",
            "From (original): https://drive.google.com/uc?id=1h8kqppLRjNoK2uwcnXfK7iQbe_sXQmor\n",
            "From (redirected): https://drive.google.com/uc?id=1h8kqppLRjNoK2uwcnXfK7iQbe_sXQmor&confirm=t&uuid=c6bce6e1-efc9-45d0-a5aa-77f9c3f6ffb8\n",
            "To: /content/SRR5836473_1_bismark_bt2_pe.bam\n",
            "100% 806M/806M [00:43<00:00, 18.5MB/s]\n",
            "/usr/local/lib/python3.10/site-packages/gdown/__main__.py:140: FutureWarning: Option `--id` was deprecated in version 4.3.1 and will be removed in 5.0. You don't need to pass it anymore to use a file ID.\n",
            "  warnings.warn(\n",
            "Downloading...\n",
            "From: https://drive.google.com/uc?id=1ybuIMaW1BWdz-ehkzg7BmlZVWu4R6pte\n",
            "To: /content/SRR5836473_1_bismark_bt2_PE_report.txt\n",
            "100% 1.74k/1.74k [00:00<00:00, 7.58MB/s]\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Качаем bam файл и отчет SRR5836475"
      ],
      "metadata": {
        "id": "nD1mIwTMDbuD"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "! gdown --id 1QE-3m9IoJehidSrLM7Npg8larZ4Tt1Ls\n",
        "! gdown --id 1L-kZURPu5A7QHxEz8d2pnH-cfaWlYggP"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "aBUMeLMZBgfO",
        "outputId": "883fb244-8cb6-4acd-b0e2-8730e8965c05"
      },
      "execution_count": 8,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "/usr/local/lib/python3.10/site-packages/gdown/__main__.py:140: FutureWarning: Option `--id` was deprecated in version 4.3.1 and will be removed in 5.0. You don't need to pass it anymore to use a file ID.\n",
            "  warnings.warn(\n",
            "Downloading...\n",
            "From (original): https://drive.google.com/uc?id=1QE-3m9IoJehidSrLM7Npg8larZ4Tt1Ls\n",
            "From (redirected): https://drive.google.com/uc?id=1QE-3m9IoJehidSrLM7Npg8larZ4Tt1Ls&confirm=t&uuid=8189969e-3990-40e3-86d2-688fb806b809\n",
            "To: /content/SRR5836475_1_bismark_bt2_pe.bam\n",
            "100% 1.20G/1.20G [00:38<00:00, 31.2MB/s]\n",
            "/usr/local/lib/python3.10/site-packages/gdown/__main__.py:140: FutureWarning: Option `--id` was deprecated in version 4.3.1 and will be removed in 5.0. You don't need to pass it anymore to use a file ID.\n",
            "  warnings.warn(\n",
            "Downloading...\n",
            "From: https://drive.google.com/uc?id=1L-kZURPu5A7QHxEz8d2pnH-cfaWlYggP\n",
            "To: /content/SRR5836475_1_bismark_bt2_PE_report.txt\n",
            "100% 1.74k/1.74k [00:00<00:00, 6.36MB/s]\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Качаем bam файл и отчет SRR3824222"
      ],
      "metadata": {
        "id": "4oSynt2SDco1"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "! gdown --id 1vZUsitBBYAYaVHbfonRuoHEK_toNZnN5\n",
        "! gdown --id 1RC6FJEmdWwz8iGwBTJdEcf08wwHHyp1x"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "CRnV87ZOBgqk",
        "outputId": "82ec152c-9412-43c9-ee10-b9b987f87103"
      },
      "execution_count": 9,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "/usr/local/lib/python3.10/site-packages/gdown/__main__.py:140: FutureWarning: Option `--id` was deprecated in version 4.3.1 and will be removed in 5.0. You don't need to pass it anymore to use a file ID.\n",
            "  warnings.warn(\n",
            "Downloading...\n",
            "From (original): https://drive.google.com/uc?id=1vZUsitBBYAYaVHbfonRuoHEK_toNZnN5\n",
            "From (redirected): https://drive.google.com/uc?id=1vZUsitBBYAYaVHbfonRuoHEK_toNZnN5&confirm=t&uuid=79ed56c9-236c-48d6-81ae-32da82afd25c\n",
            "To: /content/SRR3824222_1_bismark_bt2_pe.bam\n",
            "100% 2.04G/2.04G [01:20<00:00, 25.2MB/s]\n",
            "/usr/local/lib/python3.10/site-packages/gdown/__main__.py:140: FutureWarning: Option `--id` was deprecated in version 4.3.1 and will be removed in 5.0. You don't need to pass it anymore to use a file ID.\n",
            "  warnings.warn(\n",
            "Downloading...\n",
            "From: https://drive.google.com/uc?id=1RC6FJEmdWwz8iGwBTJdEcf08wwHHyp1x\n",
            "To: /content/SRR3824222_1_bismark_bt2_PE_report.txt\n",
            "100% 1.75k/1.75k [00:00<00:00, 5.45MB/s]\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 10,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "AxMpAQFVQFR8",
        "outputId": "d697b7d6-718d-4394-8b90-12038e10230c"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "total 3.8G\n",
            "-rw-r--r-- 1 root root  23K Jun 25 10:47 condacolab_install.log\n",
            "drwxr-xr-x 8 root root 4.0K Jun 25 09:26 FastQC\n",
            "-rw-r--r-- 1 root root 9.8M Jan 15  2020 fastqc_v0.11.9.zip\n",
            "drwxr-xr-x 1 root root 4.0K Jun 21 13:28 sample_data\n",
            "-rw-r--r-- 1 root root 1.9G Jan 30 08:06 SRR3824222_1_bismark_bt2_pe.bam\n",
            "-rw-r--r-- 1 root root 1.8K Jan 31 18:19 SRR3824222_1_bismark_bt2_PE_report.txt\n",
            "-rw-r--r-- 1 root root 769M Jan 30 07:20 SRR5836473_1_bismark_bt2_pe.bam\n",
            "-rw-r--r-- 1 root root 1.8K Jan 31 18:19 SRR5836473_1_bismark_bt2_PE_report.txt\n",
            "-rw-r--r-- 1 root root 570K Jun 25 10:01 SRR5836473_1_fastqc.html\n",
            "-rw-r--r-- 1 root root 296K Jun 25 10:01 SRR5836473_1_fastqc.zip\n",
            "-rw-r--r-- 1 root root 582K Jun 25 10:43 SRR5836473_2_fastqc.html\n",
            "-rw-r--r-- 1 root root 310K Jun 25 10:43 SRR5836473_2_fastqc.zip\n",
            "-rw-r--r-- 1 root root 1.2G Jan 30 07:01 SRR5836475_1_bismark_bt2_pe.bam\n",
            "-rw-r--r-- 1 root root 1.8K Jan 31 18:19 SRR5836475_1_bismark_bt2_PE_report.txt\n"
          ]
        }
      ],
      "source": [
        "! ls -lh"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 11,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "yMQy_LQ8xSOP",
        "outputId": "91c86858-a705-4b21-9192-7545eafef531"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "SRR5836473.41_41_length=140\t99\tchr11\t40076664\t0\t140M\t=\t40076703\t178\tAATTTTTTTGGGGATTGTATTGAATTTGTAGATTGTTTTTGGTAAGAGAGTTATTTTTATTATATTGATTTTGTTAATTTATGAATATGGGAGATTTTTTTATTTTTTGAGATTTTTTTTAATTTTTTTTTTTAGAGATT\tKKKKKKKKKKKKKKKFKKKKKKKKKKKKKKFKKKKKKKKKFKKKKKKKKKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKF7<77FF\tNM:i:28\tMD:Z:6G0A9C7C9C6C4T2C0C7C9C0C2C0C3C0C5C9C3C0C2C2C6C2C8C3C2C5C1\tXM:Z:.................h.......x.........h......h.......hh.......h.........hx..hh...hh.....h.........h...hh..h..x......h..h........h...h..x.....h.\tXR:Z:CT\tXG:Z:CT\n",
            "SRR5836473.41_41_length=140\t147\tchr11\t40076703\t0\t139M\t=\t40076664\t-178\tTGGTAAGAGAGTTATTTTTATTATATTGATTTTGTTAATTTATGAATATGGGAGATTTTTTTATTTTTTGAGATTTTTTTTAATTTTTTTTTTTAGAGATTTGAAGTTTTTATTATATAGATTTTTTATTTTTTTAGTG\tKKKKKAAFFFKKKKKKKKKKKKKFFKKKKKKKKKKFAFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFKKKKKKKKKKKKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKK\tNM:i:31\tMD:Z:3C4T2C0C7C9C0C2C0C3C0C5C9C3C0C2C2C6C2C8C3C2C5C8C4C3A4C3C1C2C6T0\tXM:Z:...h.......hh.......h.........hx..hh...hh.....h.........h...hh..h..x......h..h........h...h..x.....h........h....h........h...h.h..h.......\tXR:Z:GA\tXG:Z:CT\n",
            "SRR5836473.231_231_length=140\t83\tchr11\t48401999\t0\t140M\t=\t48401992\t-147\tACAAATTACTCCTTCTAATTCATTAAAAAATTAAATTAAAATTATAATAAAAATTACATTAAATCTATAAATTACTTTTAACAAAATAACCATTTTTACAATATTAATCCTACCAATCCATAAACACAAAAAAACTTTCC\tKKKKKKKKF<KKKKKKKKKFKKKKKKKKKKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFKKKKKKKKKKKKKKKK\tNM:i:34\tMD:Z:0G2G3G13G2G2G4G1G2G0G6G2G0G0G0G3G4G5G2G3G5G0G3G3G13G2G5G9G1G2T1G0G1G1T6\tXM:Z:h..x...h.............z..h..h....h.h..hh......h..hhhh...h....h.....x..h...h.....hh...h...h.............h..h.....x.........h.h....hh.h........\tXR:Z:CT\tXG:Z:GA\n",
            "SRR5836473.231_231_length=140\t163\tchr11\t48401992\t0\t139M\t=\t48401999\t147\tTAAATTTACAAATTACTCCTTCTAATTCATTAAAAAATTAAATTAAAATTATAATAAAAATTACATTAAATCTATAAATTACTTTTAACAAAATAACCATTTTTACAATATTAATCCTACCAATCCATAAACACAAAAA\tFFKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKAKFFKKKKKKKKKKKKKKKKFKAF<FF<FKKKKKKKKKKFKKKK<FFKAFF<FKF<FK7AF7A,7<AAF<KF\tNM:i:35\tMD:Z:1G3A1G2G3G13G2G2G4G1G2G0G6G2G0G0G0G3G4G5G2G3G5G0G3G3G13G2G5G9G1G2T1G0G1G0\tXM:Z:.h.....h..x...h.............z..h..h....h.h..hh......h..hhhh...h....h.....x..h...h.....hh...h...h.............h..h.....x.........h.h....hh.h\tXR:Z:GA\tXG:Z:GA\n",
            "SRR5836473.1421_1421_length=140\t99\tchr11\t30560802\t0\t140M\t=\t30560952\t289\tTGTTGAGGGGTATTTGGGTTTTTTTTAGTTTTCGGTTATTATAAATAAGGTTGTTATGAATATAGTGTAGTATGTGTTTTTTTTATTGGTTGGAATATTTTTTGGATATATGTTTAGATGAGGTATTGTGGGATTTTTTG\tKKKKKKKKFKFKKKKKKKFKKKKKFKKKKKAK,FAKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFKKKKKKKKKKKKKKKKKKAKKKKKKKKKKKKKKKKKKKKKKKKKKKKKAKKKKAFFK<AFKKKKK\tNM:i:30\tMD:Z:10C2C6C3C0C2C2C0T2C9C4C2C6C6G2A6C3C3C0C8C2C2C10C0C0C3A9C5C0C1C2\tXM:Z:..........h..x......h...hx..h..x...h.........h....x..h......h................h...h...xz........h..h..x..........hhx.............z.....hh.x..\tXR:Z:CT\tXG:Z:CT\n",
            "SRR5836473.1421_1421_length=140\t147\tchr11\t30560952\t0\t139M\t=\t30560802\t-289\tGTTTAATTTTTTGAAGAATTGTTAGATTGATTTTTAGAGTGGTTGTAAAAGTTTGTAATTTTATTAATAATGGAGGAGTGTTTTTTTTTTTTTATATTTTTTTTAGTATTTGTTGTTATTTGAATTTTTGATTTTAGTT\t<<<<<<KF<KKKKKFFFKKAA<KAFKFKKKKKKKKKFFFFFFKKKFFAFAAKKKKFKKKKKKKKKKKKKKKKKKKKFKKKKKKKKKKKKKKKKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKFKKKKKKKFKKKKKKFF\tNM:i:41\tMD:Z:2C0C6C3G3C2C0C3C6C0C12C3C3C3C0C0C1C0C2C14C0C1C3C1C0C1C1G0C0C1C0G0C0C2C2C2C3C1C0C12C5C0\tXM:Z:..hh......x.......x..hx...x......hx................h...h...hhh.hh..h..............hh.h...h.hh.x..hh.z.hx..h..x..x...h.hx............h.....h\tXR:Z:GA\tXG:Z:CT\n",
            "SRR5836473.2061_2061_length=140\t99\tchr11\t73270947\t0\t139M\t=\t73270962\t154\tATAAGAGATAGTTATATTAGGGTTTTGTTAGTAAAATTTTGTTGGTATTTGTAGTAGTGTTTGGGTTTGGTGGTTGTTTATGGGATGGATTTATGGGTGGGGTAGTTTTTGGATGGTTTTTTTTTTTATTTTAGTTTTG\tKKKKKKKKKKKFKKKKKKKKKKFKKKKKKKKKKKKKKFKKKKKKKKKKKKKKKKFKKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFFKKKKKKKFKKKK7AFAFKKKKKF7FKKKKKKFKFAKKKKKKKKF7FFKF\tNM:i:31\tMD:Z:1C6C2C5C5C0C3C2C5C3C3C2A2C1A4A1C1A27C0C10C3C1C9C2C0C2C0C2C1C2C1C2\tXM:Z:.h......x..h.....x.....hx...x..h.....h...x...h.....h........h.............................hh..........x...h.x.........h..hh..hh..h.x..h.x..\tXR:Z:CT\tXG:Z:CT\n",
            "SRR5836473.2061_2061_length=140\t147\tchr11\t73270962\t0\t139M\t=\t73270947\t-154\tATTAGGGTTTTGTTAGTAAAATTTTGTTGGTATTTGTAGTAGTGTTTGGGTTTGGTGGTTGTTTATGGGATGGATTTATGGGTGGGGTAGTTTTTGGATGGTTTTTTTTTTTATTTTAGTTTTGAATTTTGTTTTTGTA\tA7<AAAAA77AKKKFKKKFKKKKAKF<KKKFKKFKKFKAAKKKKFKAKKKKKKKFKKKKKKKKKKKKKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKK\tNM:i:31\tMD:Z:2C5C0C3C2C5C3C3C2A2C1A4A1C1A27C0C10C3C1C9C2C0C2C0C2C1C2C1C4C5C1C4\tXM:Z:..x.....hx...x..h.....h...x...h.....h........h.............................hh..........x...h.x.........h..hh..hh..h.x..h.x....h.....h.x....\tXR:Z:GA\tXG:Z:CT\n",
            "SRR5836473.3331_3331_length=140\t99\tchr11\t43350802\t40\t140M\t=\t43350876\t214\tTTTTAATGTAAGTTGTATAAGGAGTATAATTAGGTTTAGGTGATGTTGAAAATAATATAATTTTATTTATAAATATTATTATTATTATGATAAAGGGAAAGGTGATTATTTTTAAAAGATTATGATGGTTTGTTGGTAAT\tKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFKFFKKKKKKKKKF\tNM:i:29\tMD:Z:0C5C1C6C0T0C7G0C3C3C1C23C2C1C0C6C2C1C0C1C23C0C3C0C0C7C7C0C6C3\tXM:Z:h.....z.h......h.h........h...x...h.x.......................h..h.hh......h..h.hh.h.......................hh...hhh.......h.......hx......h...\tXR:Z:CT\tXG:Z:CT\n",
            "SRR5836473.3331_3331_length=140\t147\tchr11\t43350876\t40\t140M\t=\t43350802\t-214\tATTATTATTATTATGATAAAGGGAAAGGTGATTATTTTTAAAAGATTATGATGGTTTGTTGGTAATGTTATTGTTATACGTAGTATGTTTTGGTGTTATTTGGATTTTGTATTTTTTATAGTTTTTTTATTTTGTTAATT\tKFKKKKK<KKKKKFKAKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKFKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKKK\tNM:i:34\tMD:Z:2C1C0C1C23C0C3C0C0C7C7C0C6C4C2C2C6C2C3C0C7C2C0C3C0C1C1C4C0C11C1C0C4C3C0\tXM:Z:..h.hh.h.......................hh...hhh.......h.......hx......h....h..x..h....Z.x..h...hh.......h..xz...hh.z.h....hh...........h.hh....h...h\tXR:Z:GA\tXG:Z:CT\n"
          ]
        }
      ],
      "source": [
        "! samtools view SRR5836473_1_bismark_bt2_pe.bam | head"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 12,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "SrGGCxZQ2Ao3",
        "outputId": "6a172733-ea13-43c8-d892-c213fd248768"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[bam_sort_core] merging from 4 files and 1 in-memory blocks...\n",
            "[bam_sort_core] merging from 5 files and 1 in-memory blocks...\n",
            "[bam_sort_core] merging from 9 files and 1 in-memory blocks...\n"
          ]
        }
      ],
      "source": [
        "! samtools sort SRR5836473_1_bismark_bt2_pe.bam -o sorted_8cell.bam\n",
        "! samtools sort SRR5836475_1_bismark_bt2_pe.bam -o sorted_icm.bam\n",
        "! samtools sort SRR3824222_1_bismark_bt2_pe.bam -o sorted_epiblast.bam"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 13,
      "metadata": {
        "id": "GErvoOM-3Axj"
      },
      "outputs": [],
      "source": [
        "! samtools index  sorted_8cell.bam\n",
        "! samtools index  sorted_icm.bam\n",
        "! samtools index  sorted_epiblast.bam"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "Lw--6KavBoXp"
      },
      "source": [
        "Сколько ридов закартировалось на регионы 11347700-11367700 и 40185800-40195800 в образце 8-cell?"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 22,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "b-ZhJMlax75S",
        "outputId": "a143d4f6-24e6-4efe-a857-449a985d90b2"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "551\n",
            "194\n"
          ]
        }
      ],
      "source": [
        "! samtools view sorted_8cell.bam chr11:11347700-11367700 | cut -f1 | sort -u | wc -l\n",
        "! samtools view sorted_8cell.bam chr11:40185800-40195800 | cut -f1 | sort -u | wc -l"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Сколько ридов закартировалось на регионы 11347700-11367700 и 40185800-40195800 в образце icm?"
      ],
      "metadata": {
        "id": "m41JwfEQG2HJ"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "! samtools view sorted_icm.bam chr11:11347700-11367700 | cut -f1 | sort -u | wc -l\n",
        "! samtools view sorted_icm.bam chr11:40185800-40195800 | cut -f1 | sort -u | wc -l"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "C-QvEo-cHaOX",
        "outputId": "5e4bc236-d6c5-488f-fa7b-ebd4d5dd0575"
      },
      "execution_count": 15,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "797\n",
            "274\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Сколько ридов закартировалось на регионы 11347700-11367700 и 40185800-40195800 в образце epiblast?"
      ],
      "metadata": {
        "id": "lvGs9y6yG26-"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "! samtools view sorted_epiblast.bam chr11:11347700-11367700 | cut -f1 | sort -u | wc -l\n",
        "! samtools view sorted_epiblast.bam chr11:40185800-40195800 | cut -f1 | sort -u | wc -l"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "X4l3gTdrHfgG",
        "outputId": "09a81438-354b-48c3-b1c2-3303da97eeac"
      },
      "execution_count": 16,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "1344\n",
            "565\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "CliRCSUQTmV-"
      },
      "source": [
        "Дедупликация"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 24,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ShW-Fhzq19kE",
        "outputId": "e86756e7-25e6-4243-8cd4-0a248e1bbac2"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Output filename was given as: s_8_cell\n",
            "Processing paired-end Bismark output file(s) (SAM format):\n",
            "SRR5836473_1_bismark_bt2_pe.bam\n",
            "\n",
            "\n",
            "If there are several alignments to a single position in the genome the first alignment will be chosen. Since the input files are not in any way sorted this is a near-enough random selection of reads.\n",
            "\n",
            "Checking file >>SRR5836473_1_bismark_bt2_pe.bam<< for signs of file truncation...\n",
            "\n",
            "\n",
            "\n",
            "Now testing Bismark result file SRR5836473_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\t...passed!\n",
            "Output file is: s_8_cell.deduplicated.bam\n",
            "\n",
            "skipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
            "skipping header line:\t@SQ\tSN:chr11\tLN:121973369\n",
            "skipping header line:\t@PG\tID:Bismark\tVN:v0.24.2\tCL:\"bismark --genome_folder /mnt/cache-1/for_minor/mm39_11/ -q --non_directional -parallel 10 -1 SRR3824222_1.fq,SRR5836473_1.fq,SRR5836473_1.fq -2 SRR3824222_2.fq,SRR5836473_2.fq,SRR5836473_2.fq\"\n",
            "skipping header line:\t@PG\tID:samtools\tPN:samtools\tPP:Bismark\tVN:1.19.2\tCL:/usr/local/bin/samtools view -h --threads 1 SRR5836473_1_bismark_bt2_pe.bam\n",
            "\n",
            "Total number of alignments analysed in SRR5836473_1_bismark_bt2_pe.bam:\t2858170\n",
            "Total number duplicated alignments removed:\t522512 (18.28%)\n",
            "Duplicated alignments were found at:\t230677 different position(s)\n",
            "\n",
            "Total count of deduplicated leftover sequences: 2335658 (81.72% of total)\n",
            "\n"
          ]
        }
      ],
      "source": [
        "! deduplicate_bismark  --bam  --paired  -o s_8_cell  SRR5836473_1_bismark_bt2_pe.bam"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! deduplicate_bismark  --bam  --paired  -o s_icm  SRR5836475_1_bismark_bt2_pe.bam"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "P3LKb4v3I3i9",
        "outputId": "935de387-5eaa-40f7-ee7e-fcc42657244c"
      },
      "execution_count": 25,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Output filename was given as: s_icm\n",
            "Processing paired-end Bismark output file(s) (SAM format):\n",
            "SRR5836475_1_bismark_bt2_pe.bam\n",
            "\n",
            "\n",
            "If there are several alignments to a single position in the genome the first alignment will be chosen. Since the input files are not in any way sorted this is a near-enough random selection of reads.\n",
            "\n",
            "Checking file >>SRR5836475_1_bismark_bt2_pe.bam<< for signs of file truncation...\n",
            "\n",
            "\n",
            "\n",
            "Now testing Bismark result file SRR5836475_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\t...passed!\n",
            "Output file is: s_icm.deduplicated.bam\n",
            "\n",
            "skipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
            "skipping header line:\t@SQ\tSN:chr11\tLN:121973369\n",
            "skipping header line:\t@PG\tID:Bismark\tVN:v0.24.2\tCL:\"bismark --genome_folder /mnt/cache-1/for_minor/mm39_11/ -q --non_directional -parallel 10 -1 SRR5836475_1.fq -2 SRR5836475_2.fq\"\n",
            "skipping header line:\t@PG\tID:samtools\tPN:samtools\tPP:Bismark\tVN:1.19.2\tCL:/usr/local/bin/samtools view -h --threads 1 SRR5836475_1_bismark_bt2_pe.bam\n",
            "\n",
            "Total number of alignments analysed in SRR5836475_1_bismark_bt2_pe.bam:\t4170098\n",
            "Total number duplicated alignments removed:\t378125 (9.07%)\n",
            "Duplicated alignments were found at:\t229940 different position(s)\n",
            "\n",
            "Total count of deduplicated leftover sequences: 3791973 (90.93% of total)\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! deduplicate_bismark  --bam  --paired  -o s_epiblast  SRR3824222_1_bismark_bt2_pe.bam"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "HAKgZsKVI4Dr",
        "outputId": "9758493c-50a7-4b7e-8d6b-87dc4710965f"
      },
      "execution_count": 26,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Output filename was given as: s_epiblast\n",
            "Processing paired-end Bismark output file(s) (SAM format):\n",
            "SRR3824222_1_bismark_bt2_pe.bam\n",
            "\n",
            "\n",
            "If there are several alignments to a single position in the genome the first alignment will be chosen. Since the input files are not in any way sorted this is a near-enough random selection of reads.\n",
            "\n",
            "Checking file >>SRR3824222_1_bismark_bt2_pe.bam<< for signs of file truncation...\n",
            "\n",
            "\n",
            "\n",
            "Now testing Bismark result file SRR3824222_1_bismark_bt2_pe.bam for positional sorting (which would be bad...)\t...passed!\n",
            "Output file is: s_epiblast.deduplicated.bam\n",
            "\n",
            "skipping header line:\t@HD\tVN:1.0\tSO:unsorted\n",
            "skipping header line:\t@SQ\tSN:chr11\tLN:121973369\n",
            "skipping header line:\t@PG\tID:Bismark\tVN:v0.24.2\tCL:\"bismark --genome_folder /mnt/cache-1/for_minor/mm39_11/ -q --non_directional -parallel 10 -1 SRR3824222_1.fq,SRR5836473_1.fq,SRR5836473_1.fq -2 SRR3824222_2.fq,SRR5836473_2.fq,SRR5836473_2.fq\"\n",
            "skipping header line:\t@PG\tID:samtools\tPN:samtools\tPP:Bismark\tVN:1.19.2\tCL:/usr/local/bin/samtools view -h --threads 1 SRR3824222_1_bismark_bt2_pe.bam\n",
            "\n",
            "Total number of alignments analysed in SRR3824222_1_bismark_bt2_pe.bam:\t7051731\n",
            "Total number duplicated alignments removed:\t205287 (2.91%)\n",
            "Duplicated alignments were found at:\t164998 different position(s)\n",
            "\n",
            "Total count of deduplicated leftover sequences: 6846444 (97.09% of total)\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 27,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "OkrrqUQPXCCI",
        "outputId": "dfce98c0-8d07-4510-a374-da5300810ccc"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "condacolab_install.log\t\t     sorted_epiblast.bam\n",
            "FastQC\t\t\t\t     sorted_epiblast.bam.bai\n",
            "fastqc_v0.11.9.zip\t\t     sorted_icm.bam\n",
            "mouse_chr\t\t\t     sorted_icm.bam.bai\n",
            "mouse_chr.fa\t\t\t     SRR3824222_1_bismark_bt2_pe.bam\n",
            "mouse_chr.gz\t\t\t     SRR3824222_1_bismark_bt2_PE_report.txt\n",
            "s_8_cell.deduplicated.bam\t     SRR5836473_1_bismark_bt2_pe.bam\n",
            "s_8_cell.deduplication_report.txt    SRR5836473_1_bismark_bt2_PE_report.txt\n",
            "sample_data\t\t\t     SRR5836473_1_fastqc.html\n",
            "s_epiblast.deduplicated.bam\t     SRR5836473_1_fastqc.zip\n",
            "s_epiblast.deduplication_report.txt  SRR5836473_2_fastqc.html\n",
            "s_icm.deduplicated.bam\t\t     SRR5836473_2_fastqc.zip\n",
            "s_icm.deduplication_report.txt\t     SRR5836475_1_bismark_bt2_pe.bam\n",
            "sorted_8cell.bam\t\t     SRR5836475_1_bismark_bt2_PE_report.txt\n",
            "sorted_8cell.bam.bai\n"
          ]
        }
      ],
      "source": [
        "! ls"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "11-ю хромосому мыши можно скачать с UCSC (сборка mm39). Скачаем при помощи table browser, распакуем и загрузим файл сюда"
      ],
      "metadata": {
        "id": "969PlQMGwBO3"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "! mkdir mouse_chr\n",
        "! mv chr11.fa ./mouse_chr"
      ],
      "metadata": {
        "id": "b6y3Gm1APhu2"
      },
      "execution_count": 32,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "sf_yqXE71QjG"
      },
      "source": [
        "Время работы: ±30 мин"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! head s_8_cell.deduplicated.bedGraph"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "buKTYu3PBuA3",
        "outputId": "1b0e379f-73b7-491a-878e-b6708fc9aa07"
      },
      "execution_count": 21,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "head: cannot open 's_8_cell.deduplicated.bedGraph' for reading: No such file or directory\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Коллинг метилирования"
      ],
      "metadata": {
        "id": "LKBD3cwo1wLx"
      }
    },
    {
      "cell_type": "code",
      "execution_count": 53,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "4c5BEvKpDw5K",
        "outputId": "0362b194-1b01-460d-a5ea-b091465599dd"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            " *** Bismark methylation extractor version v0.24.2 ***\n",
            "\n",
            "Trying to determine the type of mapping from the SAM header line of file s_8_cell.deduplicated.bam\n",
            "Treating file(s) as paired-end data (as extracted from @PG line)\n",
            "\n",
            "Setting option '--no_overlap' since this is (normally) the right thing to do for paired-end data\n",
            "\n",
            "\n",
            "Setting core usage to single-threaded (default). Consider using --multicore <int> to speed up the extraction process.\n",
            "\n",
            "Summarising Bismark methylation extractor parameters:\n",
            "===============================================================\n",
            "Bismark paired-end SAM format specified (default)\n",
            "Number of cores to be used: 1\n",
            "Output will be written to the current directory ('/content')\n",
            "\n",
            "\n",
            "Summarising bedGraph parameters:\n",
            "===============================================================\n",
            "Generating additional output in bedGraph and coverage format\n",
            "bedGraph format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage>\n",
            "coverage format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage> <count methylated> <count non-methylated>\n",
            "\n",
            "Using a cutoff of 1 read(s) to report cytosine positions\n",
            "Reporting and sorting cytosine methylation information in CpG context only (default)\n",
            "The bedGraph UNIX sort command will use the following memory setting:\t'2G'. Temporary directory used for sorting is the output directory\n",
            "\n",
            "\n",
            "Summarising genome-wide cytosine methylation report parameters:\n",
            "===============================================================\n",
            "Generating comprehensive genome-wide cytosine report\n",
            "(output format: <Chromosome> <Position> <Strand> <count methylated> <count non-methylated>  <C-context>  <trinucleotide context> )\n",
            "Reporting cytosine methylation in CpG context only (default)\n",
            "Using 1-based genomic coordinates (default)\n",
            "Genome folder was specified as ./mouse_chr/\n",
            "\n",
            "Checking file >>s_8_cell.deduplicated.bam<< for signs of file truncation...\n",
            "\n",
            "\n",
            "Now testing Bismark result file >s_8_cell.deduplicated.bam< for positional sorting (which would be bad...)\t...passed!\n",
            "Writing result file containing methylation information for C in CpG context from the original top strand to CpG_OT_s_8_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the complementary to original top strand to CpG_CTOT_s_8_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the complementary to original bottom strand to CpG_CTOB_s_8_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the original bottom strand to CpG_OB_s_8_cell.deduplicated.txt.gz\n",
            "\n",
            "Writing result file containing methylation information for C in CHG context from the original top strand to CHG_OT_s_8_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the complementary to original top strand to CHG_CTOT_s_8_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the complementary to original bottom strand to CHG_CTOB_s_8_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the original bottom strand to CHG_OB_s_8_cell.deduplicated.txt.gz\n",
            "\n",
            "Writing result file containing methylation information for C in CHH context from the original top strand to CHH_OT_s_8_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the complementary to original top strand to CHH_CTOT_s_8_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the complementary to original bottom strand to CHH_CTOB_s_8_cell.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the original bottom strand to CHH_OB_s_8_cell.deduplicated.txt.gz\n",
            "\n",
            "Now reading in Bismark result file s_8_cell.deduplicated.bam\n",
            "skipping SAM header line:\t@HD\tVN:1.0\tSO:unsorted\n",
            "skipping SAM header line:\t@SQ\tSN:chr11\tLN:121973369\n",
            "skipping SAM header line:\t@PG\tID:Bismark\tVN:v0.24.2\tCL:\"bismark --genome_folder /mnt/cache-1/for_minor/mm39_11/ -q --non_directional -parallel 10 -1 SRR3824222_1.fq,SRR5836473_1.fq,SRR5836473_1.fq -2 SRR3824222_2.fq,SRR5836473_2.fq,SRR5836473_2.fq\"\n",
            "skipping SAM header line:\t@PG\tID:samtools\tPN:samtools\tPP:Bismark\tVN:1.19.2\tCL:/usr/local/bin/samtools view -h --threads 1 SRR5836473_1_bismark_bt2_pe.bam\n",
            "skipping SAM header line:\t@PG\tID:samtools.1\tPN:samtools\tPP:samtools\tVN:1.19.2\tCL:/usr/local/bin/samtools view -bS --threads 1 -\n",
            "skipping SAM header line:\t@PG\tID:samtools.2\tPN:samtools\tPP:samtools.1\tVN:1.19.2\tCL:/usr/local/bin/samtools view -h s_8_cell.deduplicated.bam\n",
            "Processed lines: 500000\n",
            "Processed lines: 1000000\n",
            "Processed lines: 1500000\n",
            "Processed lines: 2000000\n",
            "\n",
            "Processed 2335658 lines in total\n",
            "Total number of methylation call strings processed: 4671316\n",
            "\n",
            "Final Cytosine Methylation Report\n",
            "=================================\n",
            "Total number of C's analysed:\t104748620\n",
            "\n",
            "Total methylated C's in CpG context:\t2103916\n",
            "Total methylated C's in CHG context:\t270195\n",
            "Total methylated C's in CHH context:\t1236285\n",
            "\n",
            "Total C to T conversions in CpG context:\t2729982\n",
            "Total C to T conversions in CHG context:\t22951565\n",
            "Total C to T conversions in CHH context:\t75456677\n",
            "\n",
            "C methylated in CpG context:\t43.5%\n",
            "C methylated in CHG context:\t1.2%\n",
            "C methylated in CHH context:\t1.6%\n",
            "\n",
            "\n",
            "Determining maximum read lengths for M-Bias plots\n",
            "Maximum read length of Read 1: 140\n",
            "Maximum read length of Read 2: 140\n",
            "\n",
            "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
            "Determining maximum read lengths for M-Bias plots\n",
            "Maximum read length of Read 1: 140\n",
            "Maximum read length of Read 2: 140\n",
            "\n",
            "Perl module GD::Graph::lines is not installed, skipping drawing M-bias plots (only writing out M-bias plot table)\n",
            "Deleting unused files ...\n",
            "\n",
            "CpG_OT_s_8_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CpG_CTOT_s_8_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CpG_CTOB_s_8_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CpG_OB_s_8_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CHG_OT_s_8_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CHG_CTOT_s_8_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CHG_CTOB_s_8_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CHG_OB_s_8_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CHH_OT_s_8_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CHH_CTOT_s_8_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CHH_CTOB_s_8_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "CHH_OB_s_8_cell.deduplicated.txt.gz contains data ->\tkept\n",
            "\n",
            "\n",
            "Using these input files: CpG_OT_s_8_cell.deduplicated.txt.gz CpG_CTOT_s_8_cell.deduplicated.txt.gz CpG_CTOB_s_8_cell.deduplicated.txt.gz CpG_OB_s_8_cell.deduplicated.txt.gz CHG_OT_s_8_cell.deduplicated.txt.gz CHG_CTOT_s_8_cell.deduplicated.txt.gz CHG_CTOB_s_8_cell.deduplicated.txt.gz CHG_OB_s_8_cell.deduplicated.txt.gz CHH_OT_s_8_cell.deduplicated.txt.gz CHH_CTOT_s_8_cell.deduplicated.txt.gz CHH_CTOB_s_8_cell.deduplicated.txt.gz CHH_OB_s_8_cell.deduplicated.txt.gz\n",
            "\n",
            "Summary of parameters for bismark2bedGraph conversion:\n",
            "======================================================\n",
            "bedGraph output:\t\ts_8_cell.deduplicated.bedGraph.gz\n",
            "output directory:\t\t><\n",
            "remove whitespaces:\t\tno\n",
            "CX context:\t\t\tno (CpG context only, default)\n",
            "No-header selected:\t\tno\n",
            "Sorting method:\t\t\tUnix sort-based (smaller memory footprint, but slower)\n",
            "Sort buffer size:\t\t2G\n",
            "Coverage threshold:\t\t1\n",
            "=============================================================================\n",
            "Methylation information will now be written into a bedGraph and coverage file\n",
            "=============================================================================\n",
            "\n",
            "Using the following files as Input:\n",
            "/content/CpG_OT_s_8_cell.deduplicated.txt.gz\t/content/CpG_CTOT_s_8_cell.deduplicated.txt.gz\t/content/CpG_CTOB_s_8_cell.deduplicated.txt.gz\t/content/CpG_OB_s_8_cell.deduplicated.txt.gz\n",
            "\n",
            "Writing bedGraph to file: s_8_cell.deduplicated.bedGraph.gz\n",
            "Also writing out a coverage file including counts methylated and unmethylated residues to file: s_8_cell.deduplicated.bismark.cov.gz\n",
            "\n",
            "Now writing methylation information for file >>CpG_OT_s_8_cell.deduplicated.txt.gz<< to individual files for each chromosome\n",
            "Finished writing out individual chromosome files for CpG_OT_s_8_cell.deduplicated.txt.gz\n",
            "Now writing methylation information for file >>CpG_CTOT_s_8_cell.deduplicated.txt.gz<< to individual files for each chromosome\n",
            "Finished writing out individual chromosome files for CpG_CTOT_s_8_cell.deduplicated.txt.gz\n",
            "Now writing methylation information for file >>CpG_CTOB_s_8_cell.deduplicated.txt.gz<< to individual files for each chromosome\n",
            "Finished writing out individual chromosome files for CpG_CTOB_s_8_cell.deduplicated.txt.gz\n",
            "Now writing methylation information for file >>CpG_OB_s_8_cell.deduplicated.txt.gz<< to individual files for each chromosome\n",
            "Finished writing out individual chromosome files for CpG_OB_s_8_cell.deduplicated.txt.gz\n",
            "\n",
            "Collecting temporary chromosome file information... Processing the following input file(s):\n",
            "CpG_OT_s_8_cell.deduplicated.txt.gz.chrchr11.methXtractor.temp\n",
            "\n",
            "Sorting input file CpG_OT_s_8_cell.deduplicated.txt.gz.chrchr11.methXtractor.temp by positions (using -S of 2G)\n",
            "Successfully deleted the temporary input file CpG_OT_s_8_cell.deduplicated.txt.gz.chrchr11.methXtractor.temp\n",
            "\n",
            "Finished BedGraph conversion ...\n",
            "\n",
            "Summary of parameters for genome-wide cytosine report:\n",
            "==============================================================================\n",
            "Coverage infile:\t\t\t\ts_8_cell.deduplicated.bismark.cov.gz\n",
            "Output directory:\t\t\t\t><\n",
            "Parent directory:\t\t\t\t>/content/<\n",
            "Genome directory:\t\t\t\t>/content/mouse_chr/<\n",
            "Coverage threshold\t\t\t\t0 (all positions will be reported, default)\n",
            "CX context:\t\t\t\t\tno (CpG context only, default)\n",
            "Genome coordinates used:\t\t\t1-based (default)\n",
            "GZIP compression:\t\t\t\tyes\n",
            "Split by chromosome:\t\t\t\tno\n",
            "\n",
            "\n",
            "Now reading in and storing sequence information of the genome specified in: /content/mouse_chr/\n",
            "\n",
            "chr chr11 (8224119 bp)\n",
            "\n",
            "Stored sequence information of 1 chromosomes/scaffolds in total\n",
            "\n",
            "==============================================================================\n",
            "Methylation information will now be written into a genome-wide cytosine report\n",
            "==============================================================================\n",
            "\n",
            "Adding context-specific methylation summaries\n",
            "\n",
            ">>> Writing genome-wide cytosine report to: s_8_cell.deduplicated.CpG_report.txt.gz <<<\n",
            "\n",
            ">>> Writing all cytosine context summary file to: s_8_cell.deduplicated.cytosine_context_summary.txt <<<\n",
            "\n",
            "Storing all covered cytosine positions for chromosome: chr11\n",
            "Writing cytosine report for last chromosome chr11 (stored 1737533 different covered positions)\n",
            "Finished writing out cytosine report for covered chromosomes (processed 1 chromosomes/scaffolds in total)\n",
            "\n",
            "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
            "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
            "\n",
            "\n",
            "\n",
            "Finished generating genome-wide cytosine report\n",
            "\n"
          ]
        }
      ],
      "source": [
        "! bismark_methylation_extractor --gzip --bedGraph --cytosine_report --genome_folder ./mouse_chr s_8_cell.deduplicated.bam"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! bismark_methylation_extractor --gzip --bedGraph --cytosine_report --genome_folder ./mouse_chr s_icm.deduplicated.bam"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "P_DBJgmMT-g1",
        "outputId": "4a108846-254a-4b63-8b80-de6159db3fcd"
      },
      "execution_count": 54,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            " *** Bismark methylation extractor version v0.24.2 ***\n",
            "\n",
            "Trying to determine the type of mapping from the SAM header line of file s_icm.deduplicated.bam\n",
            "Treating file(s) as paired-end data (as extracted from @PG line)\n",
            "\n",
            "Setting option '--no_overlap' since this is (normally) the right thing to do for paired-end data\n",
            "\n",
            "\n",
            "Setting core usage to single-threaded (default). Consider using --multicore <int> to speed up the extraction process.\n",
            "\n",
            "Summarising Bismark methylation extractor parameters:\n",
            "===============================================================\n",
            "Bismark paired-end SAM format specified (default)\n",
            "Number of cores to be used: 1\n",
            "Output will be written to the current directory ('/content')\n",
            "\n",
            "\n",
            "Summarising bedGraph parameters:\n",
            "===============================================================\n",
            "Generating additional output in bedGraph and coverage format\n",
            "bedGraph format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage>\n",
            "coverage format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage> <count methylated> <count non-methylated>\n",
            "\n",
            "Using a cutoff of 1 read(s) to report cytosine positions\n",
            "Reporting and sorting cytosine methylation information in CpG context only (default)\n",
            "The bedGraph UNIX sort command will use the following memory setting:\t'2G'. Temporary directory used for sorting is the output directory\n",
            "\n",
            "\n",
            "Summarising genome-wide cytosine methylation report parameters:\n",
            "===============================================================\n",
            "Generating comprehensive genome-wide cytosine report\n",
            "(output format: <Chromosome> <Position> <Strand> <count methylated> <count non-methylated>  <C-context>  <trinucleotide context> )\n",
            "Reporting cytosine methylation in CpG context only (default)\n",
            "Using 1-based genomic coordinates (default)\n",
            "Genome folder was specified as ./mouse_chr/\n",
            "\n",
            "Checking file >>s_icm.deduplicated.bam<< for signs of file truncation...\n",
            "\n",
            "\n",
            "Now testing Bismark result file >s_icm.deduplicated.bam< for positional sorting (which would be bad...)\t...passed!\n",
            "Writing result file containing methylation information for C in CpG context from the original top strand to CpG_OT_s_icm.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the complementary to original top strand to CpG_CTOT_s_icm.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the complementary to original bottom strand to CpG_CTOB_s_icm.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CpG context from the original bottom strand to CpG_OB_s_icm.deduplicated.txt.gz\n",
            "\n",
            "Writing result file containing methylation information for C in CHG context from the original top strand to CHG_OT_s_icm.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the complementary to original top strand to CHG_CTOT_s_icm.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the complementary to original bottom strand to CHG_CTOB_s_icm.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHG context from the original bottom strand to CHG_OB_s_icm.deduplicated.txt.gz\n",
            "\n",
            "Writing result file containing methylation information for C in CHH context from the original top strand to CHH_OT_s_icm.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the complementary to original top strand to CHH_CTOT_s_icm.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the complementary to original bottom strand to CHH_CTOB_s_icm.deduplicated.txt.gz\n",
            "Writing result file containing methylation information for C in CHH context from the original bottom strand to CHH_OB_s_icm.deduplicated.txt.gz\n",
            "\n",
            "Now reading in Bismark result file s_icm.deduplicated.bam\n",
            "skipping SAM header line:\t@HD\tVN:1.0\tSO:unsorted\n",
            "skipping SAM header line:\t@SQ\tSN:chr11\tLN:121973369\n",
            "skipping SAM header line:\t@PG\tID:Bismark\tVN:v0.24.2\tCL:\"bismark --genome_folder /mnt/cache-1/for_minor/mm39_11/ -q --non_directional -parallel 10 -1 SRR5836475_1.fq -2 SRR5836475_2.fq\"\n",
            "skipping SAM header line:\t@PG\tID:samtools\tPN:samtools\tPP:Bismark\tVN:1.19.2\tCL:/usr/local/bin/samtools view -h --threads 1 SRR5836475_1_bismark_bt2_pe.bam\n",
            "skipping SAM header line:\t@PG\tID:samtools.1\tPN:samtools\tPP:samtools\tVN:1.19.2\tCL:/usr/local/bin/samtools view -bS --threads 1 -\n",
            "skipping SAM header line:\t@PG\tID:samtools.2\tPN:samtools\tPP:samtools.1\tVN:1.19.2\tCL:/usr/local/bin/samtools view -h s_icm.deduplicated.bam\n",
            "Processed lines: 500000\n",
            "Processed lines: 1000000\n",
            "Processed lines: 1500000\n",
            "Processed lines: 2000000\n",
            "Processed lines: 2500000\n",
            "^C\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! bismark_methylation_extractor --gzip --bedGraph --cytosine_report --genome_folder ./mouse_chr s_epiblast.deduplicated.bam"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "2_dgeSTPT_B8",
        "outputId": "d6cb9639-022a-4e2d-8965-1ba438159a5a"
      },
      "execution_count": 55,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            " *** Bismark methylation extractor version v0.24.2 ***\n",
            "\n",
            "Trying to determine the type of mapping from the SAM header line of file s_epiblast.deduplicated.bam\n",
            "Treating file(s) as paired-end data (as extracted from @PG line)\n",
            "\n",
            "Setting option '--no_overlap' since this is (normally) the right thing to do for paired-end data\n",
            "\n",
            "\n",
            "Setting core usage to single-threaded (default). Consider using --multicore <int> to speed up the extraction process.\n",
            "\n",
            "Summarising Bismark methylation extractor parameters:\n",
            "===============================================================\n",
            "Bismark paired-end SAM format specified (default)\n",
            "Number of cores to be used: 1\n",
            "Output will be written to the current directory ('/content')\n",
            "\n",
            "\n",
            "Summarising bedGraph parameters:\n",
            "===============================================================\n",
            "Generating additional output in bedGraph and coverage format\n",
            "bedGraph format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage>\n",
            "coverage format:\t<Chromosome> <Start Position> <End Position> <Methylation Percentage> <count methylated> <count non-methylated>\n",
            "\n",
            "Using a cutoff of 1 read(s) to report cytosine positions\n",
            "Reporting and sorting cytosine methylation information in CpG context only (default)\n",
            "The bedGraph UNIX sort command will use the following memory setting:\t'2G'. Temporary directory used for sorting is the output directory\n",
            "\n",
            "\n",
            "Summarising genome-wide cytosine methylation report parameters:\n",
            "===============================================================\n",
            "Generating comprehensive genome-wide cytosine report\n",
            "(output format: <Chromosome> <Position> <Strand> <count methylated> <count non-methylated>  <C-context>  <trinucleotide context> )\n",
            "Reporting cytosine methylation in CpG context only (default)\n",
            "Using 1-based genomic coordinates (default)\n",
            "Genome folder was specified as ./mouse_chr/\n",
            "\n",
            "Checking file >>s_epiblast.deduplicated.bam<< for signs of file truncation...\n",
            "\n",
            "\n",
            "Now testing Bismark result file >s_epiblast.deduplicated.bam< for positional sorting (which would be bad...)\t^C\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! ls -lh"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "iNwzMUeAkQ4f",
        "outputId": "28c378c5-beb6-4c1b-ed90-3f373fee6f82"
      },
      "execution_count": 36,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "total 13G\n",
            "-rw-r--r-- 1 root root  7.7K Jun 25 13:55 CHG_CTOB_s_8_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root   13K Jun 25 15:51 CHG_CTOB_s_epiblast.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  9.4K Jun 25 14:36 CHG_CTOB_s_icm.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  8.2K Jun 25 13:55 CHG_CTOT_s_8_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root   14K Jun 25 15:51 CHG_CTOT_s_epiblast.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root   10K Jun 25 14:36 CHG_CTOT_s_icm.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root   51M Jun 25 13:55 CHG_OB_s_8_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  157M Jun 25 15:51 CHG_OB_s_epiblast.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root   84M Jun 25 14:36 CHG_OB_s_icm.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root   51M Jun 25 13:55 CHG_OT_s_8_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  157M Jun 25 15:51 CHG_OT_s_epiblast.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root   85M Jun 25 14:36 CHG_OT_s_icm.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  1.1M Jun 25 13:55 CHH_CTOB_s_8_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  1.9M Jun 25 15:51 CHH_CTOB_s_epiblast.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  1.6M Jun 25 14:36 CHH_CTOB_s_icm.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root 1014K Jun 25 13:55 CHH_CTOT_s_8_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  1.7M Jun 25 15:51 CHH_CTOT_s_epiblast.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  1.4M Jun 25 14:36 CHH_CTOT_s_icm.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  133M Jun 25 13:55 CHH_OB_s_8_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  419M Jun 25 15:51 CHH_OB_s_epiblast.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  223M Jun 25 14:36 CHH_OB_s_icm.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  133M Jun 25 13:55 CHH_OT_s_8_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  419M Jun 25 15:51 CHH_OT_s_epiblast.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  223M Jun 25 14:36 CHH_OT_s_icm.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  111M Jun 25 13:42 chr11.fa\n",
            "-rw-r--r-- 1 root root   23K Jun 25 10:47 condacolab_install.log\n",
            "-rw-r--r-- 1 root root  4.2K Jun 25 13:55 CpG_CTOB_s_8_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  7.5K Jun 25 15:51 CpG_CTOB_s_epiblast.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  4.7K Jun 25 14:36 CpG_CTOB_s_icm.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  3.8K Jun 25 13:55 CpG_CTOT_s_8_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  8.0K Jun 25 15:51 CpG_CTOT_s_epiblast.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root  4.6K Jun 25 14:36 CpG_CTOT_s_icm.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root   17M Jun 25 13:55 CpG_OB_s_8_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root   49M Jun 25 15:51 CpG_OB_s_epiblast.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root   28M Jun 25 14:36 CpG_OB_s_icm.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root   17M Jun 25 13:55 CpG_OT_s_8_cell.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root   49M Jun 25 15:51 CpG_OT_s_epiblast.deduplicated.txt.gz\n",
            "-rw-r--r-- 1 root root   28M Jun 25 14:36 CpG_OT_s_icm.deduplicated.txt.gz\n",
            "drwxr-xr-x 8 root root  4.0K Jun 25 09:26 FastQC\n",
            "-rw-r--r-- 1 root root  9.8M Jan 15  2020 fastqc_v0.11.9.zip\n",
            "drwxr-xr-x 2 root root  4.0K Jun 25 13:31 mouse_chr\n",
            "-rw-r--r-- 1 root root  620M Jun 25 12:08 s_8_cell.deduplicated.bam\n",
            "-rw-r--r-- 1 root root  9.8M Jun 25 13:56 s_8_cell.deduplicated.bedGraph.gz\n",
            "-rw-r--r-- 1 root root  9.4M Jun 25 13:56 s_8_cell.deduplicated.bismark.cov.gz\n",
            "-rw-r--r-- 1 root root  486K Jun 25 13:57 s_8_cell.deduplicated.CpG_report.txt.gz\n",
            "-rw-r--r-- 1 root root  1.5K Jun 25 13:57 s_8_cell.deduplicated.cytosine_context_summary.txt\n",
            "-rw-r--r-- 1 root root   22K Jun 25 13:55 s_8_cell.deduplicated.M-bias.txt\n",
            "-rw-r--r-- 1 root root   841 Jun 25 13:55 s_8_cell.deduplicated_splitting_report.txt\n",
            "-rw-r--r-- 1 root root   283 Jun 25 12:08 s_8_cell.deduplication_report.txt\n",
            "drwxr-xr-x 1 root root  4.0K Jun 21 13:28 sample_data\n",
            "-rw-r--r-- 1 root root  1.9G Jun 25 12:18 s_epiblast.deduplicated.bam\n",
            "-rw-r--r-- 1 root root   14M Jun 25 15:53 s_epiblast.deduplicated.bedGraph.gz\n",
            "-rw-r--r-- 1 root root   14M Jun 25 15:53 s_epiblast.deduplicated.bismark.cov.gz\n",
            "-rw-r--r-- 1 root root  528K Jun 25 15:54 s_epiblast.deduplicated.CpG_report.txt.gz\n",
            "-rw-r--r-- 1 root root  1.6K Jun 25 15:54 s_epiblast.deduplicated.cytosine_context_summary.txt\n",
            "-rw-r--r-- 1 root root   24K Jun 25 15:51 s_epiblast.deduplicated.M-bias.txt\n",
            "-rw-r--r-- 1 root root   847 Jun 25 15:51 s_epiblast.deduplicated_splitting_report.txt\n",
            "-rw-r--r-- 1 root root   282 Jun 25 12:18 s_epiblast.deduplication_report.txt\n",
            "-rw-r--r-- 1 root root  1.1G Jun 25 12:11 s_icm.deduplicated.bam\n",
            "-rw-r--r-- 1 root root   12M Jun 25 14:38 s_icm.deduplicated.bedGraph.gz\n",
            "-rw-r--r-- 1 root root   12M Jun 25 14:38 s_icm.deduplicated.bismark.cov.gz\n",
            "-rw-r--r-- 1 root root  502K Jun 25 14:38 s_icm.deduplicated.CpG_report.txt.gz\n",
            "-rw-r--r-- 1 root root  1.5K Jun 25 14:38 s_icm.deduplicated.cytosine_context_summary.txt\n",
            "-rw-r--r-- 1 root root   23K Jun 25 14:36 s_icm.deduplicated.M-bias.txt\n",
            "-rw-r--r-- 1 root root   839 Jun 25 14:36 s_icm.deduplicated_splitting_report.txt\n",
            "-rw-r--r-- 1 root root   282 Jun 25 12:11 s_icm.deduplication_report.txt\n",
            "-rw-r--r-- 1 root root  528M Jun 25 11:13 sorted_8cell.bam\n",
            "-rw-r--r-- 1 root root  203K Jun 25 11:21 sorted_8cell.bam.bai\n",
            "-rw-r--r-- 1 root root  1.3G Jun 25 11:21 sorted_epiblast.bam\n",
            "-rw-r--r-- 1 root root  349K Jun 25 11:22 sorted_epiblast.bam.bai\n",
            "-rw-r--r-- 1 root root  777M Jun 25 11:16 sorted_icm.bam\n",
            "-rw-r--r-- 1 root root  341K Jun 25 11:21 sorted_icm.bam.bai\n",
            "-rw-r--r-- 1 root root  1.9G Jan 30 08:06 SRR3824222_1_bismark_bt2_pe.bam\n",
            "-rw-r--r-- 1 root root  1.8K Jan 31 18:19 SRR3824222_1_bismark_bt2_PE_report.txt\n",
            "-rw-r--r-- 1 root root  769M Jan 30 07:20 SRR5836473_1_bismark_bt2_pe.bam\n",
            "-rw-r--r-- 1 root root  1.8K Jan 31 18:19 SRR5836473_1_bismark_bt2_PE_report.txt\n",
            "-rw-r--r-- 1 root root  570K Jun 25 10:01 SRR5836473_1_fastqc.html\n",
            "-rw-r--r-- 1 root root  296K Jun 25 10:01 SRR5836473_1_fastqc.zip\n",
            "-rw-r--r-- 1 root root  582K Jun 25 10:43 SRR5836473_2_fastqc.html\n",
            "-rw-r--r-- 1 root root  310K Jun 25 10:43 SRR5836473_2_fastqc.zip\n",
            "-rw-r--r-- 1 root root  1.2G Jan 30 07:01 SRR5836475_1_bismark_bt2_pe.bam\n",
            "-rw-r--r-- 1 root root  1.8K Jan 31 18:19 SRR5836475_1_bismark_bt2_PE_report.txt\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [],
      "metadata": {
        "id": "1yuRuud-mMK-"
      }
    },
    {
      "cell_type": "code",
      "execution_count": 50,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "6RQ7FwM-UsSi",
        "outputId": "8cec4bf7-481b-4cc2-8662-5519ea5c7bb8"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Found 3 alignment reports in current directory. Now trying to figure out whether there are corresponding optional reports\n",
            "\n",
            "Writing Bismark HTML report to >> SRR3824222_1_bismark_bt2_PE_report.html <<\n",
            "\n",
            "==============================================================================================================\n",
            "Using the following alignment report:\t\t> SRR3824222_1_bismark_bt2_PE_report.txt <\n",
            "Processing alignment report SRR3824222_1_bismark_bt2_PE_report.txt ...\n",
            "Complete\n",
            "\n",
            "No deduplication report file specified, skipping this step\n",
            "No splitting report file specified, skipping this step\n",
            "Using the following M-bias report:\t\t> s_8_cell.deduplicated.M-bias.txt <\n",
            "Processing M-bias report s_8_cell.deduplicated.M-bias.txt ...\n",
            "Complete\n",
            "\n",
            "No nucleotide coverage report file specified, skipping this step\n",
            "==============================================================================================================\n",
            "\n",
            "\n",
            "\n",
            "Writing Bismark HTML report to >> SRR5836473_1_bismark_bt2_PE_report.html <<\n",
            "\n",
            "==============================================================================================================\n",
            "Using the following alignment report:\t\t> SRR5836473_1_bismark_bt2_PE_report.txt <\n",
            "Processing alignment report SRR5836473_1_bismark_bt2_PE_report.txt ...\n",
            "Complete\n",
            "\n",
            "No deduplication report file specified, skipping this step\n",
            "No splitting report file specified, skipping this step\n",
            "No M-bias report file specified, skipping this step\n",
            "No nucleotide coverage report file specified, skipping this step\n",
            "==============================================================================================================\n",
            "\n",
            "\n",
            "\n",
            "Writing Bismark HTML report to >> SRR5836475_1_bismark_bt2_PE_report.html <<\n",
            "\n",
            "==============================================================================================================\n",
            "Using the following alignment report:\t\t> SRR5836475_1_bismark_bt2_PE_report.txt <\n",
            "Processing alignment report SRR5836475_1_bismark_bt2_PE_report.txt ...\n",
            "Complete\n",
            "\n",
            "No deduplication report file specified, skipping this step\n",
            "No splitting report file specified, skipping this step\n",
            "No M-bias report file specified, skipping this step\n",
            "No nucleotide coverage report file specified, skipping this step\n",
            "==============================================================================================================\n",
            "\n",
            "\n"
          ]
        }
      ],
      "source": [
        "! bismark2report -mbias_report s_8_cell.deduplicated.M-bias.txt\n"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! head s_8_cell.deduplicated.bedGraph"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "tI5AUYTPliqA",
        "outputId": "63e700e9-e990-4981-f53f-2f3aa5facfa8"
      },
      "execution_count": 38,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "head: cannot open 's_8_cell.deduplicated.bedGraph' for reading: No such file or directory\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! bismark2report"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "vnEpvtdxIy1E",
        "outputId": "a4a2d9d9-6e6c-4a66-c3a2-bf169d3b9d64"
      },
      "execution_count": 39,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Found 3 alignment reports in current directory. Now trying to figure out whether there are corresponding optional reports\n",
            "\n",
            "Writing Bismark HTML report to >> SRR3824222_1_bismark_bt2_PE_report.html <<\n",
            "\n",
            "==============================================================================================================\n",
            "Using the following alignment report:\t\t> SRR3824222_1_bismark_bt2_PE_report.txt <\n",
            "Processing alignment report SRR3824222_1_bismark_bt2_PE_report.txt ...\n",
            "Complete\n",
            "\n",
            "No deduplication report file specified, skipping this step\n",
            "No splitting report file specified, skipping this step\n",
            "No M-bias report file specified, skipping this step\n",
            "No nucleotide coverage report file specified, skipping this step\n",
            "==============================================================================================================\n",
            "\n",
            "\n",
            "\n",
            "Writing Bismark HTML report to >> SRR5836473_1_bismark_bt2_PE_report.html <<\n",
            "\n",
            "==============================================================================================================\n",
            "Using the following alignment report:\t\t> SRR5836473_1_bismark_bt2_PE_report.txt <\n",
            "Processing alignment report SRR5836473_1_bismark_bt2_PE_report.txt ...\n",
            "Complete\n",
            "\n",
            "No deduplication report file specified, skipping this step\n",
            "No splitting report file specified, skipping this step\n",
            "No M-bias report file specified, skipping this step\n",
            "No nucleotide coverage report file specified, skipping this step\n",
            "==============================================================================================================\n",
            "\n",
            "\n",
            "\n",
            "Writing Bismark HTML report to >> SRR5836475_1_bismark_bt2_PE_report.html <<\n",
            "\n",
            "==============================================================================================================\n",
            "Using the following alignment report:\t\t> SRR5836475_1_bismark_bt2_PE_report.txt <\n",
            "Processing alignment report SRR5836475_1_bismark_bt2_PE_report.txt ...\n",
            "Complete\n",
            "\n",
            "No deduplication report file specified, skipping this step\n",
            "No splitting report file specified, skipping this step\n",
            "No M-bias report file specified, skipping this step\n",
            "No nucleotide coverage report file specified, skipping this step\n",
            "==============================================================================================================\n",
            "\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! ls *bedGraph.gz | xargs -tI{} gzip -d {}"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "8sfumZfyJvkk",
        "outputId": "5d872310-c4c2-4c51-ae8b-b7cf56e2ab03"
      },
      "execution_count": 40,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "gzip -d s_8_cell.deduplicated.bedGraph.gz\n",
            "gzip -d s_epiblast.deduplicated.bedGraph.gz\n",
            "gzip -d s_icm.deduplicated.bedGraph.gz\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! head s_8_cell.deduplicated.bedGraph"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "NdppCQOVJ1AI",
        "outputId": "b0f7ea1e-2845-4cb7-fff4-8e6c49bc32ef"
      },
      "execution_count": 41,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "track type=bedGraph\n",
            "chr11\t3050030\t3050031\t50\n",
            "chr11\t3050047\t3050048\t66.6666666666667\n",
            "chr11\t3050048\t3050049\t50\n",
            "chr11\t3050223\t3050224\t87.5\n",
            "chr11\t3050224\t3050225\t53.3333333333333\n",
            "chr11\t3050285\t3050286\t66.6666666666667\n",
            "chr11\t3050286\t3050287\t33.3333333333333\n",
            "chr11\t3050309\t3050310\t62.5\n",
            "chr11\t3050310\t3050311\t50\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "import matplotlib.pyplot as plt\n",
        "import pandas as pd"
      ],
      "metadata": {
        "id": "9_LUjn0yKLsn"
      },
      "execution_count": 42,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Распределение метилирования по хромосоме у 8_cell"
      ],
      "metadata": {
        "id": "f9dHvarKKcRq"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "df = pd.read_csv(\"s_8_cell.deduplicated.bedGraph\", delimiter='\\t', skiprows=1, header=None)\n",
        "plt.figure(figsize=[8, 5])\n",
        "plt.title(\"8cell\", fontsize=14)\n",
        "plt.hist(df[3], bins=100, density=True)\n",
        "plt.show()"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 469
        },
        "id": "DNbXy-wYLmc4",
        "outputId": "ade3307b-7a50-44ce-e1a0-72ee46aa362b"
      },
      "execution_count": 46,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<Figure size 800x500 with 1 Axes>"
            ],
            "image/png": "iVBORw0KGgoAAAANSUhEUgAAAp8AAAHECAYAAAB/SUGhAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjcuMSwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/bCgiHAAAACXBIWXMAAA9hAAAPYQGoP6dpAAAiBElEQVR4nO3df3TV5X3A8U8CJgExRGUmgmjUsqLTghLJidqqx0zcYVVW59ChcjKPblOcGnUFtaTT2eCPUlrlyGRjO7NaqGfqqlYsRtFjG0H54W9p15aBaALoJAg1OPLdH53XRoIQwCe59PU653sOee5z730uD+Dbb+43tyDLsiwAACCBwp5eAAAAvz/EJwAAyYhPAACSEZ8AACQjPgEASEZ8AgCQjPgEACAZ8QkAQDLiEwCAZMQnQB4pKCiIU089tdPYN7/5zSgoKIiFCxf2yJoAukN8AuyELMviwQcfjNNOOy0OPvjg6N+/f3zxi1+Mv/7rv45f/epXPb08gLwhPgF2wrXXXhvnnHNOrFixIsaNGxdXXHFFHH744TF79uwYOXJkvPrqqz29RIC80LenFwDQ27W0tMSMGTPisMMOi5deeikGDhyYu+073/lO1NfXx/Tp02POnDk9uEqA/ODMJ8AOrFy5Mjo6OuKkk07qFJ4REX/6p38aERHr1q3rNL527dq45ppr4otf/GL069cvDjjggKiuro477rhjm8d/+eWX47zzzouDDz44ioqK4rDDDosrrrgi3n333c/vRQH0EGc+AXZg2LBhUVRUFD/96U+jra0tSktLc7c9+uijERFx+umn58ZWrFgRp512Wrzzzjtx8sknx7hx42LTpk3x2muvxbe+9a249tprc3N/9KMfxV/8xV9EYWFhnH322TF06NB4/fXX46677oonnngiFi1aFPvvv3+6FwvwOROfADtw4IEHxrRp0+Kaa66J4cOHx9lnnx2lpaXx0ksvxVNPPRWXXXZZTJo0KTf/ggsuiHfeeSfuueeeuOSSSzo91ltvvZX79bvvvhsXXnhhDBo0KH7605/GYYcdlrtt7ty5cf7558fUqVPjzjvv/PxfJEAivu0OsBOuvvrqmDt3bnzwwQcxa9asuO222+KJJ56I6urq+Mu//Mvo2/e3/y+/ePHiePHFF+MrX/nKNuEZEXHIIYfkfv3v//7v0dbWFo2NjZ3CMyLivPPOi+OPPz7mzp37+b4wgMSc+QTYCTfddFP84z/+Y9x0001xwQUXRFlZWSxfvjyuvvrqOPXUU+M//uM/4qyzzorFixdHRMQZZ5yxw8d8/vnnIyJi0aJF8ctf/nKb2z/88MNYv359rF+/PgYNGrRnXxBADxGfADvw5JNPRkNDQ1x99dUxefLk3PjJJ58cjzzySBxxxBFxzTXXxFlnnRUbNmyIiIghQ4bs8HHfe++9iIiYOXPmZ87btGmT+AT2Gr7tDrADjz/+eEREnHbaadvcVlFREcOHD4//+q//ig8++CDKysoiImLNmjU7fNyPL1x65ZVXIsuy7R6f/pY8QD4TnwA7sGXLlojY9scpfWzdunVRWFgY++yzT4wePToiIn7yk5/s8HGrq6sjIqK5uXkPrRSg9xOfADtw0kknRUTE9OnTc99W/9isWbPirbfeipqamiguLo4TTjghTjjhhHj22Wdj9uzZ2zzW754Rrauri/322y9uuOGGeO2117aZu3nz5tz7QgH2Ft7zCbAD5557btx9993x7LPPxh/+4R/GWWedFWVlZbF06dJ46qmnol+/fjF9+vTc/Pvuuy9OPfXUuPTSS+Pee++Nmpqa+PDDD+O1116LZcuW5X54/B/8wR/ED37wgzj33HNjxIgRceaZZ8bw4cOjvb09Vq5cGc8880yceOKJMX/+/J566QB7nPgE2IE+ffrET37yk/jOd74TP/zhD+P++++PLVu2RHl5eVxwwQVx/fXXx1FHHZWbP2zYsFi6dGk0NjbGI488EjNmzIgBAwbEsGHD4sYbb+z02GPHjo1ly5bF7bffHk8++WQsWLAg9t133zjkkEOirq4uLrjggtQvF+BzVZBlWdbTiwAA4PeD93wCAJCM+AQAIBnxCQBAMuITAIBkxCcAAMmITwAAksmLn/PZ0dERb7/9duy3335RUFDQ08sBAOBTsiyLjRs3xuDBg6OwcPvnN/MiPt9+++0YOnRoTy8DAIAdWL16dRxyyCHbvT0v4nO//faLiN++mNLS0h5eDQAAn9bW1hZDhw7Nddv25EV8fvyt9tLSUvEJANCL7egtki44AgAgGfEJAEAy4hMAgGTEJwAAyYhPAACSEZ8AACQjPgEASEZ8AgCQjPgEACAZ8QkAQDLiEwCAZMQnAADJiE8AAJIRnwAAJCM+AQBIRnwCAJBM355eQG9VOfmxLsdXThubeCUAAHsPZz4BAEhGfAIAkIz4BAAgGfEJAEAy4hMAgGTEJwAAyYhPAACSEZ8AACQjPgEASEZ8AgCQjPgEACAZ8QkAQDLiEwCAZMQnAADJiE8AAJIRnwAAJCM+AQBIRnwCAJCM+AQAIBnxCQBAMuITAIBkxCcAAMmITwAAkhGfAAAkIz4BAEhGfAIAkIz4BAAgGfEJAEAy4hMAgGTEJwAAyYhPAACSEZ8AACQjPgEASEZ8AgCQzC7F58yZM6OysjJKSkqiuro6Fi9evFP3mzt3bhQUFMS4ceN25WkBAMhz3Y7PefPmRX19fTQ0NMTSpUtjxIgRMWbMmFi7du1n3m/lypVx7bXXxpe//OVdXiwAAPmt2/E5ffr0uOSSS6Kuri6OPvromDVrVvTv3z/mzJmz3fts3bo1JkyYEP/wD/8QRxxxxG4tGACA/NWt+NyyZUssWbIkamtrP3mAwsKora2N5ubm7d7vpptuioMOOiguvvjiXV8pAAB5r293Jq9fvz62bt0a5eXlncbLy8vjzTff7PI+zz33XPzLv/xLLF++fKefp729Pdrb23Nft7W1dWeZAAD0Up/r1e4bN26MCy+8MGbPnh2DBg3a6fs1NjbGwIEDc8fQoUM/x1UCAJBKt858Dho0KPr06ROtra2dxltbW6OiomKb+b/85S9j5cqV8dWvfjU31tHR8dsn7ts3VqxYEUceeeQ295syZUrU19fnvm5raxOgAAB7gW7FZ1FRUYwaNSqamppyPy6po6MjmpqaYtKkSdvMHz58eLzyyiudxm688cbYuHFjfPe7391uUBYXF0dxcXF3lgYAQB7oVnxGRNTX18fEiROjqqoqRo8eHTNmzIhNmzZFXV1dRERcdNFFMWTIkGhsbIySkpI45phjOt2/rKwsImKbcQAA9n7djs/x48fHunXrYurUqdHS0hIjR46M+fPn5y5CWrVqVRQW+uAkAAC2VZBlWdbTi9iRtra2GDhwYGzYsCFKS0uTPGfl5Me6HF85bWyS5wcAyCc722tOUQIAkIz4BAAgGfEJAEAy4hMAgGTEJwAAyYhPAACSEZ8AACQjPgEASEZ8AgCQjPgEACAZ8QkAQDLiEwCAZMQnAADJiE8AAJIRnwAAJCM+AQBIRnwCAJCM+AQAIBnxCQBAMuITAIBkxCcAAMmITwAAkhGfAAAkIz4BAEhGfAIAkIz4BAAgGfEJAEAy4hMAgGTEJwAAyYhPAACSEZ8AACQjPgEASEZ8AgCQjPgEACAZ8QkAQDLiEwCAZMQnAADJiE8AAJIRnwAAJCM+AQBIRnwCAJCM+AQAIBnxCQBAMuITAIBkxCcAAMmITwAAkhGfAAAkIz4BAEhGfAIAkIz4BAAgGfEJAEAy4hMAgGTEJwAAyYhPAACSEZ8AACQjPgEASEZ8AgCQjPgEACAZ8QkAQDLiEwCAZMQnAADJiE8AAJIRnwAAJCM+AQBIRnwCAJCM+AQAIBnxCQBAMuITAIBkxCcAAMmITwAAkhGfAAAks0vxOXPmzKisrIySkpKorq6OxYsXb3fugw8+GFVVVVFWVhb77rtvjBw5Mu69995dXjAAAPmr2/E5b968qK+vj4aGhli6dGmMGDEixowZE2vXru1y/gEHHBA33HBDNDc3x8svvxx1dXVRV1cXTzzxxG4vHgCA/FKQZVnWnTtUV1fHCSecEHfddVdERHR0dMTQoUPjiiuuiMmTJ+/UYxx//PExduzYuPnmm3dqfltbWwwcODA2bNgQpaWl3VnuLquc/FiX4yunjU3y/AAA+WRne61bZz63bNkSS5Ysidra2k8eoLAwamtro7m5eYf3z7IsmpqaYsWKFfGVr3xlu/Pa29ujra2t0wEAQP7rVnyuX78+tm7dGuXl5Z3Gy8vLo6WlZbv327BhQwwYMCCKiopi7Nixceedd8Yf//Efb3d+Y2NjDBw4MHcMHTq0O8sEAKCXSnK1+3777RfLly+PF154IW655Zaor6+PhQsXbnf+lClTYsOGDblj9erVKZYJAMDnrG93Jg8aNCj69OkTra2tncZbW1ujoqJiu/crLCyML3zhCxERMXLkyHjjjTeisbExTj311C7nFxcXR3FxcXeWBgBAHujWmc+ioqIYNWpUNDU15cY6OjqiqakpampqdvpxOjo6or29vTtPDQDAXqBbZz4jIurr62PixIlRVVUVo0ePjhkzZsSmTZuirq4uIiIuuuiiGDJkSDQ2NkbEb9+/WVVVFUceeWS0t7fHj3/847j33nvj7rvv3rOvBACAXq/b8Tl+/PhYt25dTJ06NVpaWmLkyJExf/783EVIq1atisLCT06obtq0KS677LJ46623ol+/fjF8+PD4/ve/H+PHj99zrwIAgLzQ7Z/z2RP8nE8AgN7tc/k5nwAAsDvEJwAAyYhPAACSEZ8AACQjPgEASEZ8AgCQjPgEACAZ8QkAQDLiEwCAZMQnAADJiE8AAJIRnwAAJCM+AQBIRnwCAJCM+AQAIBnxCQBAMuITAIBkxCcAAMmITwAAkhGfAAAkIz4BAEhGfAIAkIz4BAAgGfEJAEAy4hMAgGTEJwAAyYhPAACSEZ8AACQjPgEASEZ8AgCQTN+eXgAAALuncvJjXY6vnDY28Up2zJlPAACSEZ8AACQjPgEASEZ8AgCQjPgEACAZ8QkAQDLiEwCAZMQnAADJiE8AAJIRnwAAJCM+AQBIRnwCAJCM+AQAIBnxCQBAMuITAIBkxCcAAMmITwAAkhGfAAAkIz4BAEhGfAIAkIz4BAAgGfEJAEAy4hMAgGTEJwAAyYhPAACSEZ8AACQjPgEASEZ8AgCQjPgEACAZ8QkAQDLiEwCAZMQnAADJiE8AAJIRnwAAJCM+AQBIRnwCAJCM+AQAIBnxCQBAMuITAIBkxCcAAMnsUnzOnDkzKisro6SkJKqrq2Px4sXbnTt79uz48pe/HPvvv3/sv//+UVtb+5nzAQDYe3U7PufNmxf19fXR0NAQS5cujREjRsSYMWNi7dq1Xc5fuHBhnH/++fH0009Hc3NzDB06NM4444xYs2bNbi8eAID80u34nD59elxyySVRV1cXRx99dMyaNSv69+8fc+bM6XL+fffdF5dddlmMHDkyhg8fHv/8z/8cHR0d0dTUtNuLBwAgv3QrPrds2RJLliyJ2traTx6gsDBqa2ujubl5px5j8+bN8dFHH8UBBxyw3Tnt7e3R1tbW6QAAIP91Kz7Xr18fW7dujfLy8k7j5eXl0dLSslOP8fWvfz0GDx7cKWA/rbGxMQYOHJg7hg4d2p1lAgDQSyW92n3atGkxd+7ceOihh6KkpGS786ZMmRIbNmzIHatXr064SgAAPi99uzN50KBB0adPn2htbe003traGhUVFZ953zvuuCOmTZsWTz75ZHzpS1/6zLnFxcVRXFzcnaUBAJAHunXms6ioKEaNGtXpYqGPLx6qqanZ7v1uu+22uPnmm2P+/PlRVVW166sFACCvdevMZ0REfX19TJw4MaqqqmL06NExY8aM2LRpU9TV1UVExEUXXRRDhgyJxsbGiIi49dZbY+rUqXH//fdHZWVl7r2hAwYMiAEDBuzBlwIAQG/X7fgcP358rFu3LqZOnRotLS0xcuTImD9/fu4ipFWrVkVh4ScnVO++++7YsmVL/Pmf/3mnx2loaIhvfvObu7d6AADySrfjMyJi0qRJMWnSpC5vW7hwYaevV65cuStPAQDAXshnuwMAkIz4BAAgGfEJAEAy4hMAgGTEJwAAyYhPAACSEZ8AACQjPgEASEZ8AgCQjPgEACAZ8QkAQDLiEwCAZMQnAADJiE8AAJIRnwAAJCM+AQBIRnwCAJCM+AQAIBnxCQBAMuITAIBkxCcAAMmITwAAkhGfAAAkIz4BAEhGfAIAkIz4BAAgGfEJAEAy4hMAgGTEJwAAyYhPAACSEZ8AACQjPgEASEZ8AgCQjPgEACAZ8QkAQDLiEwCAZMQnAADJiE8AAJIRnwAAJCM+AQBIRnwCAJCM+AQAIBnxCQBAMuITAIBkxCcAAMmITwAAkhGfAAAkIz4BAEhGfAIAkIz4BAAgGfEJAEAy4hMAgGTEJwAAyYhPAACSEZ8AACQjPgEASEZ8AgCQjPgEACCZvj29AIDfV5WTH9tmbOW0sT2wEoB0nPkEACAZ8QkAQDLiEwCAZMQnAADJiE8AAJIRnwAAJCM+AQBIRnwCAJCM+AQAIBnxCQBAMuITAIBkdik+Z86cGZWVlVFSUhLV1dWxePHi7c597bXX4pxzzonKysooKCiIGTNm7OpaAQDIc92Oz3nz5kV9fX00NDTE0qVLY8SIETFmzJhYu3Ztl/M3b94cRxxxREybNi0qKip2e8EAAOSvbsfn9OnT45JLLom6uro4+uijY9asWdG/f/+YM2dOl/NPOOGEuP322+O8886L4uLi3V4wAAD5q1vxuWXLlliyZEnU1tZ+8gCFhVFbWxvNzc17fHEAAOxd+nZn8vr162Pr1q1RXl7eaby8vDzefPPNPbao9vb2aG9vz33d1ta2xx4bAICe0yuvdm9sbIyBAwfmjqFDh/b0kgAA2AO6FZ+DBg2KPn36RGtra6fx1tbWPXox0ZQpU2LDhg25Y/Xq1XvssQEA6Dndis+ioqIYNWpUNDU15cY6Ojqiqakpampq9tiiiouLo7S0tNMBAED+69Z7PiMi6uvrY+LEiVFVVRWjR4+OGTNmxKZNm6Kuri4iIi666KIYMmRINDY2RsRvL1J6/fXXc79es2ZNLF++PAYMGBBf+MIX9uBLAQCgt+t2fI4fPz7WrVsXU6dOjZaWlhg5cmTMnz8/dxHSqlWrorDwkxOqb7/9dhx33HG5r++4446444474pRTTomFCxfu/isAACBvdDs+IyImTZoUkyZN6vK2TwdlZWVlZFm2K08DAMBeplde7Q4AwN5JfAIAkIz4BAAgGfEJAEAy4hMAgGTEJwAAyYhPAACSEZ8AACQjPgEASEZ8AgCQjPgEACAZ8QkAQDLiEwCAZMQnAADJiE8AAJIRnwAAJCM+AQBIRnwCAJCM+AQAIBnxCQBAMuITAIBkxCcAAMmITwAAkhGfAAAkIz4BAEhGfAIAkIz4BAAgGfEJAEAy4hMAgGTEJwAAyYhPAACSEZ8AACQjPgEASEZ8AgCQjPgEACAZ8QkAQDJ9e3oBQM+pnPxYl+Mrp41NvBIAfl848wkAQDLiEwCAZMQnAADJiE8AAJIRnwAAJONqdwD2an6qA/QuznwCAJCM+AQAIBnxCQBAMuITAIBkxCcAAMmITwAAkhGfAAAkIz4BAEhGfAIAkIz4BAAgGfEJAEAy4hMAgGTEJwAAyYhPAACSEZ8AACQjPgEASEZ8AgCQjPgEACCZvj29ANhbVE5+bJuxldPG9sBKAKD3cuYTAIBkxCcAAMn4tjsA5Alv72Fv4MwnAADJiE8AAJIRnwAAJCM+AQBIRnwCAJCMq93Za7kqNL91tX8R9jCf2EOgK7t05nPmzJlRWVkZJSUlUV1dHYsXL/7M+Q888EAMHz48SkpK4thjj40f//jHu7RYACCdysmPbXPA7up2fM6bNy/q6+ujoaEhli5dGiNGjIgxY8bE2rVru5z/s5/9LM4///y4+OKLY9myZTFu3LgYN25cvPrqq7u9eAAA8ku343P69OlxySWXRF1dXRx99NExa9as6N+/f8yZM6fL+d/97nfjzDPPjOuuuy6OOuqouPnmm+P444+Pu+66a7cXDwBAfunWez63bNkSS5YsiSlTpuTGCgsLo7a2Npqbm7u8T3Nzc9TX13caGzNmTDz88MPbfZ729vZob2/Pfb1hw4aIiGhra+vOcndLR/vmLsdTruH3wTENT3Q5/uo/jNntx+5qDz/P/dsTz9fV78ee+L3Ynt7857y7a0v9e7cnpP4z2h174u9mb/nzlXod/l3rLB//buaj3vD37ePnyrLssydm3bBmzZosIrKf/exnncavu+66bPTo0V3eZ5999snuv//+TmMzZ87MDjrooO0+T0NDQxYRDofD4XA4HI48O1avXv2ZPdkrr3afMmVKp7OlHR0d8d5778WBBx4YBQUFn/vzt7W1xdChQ2P16tVRWlr6uT8fe549zG/2L//Zw/xnD/Nf6j3Msiw2btwYgwcP/sx53YrPQYMGRZ8+faK1tbXTeGtra1RUVHR5n4qKim7Nj4goLi6O4uLiTmNlZWXdWeoeUVpa6i9cnrOH+c3+5T97mP/sYf5LuYcDBw7c4ZxuXXBUVFQUo0aNiqamptxYR0dHNDU1RU1NTZf3qamp6TQ/ImLBggXbnQ8AwN6r2992r6+vj4kTJ0ZVVVWMHj06ZsyYEZs2bYq6urqIiLjoootiyJAh0djYGBERV155ZZxyyinx7W9/O8aOHRtz586NF198Me655549+0oAAOj1uh2f48ePj3Xr1sXUqVOjpaUlRo4cGfPnz4/y8vKIiFi1alUUFn5yQvXEE0+M+++/P2688ca4/vrrY9iwYfHwww/HMcccs+dexR5WXFwcDQ0N23zrn/xhD/Ob/ct/9jD/2cP811v3sCDLdnQ9PAAA7Bm79PGaAACwK8QnAADJiE8AAJIRnwAAJCM+P2XmzJlRWVkZJSUlUV1dHYsXL+7pJbEdjY2NccIJJ8R+++0XBx10UIwbNy5WrFjRac6HH34Yl19+eRx44IExYMCAOOecc7b50AN6h2nTpkVBQUFcddVVuTH71/utWbMmLrjggjjwwAOjX79+ceyxx8aLL76Yuz3Lspg6dWocfPDB0a9fv6itrY1f/OIXPbhiftfWrVvjG9/4Rhx++OHRr1+/OPLII+Pmm2/u9Nnc9rB3efbZZ+OrX/1qDB48OAoKCuLhhx/udPvO7Nd7770XEyZMiNLS0igrK4uLL744Pvjgg2SvQXz+jnnz5kV9fX00NDTE0qVLY8SIETFmzJhYu3ZtTy+NLjzzzDNx+eWXx/PPPx8LFiyIjz76KM4444zYtGlTbs7VV18djzzySDzwwAPxzDPPxNtvvx1f+9rXenDVdOWFF16If/qnf4ovfelLncbtX+/2P//zP3HSSSfFPvvsE48//ni8/vrr8e1vfzv233//3Jzbbrstvve978WsWbNi0aJFse+++8aYMWPiww8/7MGV87Fbb7017r777rjrrrvijTfeiFtvvTVuu+22uPPOO3Nz7GHvsmnTphgxYkTMnDmzy9t3Zr8mTJgQr732WixYsCAeffTRePbZZ+PSSy9N9RIiPvOT33/PjB49Orv88stzX2/dujUbPHhw1tjY2IOrYmetXbs2i4jsmWeeybIsy95///1sn332yR544IHcnDfeeCOLiKy5ubmnlsmnbNy4MRs2bFi2YMGC7JRTTsmuvPLKLMvsXz74+te/np188snbvb2joyOrqKjIbr/99tzY+++/nxUXF2c/+MEPUiyRHRg7dmz2V3/1V53Gvva1r2UTJkzIsswe9nYRkT300EO5r3dmv15//fUsIrIXXnghN+fxxx/PCgoKsjVr1iRZtzOf/2/Lli2xZMmSqK2tzY0VFhZGbW1tNDc39+DK2FkbNmyIiIgDDjggIiKWLFkSH330Uac9HT58eBx66KH2tBe5/PLLY+zYsZ32KcL+5YMf/ehHUVVVFeeee24cdNBBcdxxx8Xs2bNzt//617+OlpaWTns4cODAqK6utoe9xIknnhhNTU3x85//PCIiXnrppXjuuefiT/7kTyLCHuabndmv5ubmKCsri6qqqtyc2traKCwsjEWLFiVZZ7c/4WhvtX79+ti6dWvuk5o+Vl5eHm+++WYPrYqd1dHREVdddVWcdNJJuU/PamlpiaKioigrK+s0t7y8PFpaWnpglXza3LlzY+nSpfHCCy9sc5v96/1+9atfxd133x319fVx/fXXxwsvvBB/93d/F0VFRTFx4sTcPnX176o97B0mT54cbW1tMXz48OjTp09s3bo1brnllpgwYUJEhD3MMzuzXy0tLXHQQQd1ur1v375xwAEHJNtT8cle4fLLL49XX301nnvuuZ5eCjtp9erVceWVV8aCBQuipKSkp5fDLujo6Iiqqqr41re+FRERxx13XLz66qsxa9asmDhxYg+vjp3xwx/+MO677764//7744/+6I9i+fLlcdVVV8XgwYPtIZ8b33b/f4MGDYo+ffpscyVta2trVFRU9NCq2BmTJk2KRx99NJ5++uk45JBDcuMVFRWxZcuWeP/99zvNt6e9w5IlS2Lt2rVx/PHHR9++faNv377xzDPPxPe+973o27dvlJeX279e7uCDD46jjz6609hRRx0Vq1atiojI7ZN/V3uv6667LiZPnhznnXdeHHvssXHhhRfG1VdfHY2NjRFhD/PNzuxXRUXFNhdS/+///m+89957yfZUfP6/oqKiGDVqVDQ1NeXGOjo6oqmpKWpqanpwZWxPlmUxadKkeOihh+Kpp56Kww8/vNPto0aNin322afTnq5YsSJWrVplT3uB008/PV555ZVYvnx57qiqqooJEybkfm3/ereTTjppmx9v9vOf/zwOO+ywiIg4/PDDo6KiotMetrW1xaJFi+xhL7F58+YoLOycAn369ImOjo6IsIf5Zmf2q6amJt5///1YsmRJbs5TTz0VHR0dUV1dnWahSS5ryhNz587NiouLs3/7t3/LXn/99ezSSy/NysrKspaWlp5eGl3427/922zgwIHZwoULs3feeSd3bN68OTfnb/7mb7JDDz00e+qpp7IXX3wxq6mpyWpqanpw1XyW373aPcvsX2+3ePHirG/fvtktt9yS/eIXv8juu+++rH///tn3v//93Jxp06ZlZWVl2X/+539mL7/8cnb22Wdnhx9+ePab3/ymB1fOxyZOnJgNGTIke/TRR7Nf//rX2YMPPpgNGjQo+/u///vcHHvYu2zcuDFbtmxZtmzZsiwisunTp2fLli3L/vu//zvLsp3brzPPPDM77rjjskWLFmXPPfdcNmzYsOz8889P9hrE56fceeed2aGHHpoVFRVlo0ePzp5//vmeXhLbERFdHv/6r/+am/Ob3/wmu+yyy7L9998/69+/f/Znf/Zn2TvvvNNzi+YzfTo+7V/v98gjj2THHHNMVlxcnA0fPjy75557Ot3e0dGRfeMb38jKy8uz4uLi7PTTT89WrFjRQ6vl09ra2rIrr7wyO/TQQ7OSkpLsiCOOyG644Yasvb09N8ce9i5PP/10l//tmzhxYpZlO7df7777bnb++ednAwYMyEpLS7O6urps48aNyV5DQZb9zscYAADA58h7PgEASEZ8AgCQjPgEACAZ8QkAQDLiEwCAZMQnAADJiE8AAJIRnwAAJCM+AQBIRnwCAJCM+AQAIBnxCQBAMv8HQAVgZCDDlkcAAAAASUVORK5CYII=\n"
          },
          "metadata": {}
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Распределение метилирования по хромосоме у icm"
      ],
      "metadata": {
        "id": "IVdhknG8LhRT"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "df = pd.read_csv(\"s_icm.deduplicated.bedGraph\", delimiter='\\t', skiprows=1, header=None)\n",
        "plt.figure(figsize=[8, 5])\n",
        "plt.title(\"icm\", fontsize=14)\n",
        "plt.hist(df[3], bins=100, density=True)\n",
        "plt.show()"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 469
        },
        "id": "qa8cgid5L4vb",
        "outputId": "2f8935e8-8219-48a2-9126-52e1b38aecb9"
      },
      "execution_count": 47,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<Figure size 800x500 with 1 Axes>"
            ],
            "image/png": "iVBORw0KGgoAAAANSUhEUgAAAp8AAAHECAYAAAB/SUGhAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjcuMSwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/bCgiHAAAACXBIWXMAAA9hAAAPYQGoP6dpAAAkwklEQVR4nO3df3RW9X3A8U8SIBExAc1IgMbGXxt1KiCRLFqnW2PpDu2pq91BxwqLHT1TbNGcOUEraK0Gf3FoC5XqGetOrZPZqW2V0nLijx6PURS0/sbWX1A0AaYkiG1ik7s/tsZGEuXB+E1CX69z7jnmPt/73O/jF/XtfXKfJy/LsiwAACCB/IGeAAAAfzzEJwAAyYhPAACSEZ8AACQjPgEASEZ8AgCQjPgEACAZ8QkAQDLiEwCAZMQnwD56+eWXIy8vL/7xH/9xoKcCMGSITwAAksnz3e4A++btt9+OF154IUpKSmLcuHEDPR2AIUF8AgCQjLfdAfZRX7/zuWvXrrj88svjuOOOi5EjR0ZJSUlMmTIlLr300nj77be7x+Xl5cWpp54aW7dujb//+7+P0tLSOOigg2LGjBnx4osvRkTEs88+G6effnocfPDBcdBBB8XnP//5aGlpSfkyAfqV+AToR9u2bYtp06bFZZddFgUFBXHOOefE2WefHeXl5XH11VfH7t27e4x/44034uMf/3i89NJLMWfOnDj11FNjzZo1cdppp8VTTz0VJ554Yrz55ptx9tlnR1VVVfz3f/93nHXWWQP06gA+uGEDPQGA/cm5554bzz33XFx88cVx5ZVX9nispaUlRo0a1WPfE088ERdccEEsXbq0x3PccMMNcfLJJ8dll10W8+fPj4iILMvi05/+dKxZsyY2btwYxx9//If/ggD6mSufAP2kubk5br/99jjiiCPisssu2+PxsrKyGDas5//zjxo1Kr7+9a/32Pf7K5uHHHJIfOUrX+nen5eXF2eeeWZERPziF7/o59kDpCE+AfrJo48+GlmWxV/91V/F8OHD9+qYo446KkaOHNlj3+/vnD/uuOMiLy+v18deffXVfpgxQHriE6CftLa2RkTEhAkT9vqY4uLiPfb9/uroez32hzcuAQwl4hOgn4wePToiIrZu3TqwEwEYxMQnQD+pqqqK/Pz8uPfee12ZBOiD+AToJ2VlZXHGGWfECy+8EJdffvkej2/bti1+97vfDcDMAAYPH7UE0I++/e1vx1NPPRVXXnllrFmzJv76r/86siyL559/Pn72s59FS0tL99vzAH+MxCdAPyotLY2HHnoorrvuurjtttti+fLlUVRUFIcddlgsWLAgDjzwwIGeIsCA8t3uAAAk43c+AQBIRnwCAJCM+AQAIBnxCQBAMuITAIBkxCcAAMkMic/57OrqildffTUOOuigyMvLG+jpAADwLlmWxa5du2L8+PGRn9/39c0hEZ+vvvpqVFRUDPQ0AAB4H1u2bImPfOQjfT4+JOLzoIMOioj/ezHFxcUDPBsAAN6tra0tKioqurutL0MiPn//VntxcbH4BAAYxN7vVyTdcAQAQDLiEwCAZMQnAADJiE8AAJIRnwAAJCM+AQBIRnwCAJCM+AQAIBnxCQBAMuITAIBkxCcAAMmITwAAkhGfAAAkIz4BAEhGfAIAkIz4BAAgmWEDPYHBqnLB3b3uf3nJjMQzAQDYf7jyCQBAMuITAIBkxCcAAMmITwAAkhGfAAAkIz4BAEhGfAIAkIz4BAAgmX2KzxUrVkRlZWUUFRVFdXV1rF+//j3H79y5M+bNmxfjxo2LwsLC+NM//dNYs2bNPk0YAIChK+dvOFq9enXU19fHypUro7q6OpYtWxbTp0+PTZs2xdixY/cY39HREaeddlqMHTs2fvCDH8SECRPilVdeidGjR/fH/AEAGEJyjs+lS5fG3Llzo66uLiIiVq5cGXfffXesWrUqFixYsMf4VatWxeuvvx4PPvhgDB8+PCIiKisrP9isAQAYknJ6272joyM2bNgQtbW17zxBfn7U1tZGU1NTr8f86Ec/ipqampg3b16UlZXFMcccE1dddVV0dnb2eZ729vZoa2vrsQEAMPTlFJ87duyIzs7OKCsr67G/rKwsmpubez3mxRdfjB/84AfR2dkZa9asiUsvvTSuv/76+PrXv97neRoaGqKkpKR7q6ioyGWaAAAMUh/63e5dXV0xduzYuPHGG2Pq1Kkxc+bMuOSSS2LlypV9HrNw4cJobW3t3rZs2fJhTxMAgARy+p3P0tLSKCgoiJaWlh77W1paory8vNdjxo0bF8OHD4+CgoLufR/72Meiubk5Ojo6YsSIEXscU1hYGIWFhblMDQCAISCnK58jRoyIqVOnRmNjY/e+rq6uaGxsjJqaml6POemkk+JXv/pVdHV1de97/vnnY9y4cb2GJwAA+6+c33avr6+Pm266Kf7jP/4jnn322TjnnHNi9+7d3Xe/z549OxYuXNg9/pxzzonXX3895s+fH88//3zcfffdcdVVV8W8efP671UAADAk5PxRSzNnzozt27fHokWLorm5OSZPnhxr167tvglp8+bNkZ//TtNWVFTET3/607jgggviuOOOiwkTJsT8+fPjoosu6r9XAQDAkJCXZVk20JN4P21tbVFSUhKtra1RXFyc5JyVC+7udf/LS2YkOT8AwFCyt73mu90BAEhGfAIAkIz4BAAgGfEJAEAy4hMAgGTEJwAAyYhPAACSEZ8AACQjPgEASEZ8AgCQjPgEACAZ8QkAQDLiEwCAZMQnAADJiE8AAJIRnwAAJCM+AQBIRnwCAJCM+AQAIBnxCQBAMuITAIBkxCcAAMmITwAAkhGfAAAkIz4BAEhGfAIAkIz4BAAgGfEJAEAy4hMAgGTEJwAAyYhPAACSEZ8AACQjPgEASEZ8AgCQjPgEACAZ8QkAQDLiEwCAZMQnAADJiE8AAJIRnwAAJCM+AQBIRnwCAJCM+AQAIBnxCQBAMuITAIBkxCcAAMmITwAAkhGfAAAks0/xuWLFiqisrIyioqKorq6O9evX9zn2u9/9buTl5fXYioqK9nnCAAAMXTnH5+rVq6O+vj4WL14cGzdujEmTJsX06dNj27ZtfR5TXFwcr732Wvf2yiuvfKBJAwAwNOUcn0uXLo25c+dGXV1dHH300bFy5coYOXJkrFq1qs9j8vLyory8vHsrKyv7QJMGAGBoyik+Ozo6YsOGDVFbW/vOE+TnR21tbTQ1NfV53Jtvvhkf/ehHo6KiIj772c/G008//Z7naW9vj7a2th4bAABDX07xuWPHjujs7NzjymVZWVk0Nzf3esyf/dmfxapVq+KHP/xh3HzzzdHV1RUnnnhi/PrXv+7zPA0NDVFSUtK9VVRU5DJNAAAGqQ/9bveampqYPXt2TJ48OU455ZS4/fbb40/+5E/iO9/5Tp/HLFy4MFpbW7u3LVu2fNjTBAAggWG5DC4tLY2CgoJoaWnpsb+lpSXKy8v36jmGDx8eU6ZMiV/96ld9jiksLIzCwsJcpgYAwBCQ05XPESNGxNSpU6OxsbF7X1dXVzQ2NkZNTc1ePUdnZ2c8+eSTMW7cuNxmCgDAkJfTlc+IiPr6+pgzZ05UVVXFtGnTYtmyZbF79+6oq6uLiIjZs2fHhAkToqGhISIivva1r8Vf/MVfxJFHHhk7d+6Ma6+9Nl555ZX4p3/6p/59JQAADHo5x+fMmTNj+/btsWjRomhubo7JkyfH2rVru29C2rx5c+Tnv3NB9Y033oi5c+dGc3NzjBkzJqZOnRoPPvhgHH300f33KgAAGBLysizLBnoS76etrS1KSkqitbU1iouLk5yzcsHdve5/ecmMJOcHABhK9rbXfLc7AADJiE8AAJIRnwAAJCM+AQBIRnwCAJCM+AQAIBnxCQBAMuITAIBkxCcAAMmITwAAkhGfAAAkIz4BAEhGfAIAkIz4BAAgGfEJAEAy4hMAgGTEJwAAyYhPAACSEZ8AACQjPgEASEZ8AgCQjPgEACAZ8QkAQDLiEwCAZMQnAADJiE8AAJIRnwAAJCM+AQBIRnwCAJCM+AQAIBnxCQBAMuITAIBkxCcAAMmITwAAkhGfAAAkIz4BAEhGfAIAkIz4BAAgGfEJAEAy4hMAgGTEJwAAyYhPAACSEZ8AACQjPgEASEZ8AgCQjPgEACAZ8QkAQDL7FJ8rVqyIysrKKCoqiurq6li/fv1eHXfrrbdGXl5enH766ftyWgAAhric43P16tVRX18fixcvjo0bN8akSZNi+vTpsW3btvc87uWXX45/+Zd/iZNPPnmfJwsAwNCWc3wuXbo05s6dG3V1dXH00UfHypUrY+TIkbFq1ao+j+ns7IxZs2bF5ZdfHocffvgHmjAAAENXTvHZ0dERGzZsiNra2neeID8/amtro6mpqc/jvva1r8XYsWPji1/84r7PFACAIW9YLoN37NgRnZ2dUVZW1mN/WVlZPPfcc70e88ADD8S//du/xeOPP77X52lvb4/29vbun9va2nKZJgAAg9SHerf7rl274gtf+ELcdNNNUVpautfHNTQ0RElJSfdWUVHxIc4SAIBUcrryWVpaGgUFBdHS0tJjf0tLS5SXl+8x/oUXXoiXX345PvOZz3Tv6+rq+r8TDxsWmzZtiiOOOGKP4xYuXBj19fXdP7e1tQlQAID9QE7xOWLEiJg6dWo0NjZ2f1xSV1dXNDY2xnnnnbfH+IkTJ8aTTz7ZY99Xv/rV2LVrV3zjG9/oMygLCwujsLAwl6kBADAE5BSfERH19fUxZ86cqKqqimnTpsWyZcti9+7dUVdXFxERs2fPjgkTJkRDQ0MUFRXFMccc0+P40aNHR0TssR8AgP1fzvE5c+bM2L59eyxatCiam5tj8uTJsXbt2u6bkDZv3hz5+b44CQCAPeVlWZYN9CTeT1tbW5SUlERra2sUFxcnOWflgrt73f/ykhlJzg8AMJTsba+5RAkAQDLiEwCAZMQnAADJiE8AAJIRnwAAJCM+AQBIRnwCAJCM+AQAIBnxCQBAMuITAIBkxCcAAMmITwAAkhGfAAAkIz4BAEhGfAIAkIz4BAAgGfEJAEAy4hMAgGTEJwAAyYhPAACSEZ8AACQjPgEASEZ8AgCQjPgEACAZ8QkAQDLiEwCAZMQnAADJiE8AAJIRnwAAJCM+AQBIRnwCAJCM+AQAIBnxCQBAMuITAIBkxCcAAMmITwAAkhGfAAAkIz4BAEhGfAIAkIz4BAAgGfEJAEAy4hMAgGTEJwAAyYhPAACSEZ8AACQjPgEASEZ8AgCQjPgEACCZfYrPFStWRGVlZRQVFUV1dXWsX7++z7G33357VFVVxejRo+PAAw+MyZMnx/e+9719njAAAENXzvG5evXqqK+vj8WLF8fGjRtj0qRJMX369Ni2bVuv4w8++OC45JJLoqmpKZ544omoq6uLurq6+OlPf/qBJw8AwNCSl2VZlssB1dXVccIJJ8Ty5csjIqKrqysqKiriy1/+cixYsGCvnuP444+PGTNmxBVXXLFX49va2qKkpCRaW1ujuLg4l+nus8oFd/e6/+UlM5KcHwBgKNnbXsvpymdHR0ds2LAhamtr33mC/Pyora2Npqam9z0+y7JobGyMTZs2xV/+5V/mcmoAAPYDw3IZvGPHjujs7IyysrIe+8vKyuK5557r87jW1taYMGFCtLe3R0FBQXz729+O0047rc/x7e3t0d7e3v1zW1tbLtMEAGCQyik+99VBBx0Ujz/+eLz55pvR2NgY9fX1cfjhh8epp57a6/iGhoa4/PLLU0wNAICEcorP0tLSKCgoiJaWlh77W1paory8vM/j8vPz48gjj4yIiMmTJ8ezzz4bDQ0NfcbnwoULo76+vvvntra2qKioyGWqAAAMQjn9zueIESNi6tSp0djY2L2vq6srGhsbo6amZq+fp6urq8fb6u9WWFgYxcXFPTYAAIa+nN92r6+vjzlz5kRVVVVMmzYtli1bFrt37466urqIiJg9e3ZMmDAhGhoaIuL/3kKvqqqKI444Itrb22PNmjXxve99L2644Yb+fSUAAAx6OcfnzJkzY/v27bFo0aJobm6OyZMnx9q1a7tvQtq8eXPk579zQXX37t1x7rnnxq9//es44IADYuLEiXHzzTfHzJkz++9VAAAwJOT8OZ8Dwed8AgAMbh/K53wCAMAHIT4BAEhGfAIAkIz4BAAgGfEJAEAy4hMAgGTEJwAAyYhPAACSEZ8AACQjPgEASEZ8AgCQjPgEACAZ8QkAQDLiEwCAZMQnAADJiE8AAJIRnwAAJCM+AQBIRnwCAJCM+AQAIBnxCQBAMuITAIBkxCcAAMmITwAAkhGfAAAkIz4BAEhGfAIAkIz4BAAgGfEJAEAy4hMAgGTEJwAAyYhPAACSEZ8AACQjPgEASEZ8AgCQjPgEACAZ8QkAQDLiEwCAZMQnAADJiE8AAJIRnwAAJCM+AQBIRnwCAJCM+AQAIBnxCQBAMuITAIBkxCcAAMmITwAAktmn+FyxYkVUVlZGUVFRVFdXx/r16/sce9NNN8XJJ58cY8aMiTFjxkRtbe17jgcAYP+Vc3yuXr066uvrY/HixbFx48aYNGlSTJ8+PbZt29br+Pvuuy/OOuusuPfee6OpqSkqKirik5/8ZGzduvUDTx4AgKElL8uyLJcDqqur44QTTojly5dHRERXV1dUVFTEl7/85ViwYMH7Ht/Z2RljxoyJ5cuXx+zZs/fqnG1tbVFSUhKtra1RXFycy3T3WeWCu3vd//KSGUnODwAwlOxtr+V05bOjoyM2bNgQtbW17zxBfn7U1tZGU1PTXj3HW2+9FW+//XYcfPDBuZwaAID9wLBcBu/YsSM6OzujrKysx/6ysrJ47rnn9uo5Lrroohg/fnyPgH239vb2aG9v7/65ra0tl2kCADBIJb3bfcmSJXHrrbfGHXfcEUVFRX2Oa2hoiJKSku6toqIi4SwBAPiw5BSfpaWlUVBQEC0tLT32t7S0RHl5+Xsee91118WSJUviZz/7WRx33HHvOXbhwoXR2travW3ZsiWXaQIAMEjlFJ8jRoyIqVOnRmNjY/e+rq6uaGxsjJqamj6Pu+aaa+KKK66ItWvXRlVV1fuep7CwMIqLi3tsAAAMfTn9zmdERH19fcyZMyeqqqpi2rRpsWzZsti9e3fU1dVFRMTs2bNjwoQJ0dDQEBERV199dSxatChuueWWqKysjObm5oiIGDVqVIwaNaofXwoAAINdzvE5c+bM2L59eyxatCiam5tj8uTJsXbt2u6bkDZv3hz5+e9cUL3hhhuio6MjPv/5z/d4nsWLF8dll132wWYPAMCQkvPnfA4En/MJADC4fSif8wkAAB+E+AQAIBnxCQBAMuITAIBkxCcAAMmITwAAkhGfAAAkIz4BAEhGfAIAkIz4BAAgGfEJAEAy4hMAgGTEJwAAyYhPAACSEZ8AACQjPgEASEZ8AgCQjPgEACAZ8QkAQDLiEwCAZMQnAADJiE8AAJIRnwAAJCM+AQBIRnwCAJCM+AQAIBnxCQBAMuITAIBkxCcAAMmITwAAkhGfAAAkIz4BAEhGfAIAkIz4BAAgGfEJAEAy4hMAgGTEJwAAyYhPAACSEZ8AACQjPgEASEZ8AgCQjPgEACAZ8QkAQDLiEwCAZMQnAADJiE8AAJIRnwAAJCM+AQBIZp/ic8WKFVFZWRlFRUVRXV0d69ev73Ps008/HWeccUZUVlZGXl5eLFu2bF/nCgDAEJdzfK5evTrq6+tj8eLFsXHjxpg0aVJMnz49tm3b1uv4t956Kw4//PBYsmRJlJeXf+AJAwAwdOUcn0uXLo25c+dGXV1dHH300bFy5coYOXJkrFq1qtfxJ5xwQlx77bVx5plnRmFh4QeeMAAAQ1dO8dnR0REbNmyI2trad54gPz9qa2ujqamp3ybV3t4ebW1tPTYAAIa+nOJzx44d0dnZGWVlZT32l5WVRXNzc79NqqGhIUpKSrq3ioqKfntuAAAGzqC8233hwoXR2travW3ZsmWgpwQAQD8Ylsvg0tLSKCgoiJaWlh77W1pa+vVmosLCQr8fCgCwH8rpyueIESNi6tSp0djY2L2vq6srGhsbo6ampt8nBwDA/iWnK58REfX19TFnzpyoqqqKadOmxbJly2L37t1RV1cXERGzZ8+OCRMmRENDQ0T8301KzzzzTPdfb926NR5//PEYNWpUHHnkkf34UgAAGOxyjs+ZM2fG9u3bY9GiRdHc3ByTJ0+OtWvXdt+EtHnz5sjPf+eC6quvvhpTpkzp/vm6666L6667Lk455ZS47777PvgrAABgyMjLsiwb6Em8n7a2tigpKYnW1tYoLi5Ocs7KBXf3uv/lJTOSnB8AYCjZ214blHe7AwCwfxKfAAAkIz4BAEhGfAIAkEzOd7sD0D96u7HRTY3A/s6VTwAAkhGfAAAkIz4BAEhGfAIAkIz4BAAgGfEJAEAy4hMAgGTEJwAAyYhPAACSEZ8AACQjPgEASMZ3u8Mfsd6+WzzC94sD8OFx5RMAgGRc+QQAGOKG0jtZrnwCAJCM+AQAIBnxCQBAMuITAIBkxCcAAMmITwAAkhGfAAAkIz4BAEhGfAIAkIz4BAAgGV+vCf2kt682G4xfawYAA8mVTwAAkhGfAAAkIz4BAEhGfAIAkIz4BAAgGfEJAEAy4hMAgGTEJwAAyfiQeQD2a719AUSEL4GAgSI+2W/5xiEAGHy87Q4AQDLiEwCAZMQnAADJiE8AAJJxwxEwKLlDGWD/5MonAADJuPLJgHJ1CwD+uIhPABgifH4x+4N9ett9xYoVUVlZGUVFRVFdXR3r169/z/G33XZbTJw4MYqKiuLYY4+NNWvW7NNkARg6Khfc3esG/HHL+crn6tWro76+PlauXBnV1dWxbNmymD59emzatCnGjh27x/gHH3wwzjrrrGhoaIhPf/rTccstt8Tpp58eGzdujGOOOaZfXgSDi/8zBwD6knN8Ll26NObOnRt1dXUREbFy5cq4++67Y9WqVbFgwYI9xn/jG9+IT33qU3HhhRdGRMQVV1wR69ati+XLl8fKlSs/4PRh/yLcAdjf5RSfHR0dsWHDhli4cGH3vvz8/KitrY2mpqZej2lqaor6+voe+6ZPnx533nlnn+dpb2+P9vb27p9bW1sjIqKtrS2X6X4gXe1v9bo/5RwGwjGLf7rXY5+6fHqv+3v7e9fX37cP8+9zLvPoD/1xvsEw5/44Z19/jvr6M9ObXOfW2zlzOd9ASL3euRiINfywDJZ59IfUf2aG4j9Xf6wGw5/z358ry7L3HpjlYOvWrVlEZA8++GCP/RdeeGE2bdq0Xo8ZPnx4dsstt/TYt2LFimzs2LF9nmfx4sVZRNhsNpvNZrPZhti2ZcuW9+zJQXm3+8KFC3tcLe3q6orXX389DjnkkMjLy/vQz9/W1hYVFRWxZcuWKC4u/tDPR/+zhkOb9Rv6rOHQZw2HvtRrmGVZ7Nq1K8aPH/+e43KKz9LS0igoKIiWlpYe+1taWqK8vLzXY8rLy3MaHxFRWFgYhYWFPfaNHj06l6n2i+LiYv/ADXHWcGizfkOfNRz6rOHQl3INS0pK3ndMTh+1NGLEiJg6dWo0NjZ27+vq6orGxsaoqanp9Ziampoe4yMi1q1b1+d4AAD2Xzm/7V5fXx9z5syJqqqqmDZtWixbtix2797dfff77NmzY8KECdHQ0BAREfPnz49TTjklrr/++pgxY0bceuut8eijj8aNN97Yv68EAIBBL+f4nDlzZmzfvj0WLVoUzc3NMXny5Fi7dm2UlZVFRMTmzZsjP/+dC6onnnhi3HLLLfHVr341Lr744jjqqKPizjvvHNSf8VlYWBiLFy/e461/hg5rOLRZv6HPGg591nDoG6xrmJdl73c/PAAA9I99+npNAADYF+ITAIBkxCcAAMmITwAAkhGf77JixYqorKyMoqKiqK6ujvXr1w/0lOhDQ0NDnHDCCXHQQQfF2LFj4/TTT49Nmzb1GPPb3/425s2bF4ccckiMGjUqzjjjjD2+9IDBYcmSJZGXlxfnn39+9z7rN/ht3bo1/uEf/iEOOeSQOOCAA+LYY4+NRx99tPvxLMti0aJFMW7cuDjggAOitrY2fvnLXw7gjPlDnZ2dcemll8Zhhx0WBxxwQBxxxBFxxRVX9Phubms4uPz85z+Pz3zmMzF+/PjIy8uLO++8s8fje7Ner7/+esyaNSuKi4tj9OjR8cUvfjHefPPNZK9BfP6B1atXR319fSxevDg2btwYkyZNiunTp8e2bdsGemr04v7774958+bFQw89FOvWrYu33347PvnJT8bu3bu7x1xwwQXx4x//OG677ba4//7749VXX43Pfe5zAzhrevPII4/Ed77znTjuuON67Ld+g9sbb7wRJ510UgwfPjx+8pOfxDPPPBPXX399jBkzpnvMNddcE9/85jdj5cqV8fDDD8eBBx4Y06dPj9/+9rcDOHN+7+qrr44bbrghli9fHs8++2xcffXVcc0118S3vvWt7jHWcHDZvXt3TJo0KVasWNHr43uzXrNmzYqnn3461q1bF3fddVf8/Oc/jy996UupXkLEe37z+x+ZadOmZfPmzev+ubOzMxs/fnzW0NAwgLNib23bti2LiOz+++/PsizLdu7cmQ0fPjy77bbbusc8++yzWURkTU1NAzVN3mXXrl3ZUUcdla1bty475ZRTsvnz52dZZv2Ggosuuij7+Mc/3ufjXV1dWXl5eXbttdd279u5c2dWWFiY/ed//meKKfI+ZsyYkZ199tk99n3uc5/LZs2alWWZNRzsIiK74447un/em/V65plnsojIHnnkke4xP/nJT7K8vLxs69atSebtyuf/6+joiA0bNkRtbW33vvz8/KitrY2mpqYBnBl7q7W1NSIiDj744IiI2LBhQ7z99ts91nTixIlx6KGHWtNBZN68eTFjxowe6xRh/YaCH/3oR1FVVRV/93d/F2PHjo0pU6bETTfd1P34Sy+9FM3NzT3WsKSkJKqrq63hIHHiiSdGY2NjPP/88xER8Ytf/CIeeOCB+Ju/+ZuIsIZDzd6sV1NTU4wePTqqqqq6x9TW1kZ+fn48/PDDSeaZ8zcc7a927NgRnZ2d3d/U9HtlZWXx3HPPDdCs2FtdXV1x/vnnx0knndT97VnNzc0xYsSIGD16dI+xZWVl0dzcPACz5N1uvfXW2LhxYzzyyCN7PGb9Br8XX3wxbrjhhqivr4+LL744HnnkkfjKV74SI0aMiDlz5nSvU2//XrWGg8OCBQuira0tJk6cGAUFBdHZ2RlXXnllzJo1KyLCGg4xe7Nezc3NMXbs2B6PDxs2LA4++OBkayo+2S/MmzcvnnrqqXjggQcGeirspS1btsT8+fNj3bp1UVRUNNDTYR90dXVFVVVVXHXVVRERMWXKlHjqqadi5cqVMWfOnAGeHXvjv/7rv+L73/9+3HLLLfHnf/7n8fjjj8f5558f48ePt4Z8aLzt/v9KS0ujoKBgjztpW1paory8fIBmxd4477zz4q677op77703PvKRj3TvLy8vj46Ojti5c2eP8dZ0cNiwYUNs27Ytjj/++Bg2bFgMGzYs7r///vjmN78Zw4YNi7KyMus3yI0bNy6OPvroHvs+9rGPxebNmyMiutfJv1cHrwsvvDAWLFgQZ555Zhx77LHxhS98IS644IJoaGiICGs41OzNepWXl+9xI/Xvfve7eP3115Otqfj8fyNGjIipU6dGY2Nj976urq5obGyMmpqaAZwZfcmyLM4777y444474p577onDDjusx+NTp06N4cOH91jTTZs2xebNm63pIPCJT3winnzyyXj88ce7t6qqqpg1a1b3X1u/we2kk07a4+PNnn/++fjoRz8aERGHHXZYlJeX91jDtra2ePjhh63hIPHWW29Ffn7PFCgoKIiurq6IsIZDzd6sV01NTezcuTM2bNjQPeaee+6Jrq6uqK6uTjPRJLc1DRG33nprVlhYmH33u9/NnnnmmexLX/pSNnr06Ky5uXmgp0YvzjnnnKykpCS77777stdee617e+utt7rH/PM//3N26KGHZvfcc0/26KOPZjU1NVlNTc0Azpr38od3u2eZ9Rvs1q9fnw0bNiy78sors1/+8pfZ97///WzkyJHZzTff3D1myZIl2ejRo7Mf/vCH2RNPPJF99rOfzQ477LDsN7/5zQDOnN+bM2dONmHChOyuu+7KXnrppez222/PSktLs3/913/tHmMNB5ddu3Zljz32WPbYY49lEZEtXbo0e+yxx7JXXnkly7K9W69PfepT2ZQpU7KHH344e+CBB7KjjjoqO+uss5K9BvH5Lt/61reyQw89NBsxYkQ2bdq07KGHHhroKdGHiOh1+/d///fuMb/5zW+yc889NxszZkw2cuTI7G//9m+z1157beAmzXt6d3xav8Hvxz/+cXbMMcdkhYWF2cSJE7Mbb7yxx+NdXV3ZpZdempWVlWWFhYXZJz7xiWzTpk0DNFvera2tLZs/f3526KGHZkVFRdnhhx+eXXLJJVl7e3v3GGs4uNx77729/rdvzpw5WZbt3Xr9z//8T3bWWWdlo0aNyoqLi7O6urps165dyV5DXpb9wdcYAADAh8jvfAIAkIz4BAAgGfEJAEAy4hMAgGTEJwAAyYhPAACSEZ8AACQjPgEASEZ8AgCQjPgEACAZ8QkAQDLiEwCAZP4XC9ODGH1cxxYAAAAASUVORK5CYII=\n"
          },
          "metadata": {}
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Распределение метилирования по хромосоме у epiblast"
      ],
      "metadata": {
        "id": "r98TDvunLhYy"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "df = pd.read_csv(\"s_epiblast.deduplicated.bedGraph\", delimiter='\\t', skiprows=1, header=None)\n",
        "plt.figure(figsize=[8, 5])\n",
        "plt.title(\"epiblast\", fontsize=14)\n",
        "plt.hist(df[3], bins=100, density=True)\n",
        "plt.show()"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 469
        },
        "id": "8UtOq59kL73r",
        "outputId": "e6e5d012-c4d0-4b8e-cb9f-725d1393238b"
      },
      "execution_count": 48,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<Figure size 800x500 with 1 Axes>"
            ],
            "image/png": "iVBORw0KGgoAAAANSUhEUgAAAp8AAAHECAYAAAB/SUGhAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjcuMSwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/bCgiHAAAACXBIWXMAAA9hAAAPYQGoP6dpAAAk/ElEQVR4nO3dfXCV1Z3A8V/CS0AhCYIEQTC+sItWBSWSUtfVjllxF5m16g46rCDt2NkKVk1rC2qJrcsGrToosmCdVbcvKutUra84GAF1DaIgtUVFdy1CwQRYJaGooOTZP3aNjQRMBE4S/Hxm7gyce557z+Xw8uW5eW5ysizLAgAAEsht6wUAAPDlIT4BAEhGfAIAkIz4BAAgGfEJAEAy4hMAgGTEJwAAyYhPAACSEZ8AACQjPgH20GmnnRY5OTlNxu6+++7IycmJu+++u8WPU1xcHMXFxXt3cQDtjPgE+BK69tprIycnJxYtWtTWSwG+ZDq39QIAOrqf//zn8f7777f1MgA6BPEJsIcGDRrU1ksA6DC87Q58aTzzzDMxZsyY6NOnT+Tl5cXgwYPjmmuuaXLWctGiRZGTkxPXXnttPPfcc3HaaadFz549o7CwMM4999z4r//6r50et7mv+fxzv/nNb2LEiBFxwAEHxMEHHxzf/OY3o7a2tkVrXr9+fVRUVMRXv/rV6Nu3b+Tl5UVxcXFccsklsWHDhp3m19XVxbRp0+KYY46JHj16RH5+fhx11FExYcKEePvttxvX++Mf/zgiIr7+9a9HTk5O5OTk+HpTIAlnPoEvhTlz5sSkSZOisLAwxowZE3379o2XXnoppk+fHgsXLoyFCxdG165dG+cvWbIkKisr48wzz4xLL700Vq5cGQ8++GA8++yzsWTJkjjiiCNa9Ly//vWv48knn4zzzjsvysrKYsmSJXHXXXfFs88+G0uXLo1evXrt9vhnnnkmbrrppjj99NOjtLQ0unTpEi+//HLMmTMnnnzyyVi+fHkUFBRERESWZTFq1Kh44YUX4uSTT44zzzwzcnNz4+23346HH344LrzwwjjssMPioosuioiIxYsXx4QJExqjs7CwsNW/rgCtlgHs51auXJl17tw5Gzp0aLZp06Ym91VWVmYRkd14441ZlmXZwoULs4jIIiKbO3duk7lz587NIiI766yzmoyfeuqp2Wf/Or3rrrsaH2f+/PlN7psyZUoWEdnkyZObjB922GHZYYcd1mSstrY227Jly06v6d///d+ziMj++Z//uXHslVdeySIiO/vss3ea/+GHHzZ5nIqKiiwisoULF+40F2Bf8rY7sN+7/fbb4+OPP45Zs2ZF7969m9z3gx/8IA4++OC49957m4z/xV/8RVx88cVNxi6++OIYPHhwPPbYY7Fx48YWPXdZWVmMGjWqydjVV18dhYWF8fOf/zwaGhp2e3zfvn2jR48eO41feOGFkZ+fH0899dRO93Xv3n2nsby8vGYfByA1b7sD+70lS5ZERMSTTz4ZVVVVO93fpUuXeP3115uMnXzyyZGb2/T/57m5uXHyySfHm2++Gb/97W+jrKzsc5/7lFNO2WmsR48eMWzYsFi0aFG89dZbcdRRR+32MR544IG4/fbbY/ny5fHee+/Fjh07Gu9bv35944+PPvroOP744+Pee++NP/7xj3H22WfHaaedFsOGDdvptQC0FfEJ7PfefffdiIiYPn16i48pKira7XhdXV2Sx7npppvi+9//fhx88MFxxhlnxKGHHtp4ZnPmzJmxbdu2xrmdO3eOp59+Oq699tr49a9/Hd/73vciIuLggw+OyZMnx9VXXx2dOnVq0boB9hXxCez38vPzIyKivr4+evbs2aJjdnU1+ifjn1zksy8f5+OPP47rrrsuDjnkkFixYkX07du38b4sy+KGG27Y6ZjevXvHrFmz4tZbb43XX389nn766Zg1a1ZUVFREly5dYurUqS1aN8C+4n0YYL9XWloaEZ++/d4S//mf/7nT12M2NDTE888/Hzk5OTF06NAWPc6zzz6709if/vSnWLFiReTn5+/2qvlNmzZFXV1djBw5skl4RkS89NJL8cEHH+zy2JycnDj66KNj0qRJsWDBgoiIePjhhxvv/+QM6J+/hQ+QgvgE9nuXXHJJdO7cOS699NJYs2bNTvdv3rw5Xn755SZjb7zxRtxxxx1Nxu6444544403YvTo0XHwwQe36LmfeuqpePLJJ5uMTZ8+PTZv3hzjx4/f7ddi9u3bN7p37x7Lly9v8lmk7733Xlx66aU7zV+9enWsXr16p/FPzrJ269atceyggw6KiIi1a9e26HUA7C3edgf2e8cee2z867/+a3znO9+Jv/zLv4y/+7u/iyOPPDK2bNkSb731VixevDguuuiimDt3buMxo0aNiu9+97vx+OOPx1e+8pVYuXJlPPLII9GnT5+45ZZbWvzcZ511VowZMybOO++8KC4ujiVLlsTChQvjyCOPjJ/85Ce7PTY3NzcuueSSuOmmm2Lo0KExZsyYqK+vjyeeeCIOO+yw6N+/f5P5K1asiHPOOSdGjBgRxxxzTPTr1y/WrVsXDz30UOTm5sYVV1zROPeTD5e/6qqrYuXKlVFQUBCFhYUxefLkFr82gC+krT/rCSCVpUuXZueff37Wv3//rEuXLlmfPn2yE088MZsyZUr22muvZVn26ed8VlRUZM8++2x26qmnZgceeGCWn5+ffeMb38jefPPNnR53d5/zedddd2UPPfRQdtJJJ2Xdu3fPevfunV100UXZO++8s9PjNPc5n9u3b8+mT5+eDR48OMvLy8sGDRqUfe9738u2bNmy0/y1a9dmU6ZMyb761a9mffv2zbp27ZoNGjQoO+ecc7Lq6uqdnu/uu+/OjjvuuCwvLy+LiJ2eG2BfyMmyLGvb/AVoPxYtWhRf//rXo6KiIq699tq2Xg7AfsfXfAIAkIz4BAAgGfEJAEAyvuYTAIBknPkEACAZ8QkAQDId4kPmGxoaYv369dGzZ8/Iyclp6+UAAPAZWZbFli1bon///rv97m0dIj7Xr18fAwcObOtlAADwOdauXRuHHnroLu/vEPHZs2fPiPi/F5Ofn9/GqwEA4LPq6+tj4MCBjd22Kx0iPj95qz0/P198AgC0Y5/3JZIuOAIAIBnxCQBAMuITAIBkxCcAAMmITwAAkhGfAAAkIz4BAEhGfAIAkIz4BAAgGfEJAEAy4hMAgGTEJwAAyYhPAACSEZ8AACQjPgEASEZ8AgCQTOe2XgAAAHumeMpjzY6vnjE68Uo+nzOfAAAkIz4BAEhGfAIAkIz4BAAgGfEJAEAy4hMAgGTEJwAAyYhPAACSEZ8AACQjPgEASEZ8AgCQjPgEACAZ8QkAQDLiEwCAZMQnAADJiE8AAJIRnwAAJCM+AQBIRnwCAJCM+AQAIBnxCQBAMuITAIBkxCcAAMmITwAAkhGfAAAkIz4BAEhGfAIAkIz4BAAgGfEJAEAy4hMAgGTEJwAAyYhPAACSEZ8AACQjPgEASEZ8AgCQjPgEACAZ8QkAQDLiEwCAZMQnAADJiE8AAJIRnwAAJCM+AQBIRnwCAJCM+AQAIJkvFJ+zZ8+O4uLi6NatW5SWlsbSpUtbdNx9990XOTk5cfbZZ3+RpwUAoINrdXzOmzcvysvLo6KiIpYvXx5Dhw6NUaNGxYYNG3Z73OrVq+P73/9+nHLKKV94sQAAdGytjs+bb745Lr744pg4cWIcc8wxMXfu3DjggAPizjvv3OUxO3bsiHHjxsWPf/zjOOKII/ZowQAAdFytis/t27fHsmXLoqys7NMHyM2NsrKyqK6u3uVxP/nJT6Jv377xrW9964uvFACADq9zayZv2rQpduzYEUVFRU3Gi4qK4vXXX2/2mOeeey7+7d/+LVasWNHi59m2bVts27at8ef19fWtWSYAAO3UPr3afcuWLXHhhRfGHXfcEX369GnxcZWVlVFQUNB4Gzhw4D5cJQAAqbTqzGefPn2iU6dOUVtb22S8trY2+vXrt9P8//7v/47Vq1fHmDFjGscaGhr+74k7d45Vq1bFkUceudNxU6dOjfLy8saf19fXC1AAgP1Aq+Kza9euMXz48Kiqqmr8uKSGhoaoqqqKyZMn7zR/yJAh8bvf/a7J2DXXXBNbtmyJW265ZZdBmZeXF3l5ea1ZGgAAHUCr4jMiory8PCZMmBAlJSUxYsSImDlzZmzdujUmTpwYERHjx4+PAQMGRGVlZXTr1i2OPfbYJscXFhZGROw0DgDA/q/V8Tl27NjYuHFjTJs2LWpqamLYsGExf/78xouQ1qxZE7m5vnESAAA7y8myLGvrRXye+vr6KCgoiLq6usjPz2/r5QAAtCvFUx5rdnz1jNHJ1tDSXnOKEgCAZMQnAADJiE8AAJIRnwAAJCM+AQBIRnwCAJCM+AQAIBnxCQBAMuITAIBkxCcAAMmITwAAkhGfAAAkIz4BAEhGfAIAkIz4BAAgGfEJAEAy4hMAgGTEJwAAyYhPAACSEZ8AACQjPgEASEZ8AgCQjPgEACAZ8QkAQDLiEwCAZMQnAADJiE8AAJIRnwAAJCM+AQBIRnwCAJCM+AQAIBnxCQBAMuITAIBkxCcAAMmITwAAkhGfAAAkIz4BAEhGfAIAkIz4BAAgGfEJAEAy4hMAgGTEJwAAyYhPAACSEZ8AACQjPgEASEZ8AgCQjPgEACAZ8QkAQDLiEwCAZMQnAADJiE8AAJIRnwAAJCM+AQBIRnwCAJCM+AQAIBnxCQBAMuITAIBkxCcAAMmITwAAkhGfAAAkIz4BAEhGfAIAkIz4BAAgGfEJAEAy4hMAgGTEJwAAyYhPAACSEZ8AACQjPgEASOYLxefs2bOjuLg4unXrFqWlpbF06dJdzn3ggQeipKQkCgsL48ADD4xhw4bFL37xiy+8YAAAOq5Wx+e8efOivLw8KioqYvny5TF06NAYNWpUbNiwodn5Bx10UFx99dVRXV0dr7zySkycODEmTpwYTz755B4vHgCAjiUny7KsNQeUlpbGSSedFLfddltERDQ0NMTAgQPj0ksvjSlTprToMU488cQYPXp0XHfddS2aX19fHwUFBVFXVxf5+fmtWS4AwH6veMpjzY6vnjE62Rpa2mutOvO5ffv2WLZsWZSVlX36ALm5UVZWFtXV1Z97fJZlUVVVFatWrYq//uu/3uW8bdu2RX19fZMbAAAdX6vic9OmTbFjx44oKipqMl5UVBQ1NTW7PK6uri569OgRXbt2jdGjR8esWbPib/7mb3Y5v7KyMgoKChpvAwcObM0yAQBop5Jc7d6zZ89YsWJFvPjiizF9+vQoLy+PRYsW7XL+1KlTo66urvG2du3aFMsEAGAf69yayX369IlOnTpFbW1tk/Ha2tro16/fLo/Lzc2No446KiIihg0bFq+99lpUVlbGaaed1uz8vLy8yMvLa83SAADoAFp15rNr164xfPjwqKqqahxraGiIqqqqGDlyZIsfp6GhIbZt29aapwYAYD/QqjOfERHl5eUxYcKEKCkpiREjRsTMmTNj69atMXHixIiIGD9+fAwYMCAqKysj4v++frOkpCSOPPLI2LZtWzz++OPxi1/8IubMmbN3XwkAAO1eq+Nz7NixsXHjxpg2bVrU1NTEsGHDYv78+Y0XIa1ZsyZycz89obp169a45JJL4o9//GN07949hgwZEr/85S9j7Nixe+9VAADQIbT6cz7bgs/5BADYtf32cz4BAGBPiE8AAJIRnwAAJCM+AQBIRnwCAJCM+AQAIBnxCQBAMuITAIBkxCcAAMmITwAAkhGfAAAkIz4BAEhGfAIAkIz4BAAgGfEJAEAy4hMAgGTEJwAAyYhPAACSEZ8AACQjPgEASEZ8AgCQjPgEACAZ8QkAQDLiEwCAZMQnAADJiE8AAJIRnwAAJCM+AQBIRnwCAJCM+AQAIBnxCQBAMuITAIBkxCcAAMmITwAAkhGfAAAkIz4BAEhGfAIAkIz4BAAgGfEJAEAy4hMAgGTEJwAAyYhPAACSEZ8AACQjPgEASEZ8AgCQjPgEACAZ8QkAQDLiEwCAZMQnAADJiE8AAJIRnwAAJCM+AQBIRnwCAJCM+AQAIBnxCQBAMuITAIBkxCcAAMmITwAAkhGfAAAkIz4BAEhGfAIAkIz4BAAgGfEJAEAy4hMAgGTEJwAAyYhPAACSEZ8AACQjPgEASOYLxefs2bOjuLg4unXrFqWlpbF06dJdzr3jjjvilFNOiV69ekWvXr2irKxst/MBANh/tTo+582bF+Xl5VFRURHLly+PoUOHxqhRo2LDhg3Nzl+0aFFccMEFsXDhwqiuro6BAwfGGWecEevWrdvjxQMA0LHkZFmWteaA0tLSOOmkk+K2226LiIiGhoYYOHBgXHrppTFlypTPPX7Hjh3Rq1evuO2222L8+PEtes76+vooKCiIurq6yM/Pb81yAQD2e8VTHmt2fPWM0cnW0NJea9WZz+3bt8eyZcuirKzs0wfIzY2ysrKorq5u0WO8//778dFHH8VBBx20yznbtm2L+vr6JjcAADq+VsXnpk2bYseOHVFUVNRkvKioKGpqalr0GD/84Q+jf//+TQL2syorK6OgoKDxNnDgwNYsEwCAdirp1e4zZsyI++67Lx588MHo1q3bLudNnTo16urqGm9r165NuEoAAPaVzq2Z3KdPn+jUqVPU1tY2Ga+trY1+/frt9tgbb7wxZsyYEU899VQcf/zxu52bl5cXeXl5rVkaAAAdQKvOfHbt2jWGDx8eVVVVjWMNDQ1RVVUVI0eO3OVxN9xwQ1x33XUxf/78KCkp+eKrBQCgQ2vVmc+IiPLy8pgwYUKUlJTEiBEjYubMmbF169aYOHFiRESMHz8+BgwYEJWVlRERcf3118e0adPinnvuieLi4savDe3Ro0f06NFjL74UAADau1bH59ixY2Pjxo0xbdq0qKmpiWHDhsX8+fMbL0Jas2ZN5OZ+ekJ1zpw5sX379jjvvPOaPE5FRUVce+21e7Z6AAA6lFZ/zmdb8DmfAAC7tt9+zicAAOwJ8QkAQDLiEwCAZMQnAADJiE8AAJIRnwAAJCM+AQBIRnwCAJCM+AQAIBnxCQBAMuITAIBkxCcAAMmITwAAkhGfAAAkIz4BAEhGfAIAkIz4BAAgGfEJAEAy4hMAgGTEJwAAyYhPAACSEZ8AACQjPgEASEZ8AgCQjPgEACAZ8QkAQDLiEwCAZMQnAADJiE8AAJIRnwAAJCM+AQBIRnwCAJCM+AQAIBnxCQBAMuITAIBkxCcAAMmITwAAkhGfAAAkIz4BAEhGfAIAkIz4BAAgGfEJAEAy4hMAgGTEJwAAyYhPAACSEZ8AACQjPgEASEZ8AgCQjPgEACAZ8QkAQDLiEwCAZMQnAADJiE8AAJIRnwAAJCM+AQBIRnwCAJBM57ZeQHtVPOWxZsdXzxideCUAAPsPZz4BAEhGfAIAkIz4BAAgGfEJAEAy4hMAgGTEJwAAyYhPAACSEZ8AACQjPgEASEZ8AgCQjPgEACCZLxSfs2fPjuLi4ujWrVuUlpbG0qVLdzl35cqVce6550ZxcXHk5OTEzJkzv+haAQDo4Fodn/PmzYvy8vKoqKiI5cuXx9ChQ2PUqFGxYcOGZue///77ccQRR8SMGTOiX79+e7xgAAA6rlbH58033xwXX3xxTJw4MY455piYO3duHHDAAXHnnXc2O/+kk06Kn/70p3H++edHXl7eHi8YAICOq1XxuX379li2bFmUlZV9+gC5uVFWVhbV1dV7bVHbtm2L+vr6JjcAADq+VsXnpk2bYseOHVFUVNRkvKioKGpqavbaoiorK6OgoKDxNnDgwL322AAAtJ12ebX71KlTo66urvG2du3atl4SAAB7QefWTO7Tp0906tQpamtrm4zX1tbu1YuJ8vLyfH0oAMB+qFVnPrt27RrDhw+PqqqqxrGGhoaoqqqKkSNH7vXFAQCwf2nVmc+IiPLy8pgwYUKUlJTEiBEjYubMmbF169aYOHFiRESMHz8+BgwYEJWVlRHxfxcpvfrqq40/XrduXaxYsSJ69OgRRx111F58KQAAtHetjs+xY8fGxo0bY9q0aVFTUxPDhg2L+fPnN16EtGbNmsjN/fSE6vr16+OEE05o/PmNN94YN954Y5x66qmxaNGiPX8FAAB0GK2Oz4iIyZMnx+TJk5u977NBWVxcHFmWfZGnAQBgP9Mur3YHAGD/JD4BAEhGfAIAkIz4BAAgGfEJAEAy4hMAgGTEJwAAyYhPAACSEZ8AACQjPgEASEZ8AgCQjPgEACAZ8QkAQDLiEwCAZMQnAADJiE8AAJIRnwAAJCM+AQBIRnwCAJCM+AQAIBnxCQBAMuITAIBkxCcAAMmITwAAkhGfAAAkIz4BAEhGfAIAkIz4BAAgGfEJAEAy4hMAgGTEJwAAyYhPAACSEZ8AACQjPgEASKZzWy8AAPal4imPNTu+esboxCsBIsQnAHQYzYW0iKaj8bY7AADJiE8AAJLxtjsA0Cxv87MviE+ANuIfduDLyNvuAAAkIz4BAEhGfAIAkIz4BAAgGRccAQD7jAvr+CxnPgEASEZ8AgCQjPgEACAZ8QkAQDLiEwCAZMQnAADJ+KglAKBdaO5jmSJ8NNP+xplPAACSEZ8AACQjPgEASEZ8AgCQjPgEACAZV7sDAB2OK+M7Lmc+AQBIRnwCAJCMt90B2Ce8LQo0R3zCl5g4ACA18QkA7Df8p7r9E58AsBeJn47DXrUNFxwBAJCM+AQAIBlvu8Ne0tzbN966AYCmxCfQLvlaLNgz/gzRXn2ht91nz54dxcXF0a1btygtLY2lS5fudv79998fQ4YMiW7dusVxxx0Xjz/++BdaLABpFE95rNkbwJ5q9ZnPefPmRXl5ecydOzdKS0tj5syZMWrUqFi1alX07dt3p/nPP/98XHDBBVFZWRlnnXVW3HPPPXH22WfH8uXL49hjj90rLwKa421w6DicpYMvj1bH58033xwXX3xxTJw4MSIi5s6dG4899ljceeedMWXKlJ3m33LLLXHmmWfGlVdeGRER1113XSxYsCBuu+22mDt37h4un47OPzjsLf6zQWpf1jPBX4a/t1uzt/vT606lVfG5ffv2WLZsWUydOrVxLDc3N8rKyqK6urrZY6qrq6O8vLzJ2KhRo+Khhx7a5fNs27Yttm3b1vjzurq6iIior69vzXL3SMO295sdT7mG9uTYiid3Gvv9j0ft8dx9+evc3GPvy/3bG8/Xml+7vWFf/fo39zoiWvdaWru21Pu9N6Rec2v2ZW/83mjNY+xq7qAr7m92fF/+XtrTx92V5l5La/98N/ecu/o12pW9sYetedy9sebWzm9Oa37ftcbe+D26N7SHbvnkubIs2/3ErBXWrVuXRUT2/PPPNxm/8sorsxEjRjR7TJcuXbJ77rmnydjs2bOzvn377vJ5Kioqsohwc3Nzc3Nzc3PrYLe1a9futifb5dXuU6dObXK2tKGhId59993o3bt35OTk7PPnr6+vj4EDB8batWsjPz9/nz8fe5897NjsX8dnDzs+e9jxpd7DLMtiy5Yt0b9//93Oa1V89unTJzp16hS1tbVNxmtra6Nfv37NHtOvX79WzY+IyMvLi7y8vCZjhYWFrVnqXpGfn+8PXAdnDzs2+9fx2cOOzx52fCn3sKCg4HPntOqjlrp27RrDhw+PqqqqxrGGhoaoqqqKkSNHNnvMyJEjm8yPiFiwYMEu5wMAsP9q9dvu5eXlMWHChCgpKYkRI0bEzJkzY+vWrY1Xv48fPz4GDBgQlZWVERFx2WWXxamnnho33XRTjB49Ou6777546aWX4mc/+9nefSUAALR7rY7PsWPHxsaNG2PatGlRU1MTw4YNi/nz50dRUVFERKxZsyZycz89ofq1r30t7rnnnrjmmmviqquuisGDB8dDDz3Urj/jMy8vLyoqKnZ665+Owx52bPav47OHHZ897Pja6x7mZNnnXQ8PAAB7xxf69poAAPBFiE8AAJIRnwAAJCM+AQBIRnx+xuzZs6O4uDi6desWpaWlsXTp0rZeErtQWVkZJ510UvTs2TP69u0bZ599dqxatarJnA8//DAmTZoUvXv3jh49esS555670zc9oH2YMWNG5OTkxOWXX944Zv/av3Xr1sU//uM/Ru/evaN79+5x3HHHxUsvvdR4f5ZlMW3atDjkkEOie/fuUVZWFm+++WYbrpg/t2PHjvjRj34Uhx9+eHTv3j2OPPLIuO6665p8b2572L4888wzMWbMmOjfv3/k5OTEQw891OT+luzXu+++G+PGjYv8/PwoLCyMb33rW/GnP/0p2WsQn39m3rx5UV5eHhUVFbF8+fIYOnRojBo1KjZs2NDWS6MZixcvjkmTJsWSJUtiwYIF8dFHH8UZZ5wRW7dubZxzxRVXxCOPPBL3339/LF68ONavXx/nnHNOG66a5rz44otx++23x/HHH99k3P61b++9916cfPLJ0aVLl3jiiSfi1VdfjZtuuil69erVOOeGG26IW2+9NebOnRsvvPBCHHjggTFq1Kj48MMP23DlfOL666+POXPmxG233RavvfZaXH/99XHDDTfErFmzGufYw/Zl69atMXTo0Jg9e3az97dkv8aNGxcrV66MBQsWxKOPPhrPPPNMfPvb3071EiJ2+53fv2RGjBiRTZo0qfHnO3bsyPr3759VVla24apoqQ0bNmQRkS1evDjLsizbvHlz1qVLl+z+++9vnPPaa69lEZFVV1e31TL5jC1btmSDBw/OFixYkJ166qnZZZddlmWZ/esIfvjDH2Z/9Vd/tcv7Gxoasn79+mU//elPG8c2b96c5eXlZffee2+KJfI5Ro8enX3zm99sMnbOOedk48aNy7LMHrZ3EZE9+OCDjT9vyX69+uqrWURkL774YuOcJ554IsvJycnWrVuXZN3OfP6/7du3x7Jly6KsrKxxLDc3N8rKyqK6uroNV0ZL1dXVRUTEQQcdFBERy5Yti48++qjJng4ZMiQGDRpkT9uRSZMmxejRo5vsU4T96wgefvjhKCkpiX/4h3+Ivn37xgknnBB33HFH4/1/+MMfoqampskeFhQURGlpqT1sJ772ta9FVVVVvPHGGxER8dvf/jaee+65+Nu//duIsIcdTUv2q7q6OgoLC6OkpKRxTllZWeTm5sYLL7yQZJ2t/g5H+6tNmzbFjh07Gr9T0yeKiori9ddfb6NV0VINDQ1x+eWXx8knn9z43bNqamqia9euUVhY2GRuUVFR1NTUtMEq+az77rsvli9fHi+++OJO99m/9u+tt96KOXPmRHl5eVx11VXx4osvxne/+93o2rVrTJgwoXGfmvt71R62D1OmTIn6+voYMmRIdOrUKXbs2BHTp0+PcePGRUTYww6mJftVU1MTffv2bXJ/586d46CDDkq2p+KT/cKkSZPi97//fTz33HNtvRRaaO3atXHZZZfFggULolu3bm29HL6AhoaGKCkpiX/5l3+JiIgTTjghfv/738fcuXNjwoQJbbw6WuI//uM/4le/+lXcc8898ZWvfCVWrFgRl19+efTv398ess942/3/9enTJzp16rTTlbS1tbXRr1+/NloVLTF58uR49NFHY+HChXHooYc2jvfr1y+2b98emzdvbjLfnrYPy5Ytiw0bNsSJJ54YnTt3js6dO8fixYvj1ltvjc6dO0dRUZH9a+cOOeSQOOaYY5qMHX300bFmzZqIiMZ98vdq+3XllVfGlClT4vzzz4/jjjsuLrzwwrjiiiuisrIyIuxhR9OS/erXr99OF1J//PHH8e677ybbU/H5/7p27RrDhw+PqqqqxrGGhoaoqqqKkSNHtuHK2JUsy2Ly5Mnx4IMPxtNPPx2HH354k/uHDx8eXbp0abKnq1atijVr1tjTduD000+P3/3ud7FixYrGW0lJSYwbN67xx/avfTv55JN3+nizN954Iw477LCIiDj88MOjX79+Tfawvr4+XnjhBXvYTrz//vuRm9s0BTp16hQNDQ0RYQ87mpbs18iRI2Pz5s2xbNmyxjlPP/10NDQ0RGlpaZqFJrmsqYO47777sry8vOzuu+/OXn311ezb3/52VlhYmNXU1LT10mjGd77znaygoCBbtGhR9s477zTe3n///cY5//RP/5QNGjQoe/rpp7OXXnopGzlyZDZy5Mg2XDW78+dXu2eZ/Wvvli5dmnXu3DmbPn169uabb2a/+tWvsgMOOCD75S9/2ThnxowZWWFhYfab3/wme+WVV7K///u/zw4//PDsgw8+aMOV84kJEyZkAwYMyB599NHsD3/4Q/bAAw9kffr0yX7wgx80zrGH7cuWLVuyl19+OXv55ZeziMhuvvnm7OWXX87efvvtLMtatl9nnnlmdsIJJ2QvvPBC9txzz2WDBw/OLrjggmSvQXx+xqxZs7JBgwZlXbt2zUaMGJEtWbKkrZfELkREs7e77rqrcc4HH3yQXXLJJVmvXr2yAw44IPvGN76RvfPOO223aHbrs/Fp/9q/Rx55JDv22GOzvLy8bMiQIdnPfvazJvc3NDRkP/rRj7KioqIsLy8vO/3007NVq1a10Wr5rPr6+uyyyy7LBg0alHXr1i074ogjsquvvjrbtm1b4xx72L4sXLiw2X/7JkyYkGVZy/brf/7nf7ILLrgg69GjR5afn59NnDgx27JlS7LXkJNlf/ZtDAAAYB/yNZ8AACQjPgEASEZ8AgCQjPgEACAZ8QkAQDLiEwCAZMQnAADJiE8AAJIRnwAAJCM+AQBIRnwCAJCM+AQAIJn/BS++VrvGueQPAAAAAElFTkSuQmCC\n"
          },
          "metadata": {}
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! ls *bismark.cov.gz | xargs -tI{} gzip -d {}\n",
        "! head s_8_cell.deduplicated.bismark.cov"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "fcS724jEncto",
        "outputId": "6c4dcd9d-843e-4255-c944-1e711e2f7a45"
      },
      "execution_count": 49,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "gzip -d s_8_cell.deduplicated.bismark.cov.gz\n",
            "gzip -d s_epiblast.deduplicated.bismark.cov.gz\n",
            "gzip -d s_icm.deduplicated.bismark.cov.gz\n",
            "chr11\t3050031\t3050031\t50\t1\t1\n",
            "chr11\t3050048\t3050048\t66.6666666666667\t2\t1\n",
            "chr11\t3050049\t3050049\t50\t1\t1\n",
            "chr11\t3050224\t3050224\t87.5\t7\t1\n",
            "chr11\t3050225\t3050225\t53.3333333333333\t8\t7\n",
            "chr11\t3050286\t3050286\t66.6666666666667\t6\t3\n",
            "chr11\t3050287\t3050287\t33.3333333333333\t5\t10\n",
            "chr11\t3050310\t3050310\t62.5\t5\t3\n",
            "chr11\t3050311\t3050311\t50\t7\t7\n",
            "chr11\t3050349\t3050349\t55.5555555555556\t5\t4\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! apt-get update\n",
        "! apt-get install perl cpanminus\n",
        "! cpanm GD::Graph"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "GSSbgxEqTgLn",
        "outputId": "edffb318-2a3a-4120-cd6c-f1f218839e3f"
      },
      "execution_count": 56,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\r0% [Working]\r            \rGet:1 https://cloud.r-project.org/bin/linux/ubuntu jammy-cran40/ InRelease [3,626 B]\n",
            "Hit:2 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  InRelease\n",
            "Hit:3 http://archive.ubuntu.com/ubuntu jammy InRelease\n",
            "Get:4 http://security.ubuntu.com/ubuntu jammy-security InRelease [129 kB]\n",
            "Get:5 http://archive.ubuntu.com/ubuntu jammy-updates InRelease [128 kB]\n",
            "Hit:6 https://ppa.launchpadcontent.net/c2d4u.team/c2d4u4.0+/ubuntu jammy InRelease\n",
            "Hit:7 http://archive.ubuntu.com/ubuntu jammy-backports InRelease\n",
            "Hit:8 https://ppa.launchpadcontent.net/deadsnakes/ppa/ubuntu jammy InRelease\n",
            "Hit:9 https://ppa.launchpadcontent.net/graphics-drivers/ppa/ubuntu jammy InRelease\n",
            "Hit:10 https://ppa.launchpadcontent.net/ubuntugis/ppa/ubuntu jammy InRelease\n",
            "Fetched 261 kB in 2s (119 kB/s)\n",
            "Reading package lists... Done\n",
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "cpanminus is already the newest version (1.7045-1).\n",
            "perl is already the newest version (5.34.0-3ubuntu1.3).\n",
            "perl set to manually installed.\n",
            "0 upgraded, 0 newly installed, 0 to remove and 42 not upgraded.\n",
            "--> Working on GD::Graph\n",
            "Fetching http://www.cpan.org/authors/id/B/BP/BPS/GDGraph-1.56.tar.gz ... OK\n",
            "/usr/bin/tar: Ignoring unknown extended header keyword 'LIBARCHIVE.xattr.com.macromates.visibleIndex'\n",
            "/usr/bin/tar: Ignoring unknown extended header keyword 'LIBARCHIVE.xattr.com.macromates.selectionRange'\n",
            "/usr/bin/tar: Ignoring unknown extended header keyword 'LIBARCHIVE.xattr.com.macromates.visibleIndex'\n",
            "/usr/bin/tar: Ignoring unknown extended header keyword 'LIBARCHIVE.xattr.com.macromates.selectionRange'\n",
            "/usr/bin/tar: Ignoring unknown extended header keyword 'LIBARCHIVE.xattr.com.macromates.visibleIndex'\n",
            "/usr/bin/tar: Ignoring unknown extended header keyword 'LIBARCHIVE.xattr.com.macromates.selectionRange'\n",
            "/usr/bin/tar: Ignoring unknown extended header keyword 'LIBARCHIVE.xattr.com.macromates.visibleIndex'\n",
            "/usr/bin/tar: Ignoring unknown extended header keyword 'LIBARCHIVE.xattr.com.macromates.selectionRange'\n",
            "/usr/bin/tar: Ignoring unknown extended header keyword 'LIBARCHIVE.xattr.com.macromates.visibleIndex'\n",
            "/usr/bin/tar: Ignoring unknown extended header keyword 'LIBARCHIVE.xattr.com.macromates.selectionRange'\n",
            "/usr/bin/tar: Ignoring unknown extended header keyword 'LIBARCHIVE.xattr.com.macromates.visibleIndex'\n",
            "/usr/bin/tar: Ignoring unknown extended header keyword 'LIBARCHIVE.xattr.com.macromates.selectionRange'\n",
            "/usr/bin/tar: Ignoring unknown extended header keyword 'LIBARCHIVE.xattr.com.macromates.visibleIndex'\n",
            "/usr/bin/tar: Ignoring unknown extended header keyword 'LIBARCHIVE.xattr.com.macromates.selectionRange'\n",
            "/usr/bin/tar: Ignoring unknown extended header keyword 'LIBARCHIVE.xattr.com.macromates.visibleIndex'\n",
            "/usr/bin/tar: Ignoring unknown extended header keyword 'LIBARCHIVE.xattr.com.macromates.selectionRange'\n",
            "/usr/bin/tar: Ignoring unknown extended header keyword 'LIBARCHIVE.xattr.com.macromates.visibleIndex'\n",
            "/usr/bin/tar: Ignoring unknown extended header keyword 'LIBARCHIVE.xattr.com.macromates.selectionRange'\n",
            "/usr/bin/tar: Ignoring unknown extended header keyword 'LIBARCHIVE.xattr.com.macromates.visibleIndex'\n",
            "/usr/bin/tar: Ignoring unknown extended header keyword 'LIBARCHIVE.xattr.com.macromates.selectionRange'\n",
            "Configuring GDGraph-1.56 ... OK\n",
            "==> Found dependencies: GD::Text, GD\n",
            "--> Working on GD::Text\n",
            "Fetching http://www.cpan.org/authors/id/M/MV/MVERB/GDTextUtil-0.86.tar.gz ... OK\n",
            "Configuring GDTextUtil-0.86 ... OK\n",
            "==> Found dependencies: GD\n",
            "--> Working on GD\n",
            "Fetching http://www.cpan.org/authors/id/R/RU/RURBAN/GD-2.83.tar.gz ... OK\n",
            "Configuring GD-2.83 ... N/A\n",
            "! Configure failed for GD-2.83. See /root/.cpanm/work/1719338124.134829/build.log for details.\n",
            "! Installing the dependencies failed: Module 'GD' is not installed\n",
            "! Bailing out the installation for GDTextUtil-0.86.\n",
            "! Installing the dependencies failed: Module 'GD' is not installed, Module 'GD::Text' is not installed\n",
            "! Bailing out the installation for GDGraph-1.56.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! bismark_methylation_extractor --mbias_only --gzip --output ./ --comprehensive SRR3824222_1_bismark_bt2_PE_report.txt SRR5836473_1_bismark_bt2_PE_report.txt SRR5836475_1_bismark_bt2_PE_report.txt"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "S785xmoiUCk8",
        "outputId": "ca355d97-68d0-41ee-e016-16a48e9c06b0"
      },
      "execution_count": 52,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            " *** Bismark methylation extractor version v0.24.2 ***\n",
            "\n",
            "Option '--mbias_only' skips all sorts of methylation extraction, thus '--comprehensive' won't have any effect\n",
            "Output will be written into the directory: /content/\n",
            "Trying to determine the type of mapping from the SAM header line of file SRR3824222_1_bismark_bt2_PE_report.txt\n",
            "\n",
            "gzip: SRR3824222_1_bismark_bt2_PE_report.txt: not in gzip format\n",
            "Setting core usage to single-threaded (default). Consider using --multicore <int> to speed up the extraction process.\n",
            "\n",
            "Summarising Bismark methylation extractor parameters:\n",
            "===============================================================\n",
            "Number of cores to be used: 1\n",
            "Strand-specific outputs will be skipped. Separate output files for cytosines in CpG, CHG and CHH context will be generated\n",
            "Output path specified as: /content/\n",
            "\n",
            "Now reading in Bismark result file SRR3824222_1_bismark_bt2_PE_report.txt\n",
            "\n",
            "gzip: SRR3824222_1_bismark_bt2_PE_report.txt: not in gzip format\n",
            "Single-end or paired-end reads not specified properly\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "b6k9REsM6NXn",
        "outputId": "31c935ba-fab9-402c-bec7-0ccbecdb48eb"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Requirement already satisfied: pyGenomeTracks in /usr/local/lib/python3.7/site-packages (3.6)\n",
            "Requirement already satisfied: matplotlib<=3.3.2,>=3.1.1 in /usr/local/lib/python3.7/site-packages (from pyGenomeTracks) (3.3.2)\n",
            "Requirement already satisfied: tqdm>=4.20 in /usr/local/lib/python3.7/site-packages (from pyGenomeTracks) (4.59.0)\n",
            "Requirement already satisfied: future>=0.17.0 in /usr/local/lib/python3.7/site-packages (from pyGenomeTracks) (0.18.2)\n",
            "Requirement already satisfied: pytest in /usr/local/lib/python3.7/site-packages (from pyGenomeTracks) (6.2.5)\n",
            "Requirement already satisfied: numpy>=1.16 in /usr/local/lib/python3.7/site-packages (from pyGenomeTracks) (1.21.5)\n",
            "Requirement already satisfied: pybedtools>=0.8.1 in /usr/local/lib/python3.7/site-packages (from pyGenomeTracks) (0.9.0)\n",
            "Requirement already satisfied: intervaltree>=2.1.0 in /usr/local/lib/python3.7/site-packages (from pyGenomeTracks) (3.1.0)\n",
            "Requirement already satisfied: pysam>=0.14 in /usr/local/lib/python3.7/site-packages (from pyGenomeTracks) (0.18.0)\n",
            "Requirement already satisfied: hicmatrix>=15 in /usr/local/lib/python3.7/site-packages (from pyGenomeTracks) (15)\n",
            "Requirement already satisfied: gffutils>=0.9 in /usr/local/lib/python3.7/site-packages (from pyGenomeTracks) (0.10.1)\n",
            "Requirement already satisfied: pyBigWig>=0.3.16 in /usr/local/lib/python3.7/site-packages (from pyGenomeTracks) (0.3.18)\n",
            "Requirement already satisfied: simplejson in /usr/local/lib/python3.7/site-packages (from gffutils>=0.9->pyGenomeTracks) (3.17.6)\n",
            "Requirement already satisfied: argcomplete>=1.9.4 in /usr/local/lib/python3.7/site-packages (from gffutils>=0.9->pyGenomeTracks) (2.0.0)\n",
            "Requirement already satisfied: pyfaidx>=0.5.5.2 in /usr/local/lib/python3.7/site-packages (from gffutils>=0.9->pyGenomeTracks) (0.6.4)\n",
            "Requirement already satisfied: six>=1.12.0 in /usr/local/lib/python3.7/site-packages (from gffutils>=0.9->pyGenomeTracks) (1.15.0)\n",
            "Requirement already satisfied: argh>=0.26.2 in /usr/local/lib/python3.7/site-packages (from gffutils>=0.9->pyGenomeTracks) (0.26.2)\n",
            "Requirement already satisfied: importlib-metadata<5,>=0.23 in /usr/local/lib/python3.7/site-packages (from argcomplete>=1.9.4->gffutils>=0.9->pyGenomeTracks) (4.10.1)\n",
            "Requirement already satisfied: tables>=3.5.* in /usr/local/lib/python3.7/site-packages (from hicmatrix>=15->pyGenomeTracks) (3.7.0)\n",
            "Requirement already satisfied: scipy>=1.2.* in /usr/local/lib/python3.7/site-packages (from hicmatrix>=15->pyGenomeTracks) (1.7.3)\n",
            "Requirement already satisfied: cooler>=0.8.9 in /usr/local/lib/python3.7/site-packages (from hicmatrix>=15->pyGenomeTracks) (0.8.11)\n",
            "Requirement already satisfied: pandas>=0.25.* in /usr/local/lib/python3.7/site-packages (from hicmatrix>=15->pyGenomeTracks) (1.3.5)\n",
            "Requirement already satisfied: asciitree in /usr/local/lib/python3.7/site-packages (from cooler>=0.8.9->hicmatrix>=15->pyGenomeTracks) (0.3.3)\n",
            "Requirement already satisfied: multiprocess in /usr/local/lib/python3.7/site-packages (from cooler>=0.8.9->hicmatrix>=15->pyGenomeTracks) (0.70.12.2)\n",
            "Requirement already satisfied: pypairix in /usr/local/lib/python3.7/site-packages (from cooler>=0.8.9->hicmatrix>=15->pyGenomeTracks) (0.3.7)\n",
            "Requirement already satisfied: h5py>=2.5 in /usr/local/lib/python3.7/site-packages (from cooler>=0.8.9->hicmatrix>=15->pyGenomeTracks) (3.6.0)\n",
            "Requirement already satisfied: click>=7 in /usr/local/lib/python3.7/site-packages (from cooler>=0.8.9->hicmatrix>=15->pyGenomeTracks) (8.0.3)\n",
            "Requirement already satisfied: cytoolz<0.11 in /usr/local/lib/python3.7/site-packages (from cooler>=0.8.9->hicmatrix>=15->pyGenomeTracks) (0.10.1)\n",
            "Requirement already satisfied: pyyaml in /usr/local/lib/python3.7/site-packages (from cooler>=0.8.9->hicmatrix>=15->pyGenomeTracks) (6.0)\n",
            "Requirement already satisfied: toolz>=0.8.0 in /usr/local/lib/python3.7/site-packages (from cytoolz<0.11->cooler>=0.8.9->hicmatrix>=15->pyGenomeTracks) (0.11.2)\n",
            "Requirement already satisfied: cached-property in /usr/local/lib/python3.7/site-packages (from h5py>=2.5->cooler>=0.8.9->hicmatrix>=15->pyGenomeTracks) (1.5.2)\n",
            "Requirement already satisfied: typing-extensions>=3.6.4 in /usr/local/lib/python3.7/site-packages (from importlib-metadata<5,>=0.23->argcomplete>=1.9.4->gffutils>=0.9->pyGenomeTracks) (4.0.1)\n",
            "Requirement already satisfied: zipp>=0.5 in /usr/local/lib/python3.7/site-packages (from importlib-metadata<5,>=0.23->argcomplete>=1.9.4->gffutils>=0.9->pyGenomeTracks) (3.7.0)\n",
            "Requirement already satisfied: sortedcontainers<3.0,>=2.0 in /usr/local/lib/python3.7/site-packages (from intervaltree>=2.1.0->pyGenomeTracks) (2.4.0)\n",
            "Requirement already satisfied: certifi>=2020.06.20 in /usr/local/lib/python3.7/site-packages (from matplotlib<=3.3.2,>=3.1.1->pyGenomeTracks) (2021.10.8)\n",
            "Requirement already satisfied: pyparsing!=2.0.4,!=2.1.2,!=2.1.6,>=2.0.3 in /usr/local/lib/python3.7/site-packages (from matplotlib<=3.3.2,>=3.1.1->pyGenomeTracks) (3.0.7)\n",
            "Requirement already satisfied: python-dateutil>=2.1 in /usr/local/lib/python3.7/site-packages (from matplotlib<=3.3.2,>=3.1.1->pyGenomeTracks) (2.8.2)\n",
            "Requirement already satisfied: cycler>=0.10 in /usr/local/lib/python3.7/site-packages (from matplotlib<=3.3.2,>=3.1.1->pyGenomeTracks) (0.11.0)\n",
            "Requirement already satisfied: pillow>=6.2.0 in /usr/local/lib/python3.7/site-packages (from matplotlib<=3.3.2,>=3.1.1->pyGenomeTracks) (9.0.1)\n",
            "Requirement already satisfied: kiwisolver>=1.0.1 in /usr/local/lib/python3.7/site-packages (from matplotlib<=3.3.2,>=3.1.1->pyGenomeTracks) (1.3.2)\n",
            "Requirement already satisfied: pytz>=2017.3 in /usr/local/lib/python3.7/site-packages (from pandas>=0.25.*->hicmatrix>=15->pyGenomeTracks) (2021.3)\n",
            "Requirement already satisfied: setuptools>=0.7 in /usr/local/lib/python3.7/site-packages (from pyfaidx>=0.5.5.2->gffutils>=0.9->pyGenomeTracks) (49.6.0.post20210108)\n",
            "Requirement already satisfied: numexpr>=2.6.2 in /usr/local/lib/python3.7/site-packages (from tables>=3.5.*->hicmatrix>=15->pyGenomeTracks) (2.8.1)\n",
            "Requirement already satisfied: packaging in /usr/local/lib/python3.7/site-packages (from tables>=3.5.*->hicmatrix>=15->pyGenomeTracks) (21.3)\n",
            "Requirement already satisfied: dill>=0.3.4 in /usr/local/lib/python3.7/site-packages (from multiprocess->cooler>=0.8.9->hicmatrix>=15->pyGenomeTracks) (0.3.4)\n",
            "Requirement already satisfied: pluggy<2.0,>=0.12 in /usr/local/lib/python3.7/site-packages (from pytest->pyGenomeTracks) (1.0.0)\n",
            "Requirement already satisfied: py>=1.8.2 in /usr/local/lib/python3.7/site-packages (from pytest->pyGenomeTracks) (1.11.0)\n",
            "Requirement already satisfied: toml in /usr/local/lib/python3.7/site-packages (from pytest->pyGenomeTracks) (0.10.2)\n",
            "Requirement already satisfied: iniconfig in /usr/local/lib/python3.7/site-packages (from pytest->pyGenomeTracks) (1.1.1)\n",
            "Requirement already satisfied: attrs>=19.2.0 in /usr/local/lib/python3.7/site-packages (from pytest->pyGenomeTracks) (21.4.0)\n"
          ]
        }
      ],
      "source": [
        "! pip install pyGenomeTracks"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!pip install -U ipykernel\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "fcsJWNmHo1i-",
        "outputId": "e96a1642-abff-46ba-fede-0766677b2143"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting ipykernel\n",
            "  Downloading ipykernel-6.8.0-py3-none-any.whl (128 kB)\n",
            "\u001b[?25l\r\u001b[K     |██▌                             | 10 kB 20.9 MB/s eta 0:00:01\r\u001b[K     |█████                           | 20 kB 24.6 MB/s eta 0:00:01\r\u001b[K     |███████▋                        | 30 kB 18.4 MB/s eta 0:00:01\r\u001b[K     |██████████▏                     | 40 kB 15.0 MB/s eta 0:00:01\r\u001b[K     |████████████▊                   | 51 kB 8.7 MB/s eta 0:00:01\r\u001b[K     |███████████████▎                | 61 kB 9.4 MB/s eta 0:00:01\r\u001b[K     |█████████████████▉              | 71 kB 8.6 MB/s eta 0:00:01\r\u001b[K     |████████████████████▍           | 81 kB 9.5 MB/s eta 0:00:01\r\u001b[K     |███████████████████████         | 92 kB 10.1 MB/s eta 0:00:01\r\u001b[K     |█████████████████████████▌      | 102 kB 8.3 MB/s eta 0:00:01\r\u001b[K     |████████████████████████████    | 112 kB 8.3 MB/s eta 0:00:01\r\u001b[K     |██████████████████████████████▋ | 122 kB 8.3 MB/s eta 0:00:01\r\u001b[K     |████████████████████████████████| 128 kB 8.3 MB/s \n",
            "\u001b[?25hCollecting nest-asyncio\n",
            "  Downloading nest_asyncio-1.5.4-py3-none-any.whl (5.1 kB)\n",
            "Collecting matplotlib-inline<0.2.0,>=0.1.0\n",
            "  Downloading matplotlib_inline-0.1.3-py3-none-any.whl (8.2 kB)\n",
            "Collecting tornado<7.0,>=4.2\n",
            "  Downloading tornado-6.1-cp37-cp37m-manylinux2010_x86_64.whl (428 kB)\n",
            "\u001b[K     |████████████████████████████████| 428 kB 50.9 MB/s \n",
            "\u001b[?25hCollecting debugpy<2.0,>=1.0.0\n",
            "  Downloading debugpy-1.5.1-cp37-cp37m-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl (1.9 MB)\n",
            "\u001b[K     |████████████████████████████████| 1.9 MB 51.5 MB/s \n",
            "\u001b[?25hCollecting jupyter-client<8.0\n",
            "  Downloading jupyter_client-7.1.2-py3-none-any.whl (130 kB)\n",
            "\u001b[K     |████████████████████████████████| 130 kB 54.6 MB/s \n",
            "\u001b[?25hCollecting ipython>=7.23.1\n",
            "  Downloading ipython-7.31.1-py3-none-any.whl (792 kB)\n",
            "\u001b[K     |████████████████████████████████| 792 kB 53.2 MB/s \n",
            "\u001b[?25hCollecting traitlets<6.0,>=5.1.0\n",
            "  Downloading traitlets-5.1.1-py3-none-any.whl (102 kB)\n",
            "\u001b[K     |████████████████████████████████| 102 kB 12.1 MB/s \n",
            "\u001b[?25hCollecting prompt-toolkit!=3.0.0,!=3.0.1,<3.1.0,>=2.0.0\n",
            "  Downloading prompt_toolkit-3.0.26-py3-none-any.whl (375 kB)\n",
            "\u001b[K     |████████████████████████████████| 375 kB 54.0 MB/s \n",
            "\u001b[?25hCollecting backcall\n",
            "  Downloading backcall-0.2.0-py2.py3-none-any.whl (11 kB)\n",
            "Collecting decorator\n",
            "  Downloading decorator-5.1.1-py3-none-any.whl (9.1 kB)\n",
            "Collecting jedi>=0.16\n",
            "  Downloading jedi-0.18.1-py2.py3-none-any.whl (1.6 MB)\n",
            "\u001b[K     |████████████████████████████████| 1.6 MB 56.1 MB/s \n",
            "\u001b[?25hRequirement already satisfied: setuptools>=18.5 in /usr/local/lib/python3.7/site-packages (from ipython>=7.23.1->ipykernel) (60.7.0)\n",
            "Collecting pygments\n",
            "  Downloading Pygments-2.11.2-py3-none-any.whl (1.1 MB)\n",
            "\u001b[K     |████████████████████████████████| 1.1 MB 60.1 MB/s \n",
            "\u001b[?25hCollecting pickleshare\n",
            "  Downloading pickleshare-0.7.5-py2.py3-none-any.whl (6.9 kB)\n",
            "Collecting pexpect>4.3\n",
            "  Downloading pexpect-4.8.0-py2.py3-none-any.whl (59 kB)\n",
            "\u001b[K     |████████████████████████████████| 59 kB 6.9 MB/s \n",
            "\u001b[?25hCollecting parso<0.9.0,>=0.8.0\n",
            "  Downloading parso-0.8.3-py2.py3-none-any.whl (100 kB)\n",
            "\u001b[K     |████████████████████████████████| 100 kB 12.3 MB/s \n",
            "\u001b[?25hCollecting pyzmq>=13\n",
            "  Downloading pyzmq-22.3.0-cp37-cp37m-manylinux_2_5_x86_64.manylinux1_x86_64.whl (1.1 MB)\n",
            "\u001b[K     |████████████████████████████████| 1.1 MB 68.4 MB/s \n",
            "\u001b[?25hRequirement already satisfied: python-dateutil>=2.1 in /usr/local/lib/python3.7/site-packages (from jupyter-client<8.0->ipykernel) (2.8.2)\n",
            "Collecting entrypoints\n",
            "  Downloading entrypoints-0.4-py3-none-any.whl (5.3 kB)\n",
            "Collecting jupyter-core>=4.6.0\n",
            "  Downloading jupyter_core-4.9.1-py3-none-any.whl (86 kB)\n",
            "\u001b[K     |████████████████████████████████| 86 kB 6.2 MB/s \n",
            "\u001b[?25hCollecting ptyprocess>=0.5\n",
            "  Downloading ptyprocess-0.7.0-py2.py3-none-any.whl (13 kB)\n",
            "Collecting wcwidth\n",
            "  Downloading wcwidth-0.2.5-py2.py3-none-any.whl (30 kB)\n",
            "Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.7/site-packages (from python-dateutil>=2.1->jupyter-client<8.0->ipykernel) (1.16.0)\n",
            "Installing collected packages: wcwidth, traitlets, ptyprocess, parso, tornado, pyzmq, pygments, prompt-toolkit, pickleshare, pexpect, nest-asyncio, matplotlib-inline, jupyter-core, jedi, entrypoints, decorator, backcall, jupyter-client, ipython, debugpy, ipykernel\n",
            "Successfully installed backcall-0.2.0 debugpy-1.5.1 decorator-5.1.1 entrypoints-0.4 ipykernel-6.8.0 ipython-7.31.1 jedi-0.18.1 jupyter-client-7.1.2 jupyter-core-4.9.1 matplotlib-inline-0.1.3 nest-asyncio-1.5.4 parso-0.8.3 pexpect-4.8.0 pickleshare-0.7.5 prompt-toolkit-3.0.26 ptyprocess-0.7.0 pygments-2.11.2 pyzmq-22.3.0 tornado-6.1 traitlets-5.1.1 wcwidth-0.2.5\n"
          ]
        },
        {
          "output_type": "display_data",
          "data": {
            "application/vnd.colab-display-data+json": {
              "pip_warning": {
                "packages": [
                  "jupyter_core",
                  "pexpect",
                  "pickleshare",
                  "traitlets",
                  "wcwidth",
                  "zmq"
                ]
              }
            }
          },
          "metadata": {}
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! conda install -c bioconda ucsc-fetchchromsizes"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "UhIKyKz_t7yy",
        "outputId": "607dfac5-3f6d-4ee7-97b2-a68677691a06"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
            "Solving environment: / \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local\n",
            "\n",
            "  added / updated specs:\n",
            "    - ucsc-fetchchromsizes\n",
            "\n",
            "\n",
            "The following packages will be downloaded:\n",
            "\n",
            "    package                    |            build\n",
            "    ---------------------------|-----------------\n",
            "    ucsc-fetchchromsizes-377   |       ha8a8165_3           8 KB  bioconda\n",
            "    ------------------------------------------------------------\n",
            "                                           Total:           8 KB\n",
            "\n",
            "The following NEW packages will be INSTALLED:\n",
            "\n",
            "  ucsc-fetchchromsi~ bioconda/linux-64::ucsc-fetchchromsizes-377-ha8a8165_3\n",
            "\n",
            "\n",
            "\n",
            "Downloading and Extracting Packages\n",
            "ucsc-fetchchromsizes | 8 KB      | : 100% 1.0/1 [00:00<00:00,  4.60it/s]\n",
            "Preparing transaction: \\ \b\bdone\n",
            "Verifying transaction: / \b\bdone\n",
            "Executing transaction: \\ \b\bdone\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! fetchChromSizes mm10 > m.chrom.sizes"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "LZpspVUvt4EY",
        "outputId": "343d4ea5-ebce-42e2-dc7a-356cb3c96be9"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "INFO: trying CURL  for database mm10\n",
            "url: http://hgdownload.soe.ucsc.edu/goldenPath/mm10/bigZips/mm10.chrom.sizes\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! conda install -c bioconda ucsc-bedgraphtobigwig"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "svn9iAuGtVxX",
        "outputId": "d743482d-07bf-4f24-fd49-cc3b4a264d96"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n",
            "Solving environment: - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local\n",
            "\n",
            "  added / updated specs:\n",
            "    - ucsc-bedgraphtobigwig\n",
            "\n",
            "\n",
            "The following packages will be downloaded:\n",
            "\n",
            "    package                    |            build\n",
            "    ---------------------------|-----------------\n",
            "    libuuid-2.32.1             |    h7f98852_1000          28 KB  conda-forge\n",
            "    mysql-connector-c-6.1.11   |    h6eb9d5d_1007         2.7 MB  conda-forge\n",
            "    ucsc-bedgraphtobigwig-377  |       ha8a8165_3         155 KB  bioconda\n",
            "    ------------------------------------------------------------\n",
            "                                           Total:         2.8 MB\n",
            "\n",
            "The following NEW packages will be INSTALLED:\n",
            "\n",
            "  libpng             conda-forge/linux-64::libpng-1.6.37-h21135ba_2\n",
            "  libuuid            conda-forge/linux-64::libuuid-2.32.1-h7f98852_1000\n",
            "  mysql-connector-c  conda-forge/linux-64::mysql-connector-c-6.1.11-h6eb9d5d_1007\n",
            "  ucsc-bedgraphtobi~ bioconda/linux-64::ucsc-bedgraphtobigwig-377-ha8a8165_3\n",
            "\n",
            "\n",
            "\n",
            "Downloading and Extracting Packages\n",
            "libuuid-2.32.1       | 28 KB     | : 100% 1.0/1 [00:00<00:00,  6.53it/s]              \n",
            "ucsc-bedgraphtobigwi | 155 KB    | : 100% 1.0/1 [00:00<00:00,  9.37it/s]\n",
            "mysql-connector-c-6. | 2.7 MB    | : 100% 1.0/1 [00:00<00:00,  1.88it/s]               \n",
            "Preparing transaction: \\ \b\bdone\n",
            "Verifying transaction: / \b\b- \b\bdone\n",
            "Executing transaction: | \b\bdone\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! bedGraphToBigWig  s_8_cell.deduplicated.bedGraph m.chrom.sizes cell8_methylation.bigWig"
      ],
      "metadata": {
        "id": "rh0ba_Y_tcq5"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "! head s_8_cell.deduplicated.bedGraph"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "qofYnYciunf8",
        "outputId": "cad6d289-ceaf-48bf-be19-6715b8a6155d"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "track type=bedGraph\n",
            "chr11\t3100030\t3100031\t50\n",
            "chr11\t3100047\t3100048\t66.6666666666667\n",
            "chr11\t3100048\t3100049\t50\n",
            "chr11\t3100223\t3100224\t87.5\n",
            "chr11\t3100224\t3100225\t53.3333333333333\n",
            "chr11\t3100285\t3100286\t66.6666666666667\n",
            "chr11\t3100286\t3100287\t33.3333333333333\n",
            "chr11\t3100309\t3100310\t62.5\n",
            "chr11\t3100310\t3100311\t50\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! apt-get install bedtools"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "XEWV8p4ku-Hy",
        "outputId": "596dfae5-029a-4a64-efe0-fc42fdabfc3e"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "The following packages were automatically installed and are no longer required:\n",
            "  cuda-command-line-tools-10-0 cuda-command-line-tools-10-1\n",
            "  cuda-command-line-tools-11-0 cuda-compiler-10-0 cuda-compiler-10-1\n",
            "  cuda-compiler-11-0 cuda-cuobjdump-10-0 cuda-cuobjdump-10-1\n",
            "  cuda-cuobjdump-11-0 cuda-cupti-10-0 cuda-cupti-10-1 cuda-cupti-11-0\n",
            "  cuda-cupti-dev-11-0 cuda-documentation-10-0 cuda-documentation-10-1\n",
            "  cuda-documentation-11-0 cuda-documentation-11-1 cuda-gdb-10-0 cuda-gdb-10-1\n",
            "  cuda-gdb-11-0 cuda-gpu-library-advisor-10-0 cuda-gpu-library-advisor-10-1\n",
            "  cuda-libraries-10-0 cuda-libraries-10-1 cuda-libraries-11-0\n",
            "  cuda-memcheck-10-0 cuda-memcheck-10-1 cuda-memcheck-11-0 cuda-nsight-10-0\n",
            "  cuda-nsight-10-1 cuda-nsight-11-0 cuda-nsight-11-1 cuda-nsight-compute-10-0\n",
            "  cuda-nsight-compute-10-1 cuda-nsight-compute-11-0 cuda-nsight-compute-11-1\n",
            "  cuda-nsight-systems-10-1 cuda-nsight-systems-11-0 cuda-nsight-systems-11-1\n",
            "  cuda-nvcc-10-0 cuda-nvcc-10-1 cuda-nvcc-11-0 cuda-nvdisasm-10-0\n",
            "  cuda-nvdisasm-10-1 cuda-nvdisasm-11-0 cuda-nvml-dev-10-0 cuda-nvml-dev-10-1\n",
            "  cuda-nvml-dev-11-0 cuda-nvprof-10-0 cuda-nvprof-10-1 cuda-nvprof-11-0\n",
            "  cuda-nvprune-10-0 cuda-nvprune-10-1 cuda-nvprune-11-0 cuda-nvtx-10-0\n",
            "  cuda-nvtx-10-1 cuda-nvtx-11-0 cuda-nvvp-10-0 cuda-nvvp-10-1 cuda-nvvp-11-0\n",
            "  cuda-nvvp-11-1 cuda-samples-10-0 cuda-samples-10-1 cuda-samples-11-0\n",
            "  cuda-samples-11-1 cuda-sanitizer-11-0 cuda-sanitizer-api-10-1\n",
            "  cuda-toolkit-10-0 cuda-toolkit-10-1 cuda-toolkit-11-0 cuda-toolkit-11-1\n",
            "  cuda-tools-10-0 cuda-tools-10-1 cuda-tools-11-0 cuda-tools-11-1\n",
            "  cuda-visual-tools-10-0 cuda-visual-tools-10-1 cuda-visual-tools-11-0\n",
            "  cuda-visual-tools-11-1 default-jre dkms freeglut3 freeglut3-dev\n",
            "  keyboard-configuration libargon2-0 libcap2 libcryptsetup12\n",
            "  libdevmapper1.02.1 libfontenc1 libidn11 libip4tc0 libjansson4\n",
            "  libnvidia-cfg1-510 libnvidia-common-460 libnvidia-common-510\n",
            "  libnvidia-extra-510 libnvidia-fbc1-510 libnvidia-gl-510 libpam-systemd\n",
            "  libpolkit-agent-1-0 libpolkit-backend-1-0 libpolkit-gobject-1-0 libxfont2\n",
            "  libxi-dev libxkbfile1 libxmu-dev libxmu-headers libxnvctrl0 libxtst6\n",
            "  nsight-compute-2020.2.1 nsight-compute-2022.1.0 nsight-systems-2020.3.2\n",
            "  nsight-systems-2020.3.4 nsight-systems-2021.5.2 nvidia-dkms-510\n",
            "  nvidia-kernel-common-510 nvidia-kernel-source-510 nvidia-modprobe\n",
            "  nvidia-settings openjdk-11-jre policykit-1 policykit-1-gnome python3-xkit\n",
            "  screen-resolution-extra systemd systemd-sysv udev x11-xkb-utils\n",
            "  xserver-common xserver-xorg-core-hwe-18.04 xserver-xorg-video-nvidia-510\n",
            "Use 'apt autoremove' to remove them.\n",
            "The following NEW packages will be installed:\n",
            "  bedtools\n",
            "0 upgraded, 1 newly installed, 0 to remove and 39 not upgraded.\n",
            "Need to get 577 kB of archives.\n",
            "After this operation, 2,040 kB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu bionic/universe amd64 bedtools amd64 2.26.0+dfsg-5 [577 kB]\n",
            "Fetched 577 kB in 1s (489 kB/s)\n",
            "Selecting previously unselected package bedtools.\n",
            "(Reading database ... 155113 files and directories currently installed.)\n",
            "Preparing to unpack .../bedtools_2.26.0+dfsg-5_amd64.deb ...\n",
            "Unpacking bedtools (2.26.0+dfsg-5) ...\n",
            "Setting up bedtools (2.26.0+dfsg-5) ...\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! bedtools genomecov -bg -ibam s_8_cell.deduplicated.bam -g m.chromSizes  >  s_8_cell.cover.bedGraph"
      ],
      "metadata": {
        "id": "ofhSsnSqvBbu"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "! bedGraphToBigWig   s_8_cell.cover.bedGraph m.chrom.sizes cell8_coverage.bigWig"
      ],
      "metadata": {
        "id": "RLpT4wFCvr29"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "! head s_8_cell.deduplicated.bedGraph"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "0egk8Q4VvXss",
        "outputId": "a71cb456-70e7-4ab2-8f85-d6868b6ead94"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "track type=bedGraph\n",
            "chr11\t3100030\t3100031\t50\n",
            "chr11\t3100047\t3100048\t66.6666666666667\n",
            "chr11\t3100048\t3100049\t50\n",
            "chr11\t3100223\t3100224\t87.5\n",
            "chr11\t3100224\t3100225\t53.3333333333333\n",
            "chr11\t3100285\t3100286\t66.6666666666667\n",
            "chr11\t3100286\t3100287\t33.3333333333333\n",
            "chr11\t3100309\t3100310\t62.5\n",
            "chr11\t3100310\t3100311\t50\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "! pyGenomeTracks --tracks tracks.ini.txt --region chr11:3100030-3500030 -o image_cov.png"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "bAaOdZCKsVtX",
        "outputId": "87e15cec-5977-4453-dffa-97ad89796c34"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "INFO:pygenometracks.tracksClass:initialize 1. [bigwig file test]\n",
            "INFO:pygenometracks.tracksClass:initialize 2. [test bedgraph]\n",
            "100% 7810/7810 [00:00<00:00, 22280.01it/s]\n",
            "INFO:pygenometracks.tracksClass:time initializing track(s):\n",
            "INFO:pygenometracks.tracksClass:1.160700798034668\n",
            "DEBUG:pygenometracks.tracksClass:Figure size in cm is 40 x 9.574468085106384. Dpi is set to 72\n",
            "\n",
            "INFO:pygenometracks.tracksClass:plotting 1. [bigwig file test]\n",
            "INFO:pygenometracks.tracksClass:plotting 2. [test bedgraph]\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [],
      "metadata": {
        "id": "p7CjvnFhwpyL"
      },
      "execution_count": null,
      "outputs": []
    }
  ],
  "metadata": {
    "colab": {
      "provenance": []
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}